--- a/net/minecraft/core/dispenser/DispenseItemBehavior.java
+++ b/net/minecraft/core/dispenser/DispenseItemBehavior.java
@@ -461,9 +_,9 @@
                                 for (int i = 0; i < 5; i++) {
                                     serverlevel.sendParticles(
                                         ParticleTypes.SPLASH,
-                                        (double)blockpos.getX() + serverlevel.random.nextDouble(),
+                                        (double)blockpos.getBigX().doubleValue() + serverlevel.random.nextDouble(),
                                         (double)(blockpos.getY() + 1),
-                                        (double)blockpos.getZ() + serverlevel.random.nextDouble(),
+                                        (double)blockpos.getBigZ().doubleValue() + serverlevel.random.nextDouble(),
                                         1,
                                         0.0,
                                         0.0,
