--- a/net/minecraft/world/level/border/WorldBorder.java
+++ b/net/minecraft/world/level/border/WorldBorder.java
@@ -30,7 +_,11 @@
     public static final WorldBorder.Settings DEFAULT_SETTINGS = new WorldBorder.Settings(0.0, 0.0, 0.2, 5.0, 5, 15, 5.999997E7F, 0L, 0.0);
 
     public boolean isWithinBounds(BlockPos p_61938_) {
-        return this.isWithinBounds((double)p_61938_.getX(), (double)p_61938_.getZ());
+        return this.isWithinBounds((double)p_61938_.getBigX().doubleValue(), (double)p_61938_.getBigZ().doubleValue());
+    }
+
+    public boolean isWithinBounds(me.alphamode.mcbig.core.BigVec3 p_352959_) {
+        return this.isWithinBounds(p_352959_.x.doubleValue(), p_352959_.z.doubleValue());
     }
 
     public boolean isWithinBounds(Vec3 p_352959_) {
@@ -38,8 +_,8 @@
     }
 
     public boolean isWithinBounds(ChunkPos p_61928_) {
-        return this.isWithinBounds((double)p_61928_.getMinBlockX(), (double)p_61928_.getMinBlockZ())
-            && this.isWithinBounds((double)p_61928_.getMaxBlockX(), (double)p_61928_.getMaxBlockZ());
+        return this.isWithinBounds((double)p_61928_.getMinBlockX().doubleValue(), (double)p_61928_.getMinBlockZ().doubleValue())
+            && this.isWithinBounds((double)p_61928_.getMaxBlockX().doubleValue(), (double)p_61928_.getMaxBlockZ().doubleValue());
     }
 
     public boolean isWithinBounds(AABB p_61936_) {
@@ -62,7 +_,7 @@
     }
 
     public BlockPos clampToBounds(BlockPos p_352967_) {
-        return this.clampToBounds((double)p_352967_.getX(), (double)p_352967_.getY(), (double)p_352967_.getZ());
+        return this.clampToBounds((double)p_352967_.getBigX().doubleValue(), (double)p_352967_.getBigY().doubleValue(), (double)p_352967_.getBigZ().doubleValue());
     }
 
     public BlockPos clampToBounds(Vec3 p_352921_) {
@@ -77,6 +_,10 @@
         return this.clampVec3ToBound(p_362111_.x, p_362111_.y, p_362111_.z);
     }
 
+    public me.alphamode.mcbig.core.BigVec3 clampVec3ToBound(me.alphamode.mcbig.core.BigVec3 p_362111_) {
+        return new me.alphamode.mcbig.core.BigVec3(this.clampVec3ToBound(p_362111_.x.doubleValue(), p_362111_.y, p_362111_.z.doubleValue()));
+    }
+
     public Vec3 clampVec3ToBound(double p_362510_, double p_362166_, double p_360417_) {
         return new Vec3(
             Mth.clamp(p_362510_, this.getMinX(), this.getMaxX() - 1.0E-5F), p_362166_, Mth.clamp(p_360417_, this.getMinZ(), this.getMaxZ() - 1.0E-5F)
@@ -102,8 +_,8 @@
     }
 
     public boolean isInsideCloseToBorder(Entity p_187567_, AABB p_187568_) {
-        double d0 = Math.max(Mth.absMax(p_187568_.getXsize(), p_187568_.getZsize()), 1.0);
-        return this.getDistanceToBorder(p_187567_) < d0 * 2.0 && this.isWithinBounds(p_187567_.getX(), p_187567_.getZ(), d0);
+//        double d0 = Math.max(Mth.absMax(p_187568_.getXsize(), p_187568_.getZsize()), 1.0);
+        return false;//this.getDistanceToBorder(p_187567_) < d0 * 2.0 && this.isWithinBounds(p_187567_.getX(), p_187567_.getZ(), d0);
     }
 
     public BorderStatus getStatus() {
@@ -469,8 +_,8 @@
         }
 
         public static WorldBorder.Settings read(DynamicLike<?> p_62038_, WorldBorder.Settings p_62039_) {
-            double d0 = Mth.clamp(p_62038_.get("BorderCenterX").asDouble(p_62039_.centerX), -2.9999984E7, 2.9999984E7);
-            double d1 = Mth.clamp(p_62038_.get("BorderCenterZ").asDouble(p_62039_.centerZ), -2.9999984E7, 2.9999984E7);
+            double d0 = /*Mth.clamp(*/p_62038_.get("BorderCenterX").asDouble(p_62039_.centerX)/*, -2.9999984E7, 2.9999984E7)*/;
+            double d1 = /*Mth.clamp(*/p_62038_.get("BorderCenterZ").asDouble(p_62039_.centerZ)/*, -2.9999984E7, 2.9999984E7)*/;
             double d2 = p_62038_.get("BorderSize").asDouble(p_62039_.size);
             long i = p_62038_.get("BorderSizeLerpTime").asLong(p_62039_.sizeLerpTime);
             double d3 = p_62038_.get("BorderSizeLerpTarget").asDouble(p_62039_.sizeLerpTarget);
