--- a/net/minecraft/world/level/block/DoorBlock.java
+++ b/net/minecraft/world/level/block/DoorBlock.java
@@ -267,7 +_,7 @@
 
     @Override
     public long getSeed(BlockState p_52793_, BlockPos p_52794_) {
-        return Mth.getSeed(p_52794_.getX(), p_52794_.below(p_52793_.getValue(HALF) == DoubleBlockHalf.LOWER ? 0 : 1).getY(), p_52794_.getZ());
+        return Mth.getSeed(p_52794_.getBigX(), p_52794_.below(p_52793_.getValue(HALF) == DoubleBlockHalf.LOWER ? 0 : 1).getY(), p_52794_.getBigZ());
     }
 
     @Override
