--- a/net/minecraft/world/entity/ai/behavior/LongJumpToRandomPos.java
+++ b/net/minecraft/world/entity/ai/behavior/LongJumpToRandomPos.java
@@ -110,16 +_,16 @@
         this.findJumpTries = 20;
         this.initialPosition = Optional.of(p_147677_.position());
         BlockPos blockpos = p_147677_.blockPosition();
-        int i = blockpos.getX();
-        int j = blockpos.getY();
-        int k = blockpos.getZ();
+        me.alphamode.mcbig.math.BigInteger i = blockpos.getBigX();
+        me.alphamode.mcbig.math.BigInteger j = blockpos.getBigY();
+        me.alphamode.mcbig.math.BigInteger k = blockpos.getBigZ();
         this.jumpCandidates = BlockPos.betweenClosedStream(
-                i - this.maxLongJumpWidth,
-                j - this.maxLongJumpHeight,
-                k - this.maxLongJumpWidth,
-                i + this.maxLongJumpWidth,
-                j + this.maxLongJumpHeight,
-                k + this.maxLongJumpWidth
+                i.subtract(this.maxLongJumpWidth),
+                j.subtract(this.maxLongJumpHeight),
+                k.subtract(this.maxLongJumpWidth),
+                i.add(this.maxLongJumpWidth),
+                j.add(this.maxLongJumpHeight),
+                k.add(this.maxLongJumpWidth)
             )
             .filter(p_217317_ -> !p_217317_.equals(blockpos))
             .map(p_217314_ -> new LongJumpToRandomPos.PossibleJump(p_217314_.immutable(), Mth.ceil(blockpos.distSqr(p_217314_))))
@@ -175,9 +_,9 @@
 
     private boolean isAcceptableLandingPosition(ServerLevel p_217300_, E p_217301_, BlockPos p_217302_) {
         BlockPos blockpos = p_217301_.blockPosition();
-        int i = blockpos.getX();
-        int j = blockpos.getZ();
-        return i == p_217302_.getX() && j == p_217302_.getZ() ? false : this.acceptableLandingSpot.test(p_217301_, p_217302_);
+        me.alphamode.mcbig.math.BigInteger i = blockpos.getBigX();
+        me.alphamode.mcbig.math.BigInteger j = blockpos.getBigZ();
+        return i.equals(p_217302_.getBigX()) && j.equals(p_217302_.getBigZ()) ? false : this.acceptableLandingSpot.test(p_217301_, p_217302_);
     }
 
     @Nullable
