--- a/net/minecraft/world/entity/ai/goal/BreathAirGoal.java
+++ b/net/minecraft/world/entity/ai/goal/BreathAirGoal.java
@@ -41,12 +_,12 @@
 
     private void findAirPosition() {
         Iterable<BlockPos> iterable = BlockPos.betweenClosed(
-            Mth.floor(this.mob.getX() - 1.0),
-            this.mob.getBlockY(),
-            Mth.floor(this.mob.getZ() - 1.0),
-            Mth.floor(this.mob.getX() + 1.0),
-            Mth.floor(this.mob.getY() + 8.0),
-            Mth.floor(this.mob.getZ() + 1.0)
+            Mth.bigFloor(this.mob.getX() - 1.0),
+            me.alphamode.mcbig.math.BigInteger.val(this.mob.getBlockY()),
+            Mth.bigFloor(this.mob.getZ() - 1.0),
+            Mth.bigFloor(this.mob.getX() + 1.0),
+            Mth.bigFloor(this.mob.getY() + 8.0),
+            Mth.bigFloor(this.mob.getZ() + 1.0)
         );
         BlockPos blockpos = null;
 
