--- a/net/minecraft/world/entity/ai/goal/BreathAirGoal.java
+++ b/net/minecraft/world/entity/ai/goal/BreathAirGoal.java
@@ -41,12 +_,12 @@
 
     private void findAirPosition() {
         Iterable<BlockPos> iterable = BlockPos.betweenClosed(
-            Mth.floor(this.mob.getX() - 1.0),
-            this.mob.getBlockY(),
-            Mth.floor(this.mob.getZ() - 1.0),
-            Mth.floor(this.mob.getX() + 1.0),
-            Mth.floor(this.mob.getY() + 8.0),
-            Mth.floor(this.mob.getZ() + 1.0)
+            Mth.bigFloor(this.mob.getX() - 1.0),
+            me.alphamode.mcbig.math.BigInteger.val(this.mob.getBlockY()),
+            Mth.bigFloor(this.mob.getZ() - 1.0),
+            Mth.bigFloor(this.mob.getX() + 1.0),
+            Mth.bigFloor(this.mob.getY() + 8.0),
+            Mth.bigFloor(this.mob.getZ() + 1.0)
         );
         BlockPos blockpos = null;
 
@@ -61,7 +_,7 @@
             blockpos = BlockPos.containing(this.mob.getX(), this.mob.getY() + 8.0, this.mob.getZ());
         }
 
-        this.mob.getNavigation().moveTo((double)blockpos.getX(), (double)(blockpos.getY() + 1), (double)blockpos.getZ(), 1.0);
+        this.mob.getNavigation().moveTo((double)blockpos.getBigX().doubleValue(), (double)(blockpos.getY() + 1), (double)blockpos.getBigZ().doubleValue(), 1.0);
     }
 
     @Override
