--- a/net/minecraft/world/entity/ai/goal/FollowBoatGoal.java
+++ b/net/minecraft/world/entity/ai/goal/FollowBoatGoal.java
@@ -80,7 +_,7 @@
             if (this.currentGoal == BoatGoals.GO_TO_BOAT) {
                 BlockPos blockpos = this.following.blockPosition().relative(this.following.getDirection().getOpposite());
                 blockpos = blockpos.offset(0, -1, 0);
-                this.mob.getNavigation().moveTo((double)blockpos.getX(), (double)blockpos.getY(), (double)blockpos.getZ(), 1.0);
+                this.mob.getNavigation().moveTo((double)blockpos.getBigX().doubleValue(), (double)blockpos.getBigY().doubleValue(), (double)blockpos.getBigZ().doubleValue(), 1.0);
                 if (this.mob.distanceTo(this.following) < 4.0F) {
                     this.timeToRecalcPath = 0;
                     this.currentGoal = BoatGoals.GO_IN_BOAT_DIRECTION;
@@ -88,7 +_,7 @@
             } else if (this.currentGoal == BoatGoals.GO_IN_BOAT_DIRECTION) {
                 Direction direction = this.following.getMotionDirection();
                 BlockPos blockpos1 = this.following.blockPosition().relative(direction, 10);
-                this.mob.getNavigation().moveTo((double)blockpos1.getX(), (double)(blockpos1.getY() - 1), (double)blockpos1.getZ(), 1.0);
+                this.mob.getNavigation().moveTo((double)blockpos1.getBigX().doubleValue(), (double)(blockpos1.getY() - 1), (double)blockpos1.getBigZ().doubleValue(), 1.0);
                 if (this.mob.distanceTo(this.following) > 12.0F) {
                     this.timeToRecalcPath = 0;
                     this.currentGoal = BoatGoals.GO_TO_BOAT;
