--- a/net/minecraft/world/entity/ai/goal/PanicGoal.java
+++ b/net/minecraft/world/entity/ai/goal/PanicGoal.java
@@ -33,9 +_,9 @@
             if (this.mob.isOnFire()) {
                 BlockPos blockpos = this.lookForWater(this.mob.level(), this.mob, 5);
                 if (blockpos != null) {
-                    this.posX = (double)blockpos.getX();
-                    this.posY = (double)blockpos.getY();
-                    this.posZ = (double)blockpos.getZ();
+                    this.posX = (double)blockpos.getBigX().doubleValue();
+                    this.posY = (double)blockpos.getBigY().doubleValue();
+                    this.posZ = (double)blockpos.getBigZ().doubleValue();
                     return true;
                 }
             }
