--- a/net/minecraft/world/level/BlockGetter.java
+++ b/net/minecraft/world/level/BlockGetter.java
@@ -63,7 +_,8 @@
     }
 
     default BlockHitResult clip(ClipContext p_45548_) {
-        return traverseBlocks(p_45548_.getFrom(), p_45548_.getTo(), p_45548_, (p_151359_, p_151360_) -> {
+        Vec3 vec3 = p_45548_.getFrom().subtract(p_45548_.getTo());
+        return BlockHitResult.miss(p_45548_.getTo(), Direction.getNearest(vec3.x, vec3.y, vec3.z), BlockPos.containing(p_45548_.getTo()));/*traverseBlocks(p_45548_.getFrom(), p_45548_.getTo(), p_45548_, (p_151359_, p_151360_) -> {
             BlockState blockstate = this.getBlockState(p_151360_);
             FluidState fluidstate = this.getFluidState(p_151360_);
             Vec3 vec3 = p_151359_.getFrom();
@@ -78,7 +_,7 @@
         }, p_275153_ -> {
             Vec3 vec3 = p_275153_.getFrom().subtract(p_275153_.getTo());
             return BlockHitResult.miss(p_275153_.getTo(), Direction.getNearest(vec3.x, vec3.y, vec3.z), BlockPos.containing(p_275153_.getTo()));
-        });
+        });*/
     }
 
     @Nullable
@@ -121,9 +_,9 @@
             double d3 = Mth.lerp(-1.0E-7, p_151362_.x, p_151363_.x);
             double d4 = Mth.lerp(-1.0E-7, p_151362_.y, p_151363_.y);
             double d5 = Mth.lerp(-1.0E-7, p_151362_.z, p_151363_.z);
-            int i = Mth.floor(d3);
-            int j = Mth.floor(d4);
-            int k = Mth.floor(d5);
+            me.alphamode.mcbig.math.BigInteger i = Mth.bigFloor(d3);
+            me.alphamode.mcbig.math.BigInteger j = Mth.bigFloor(d4);
+            me.alphamode.mcbig.math.BigInteger k = Mth.bigFloor(d5);
             BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos(i, j, k);
             T t = p_151365_.apply(p_151364_, blockpos$mutableblockpos);
             if (t != null) {
@@ -145,17 +_,17 @@
                 while(d12 <= 1.0 || d13 <= 1.0 || d14 <= 1.0) {
                     if (d12 < d13) {
                         if (d12 < d14) {
-                            i += l;
+                            i = i.add(l);
                             d12 += d9;
                         } else {
-                            k += j1;
+                            k = k.add(j1);
                             d14 += d11;
                         }
                     } else if (d13 < d14) {
-                        j += i1;
+                        j = j.add(i1);
                         d13 += d10;
                     } else {
-                        k += j1;
+                        k = k.add(j1);
                         d14 += d11;
                     }
 
