--- a/net/minecraft/world/level/ClipBlockStateContext.java
+++ b/net/minecraft/world/level/ClipBlockStateContext.java
@@ -5,21 +_,21 @@
 import net.minecraft.world.phys.Vec3;
 
 public class ClipBlockStateContext {
-    private final Vec3 from;
-    private final Vec3 to;
+    private final me.alphamode.mcbig.core.BigVec3 from;
+    private final me.alphamode.mcbig.core.BigVec3 to;
     private final Predicate<BlockState> block;
 
-    public ClipBlockStateContext(Vec3 p_151401_, Vec3 p_151402_, Predicate<BlockState> p_151403_) {
+    public ClipBlockStateContext(me.alphamode.mcbig.core.BigVec3 p_151401_, me.alphamode.mcbig.core.BigVec3 p_151402_, Predicate<BlockState> p_151403_) {
         this.from = p_151401_;
         this.to = p_151402_;
         this.block = p_151403_;
     }
 
-    public Vec3 getTo() {
+    public me.alphamode.mcbig.core.BigVec3 getTo() {
         return this.to;
     }
 
-    public Vec3 getFrom() {
+    public me.alphamode.mcbig.core.BigVec3 getFrom() {
         return this.from;
     }
 
