--- a/net/minecraft/world/level/CollisionGetter.java
+++ b/net/minecraft/world/level/CollisionGetter.java
@@ -22,7 +_,7 @@
     WorldBorder getWorldBorder();
 
     @Nullable
-    BlockGetter getChunkForCollisions(int p_45774_, int p_45775_);
+    BlockGetter getChunkForCollisions(me.alphamode.mcbig.math.BigInteger p_45774_, me.alphamode.mcbig.math.BigInteger p_45775_);
 
     default boolean isUnobstructed(@Nullable Entity p_45750_, VoxelShape p_45751_) {
         return true;
@@ -30,7 +_,7 @@
 
     default boolean isUnobstructed(BlockState p_45753_, BlockPos p_45754_, CollisionContext p_45755_) {
         VoxelShape voxelshape = p_45753_.getCollisionShape(this, p_45754_, p_45755_);
-        return voxelshape.isEmpty() || this.isUnobstructed(null, voxelshape.move((double)p_45754_.getX(), (double)p_45754_.getY(), (double)p_45754_.getZ()));
+        return voxelshape.isEmpty() || this.isUnobstructed(null, voxelshape.move((double)p_45754_.getBigX().doubleValue(), (double)p_45754_.getBigY().doubleValue(), (double)p_45754_.getBigZ().doubleValue()));
     }
 
     default boolean isUnobstructed(Entity p_45785_) {
@@ -102,9 +_,9 @@
         BlockHitResult blockhitresult = this.clip(p_365129_);
         WorldBorder worldborder = this.getWorldBorder();
         if (worldborder.isWithinBounds(p_365129_.getFrom()) && !worldborder.isWithinBounds(blockhitresult.getLocation())) {
-            Vec3 vec3 = blockhitresult.getLocation().subtract(p_365129_.getFrom());
-            Direction direction = Direction.getApproximateNearest(vec3.x, vec3.y, vec3.z);
-            Vec3 vec31 = worldborder.clampVec3ToBound(blockhitresult.getLocation());
+            me.alphamode.mcbig.core.BigVec3 vec3 = blockhitresult.getLocation().subtract(p_365129_.getFrom());
+            Direction direction = Direction.getApproximateNearest(vec3.x.doubleValue(), vec3.y, vec3.z.doubleValue());
+            me.alphamode.mcbig.core.BigVec3 vec31 = worldborder.clampVec3ToBound(blockhitresult.getLocation());
             return new BlockHitResult(vec31, direction, BlockPos.containing(vec31), false, true);
         } else {
             return blockhitresult;
