--- a/net/minecraft/world/level/levelgen/feature/trunkplacers/UpwardsBranchingTrunkPlacer.java
+++ b/net/minecraft/world/level/levelgen/feature/trunkplacers/UpwardsBranchingTrunkPlacer.java
@@ -67,7 +_,7 @@
 
         for (int i = 0; i < p_226228_; i++) {
             int j = p_226229_.getY() + i;
-            if (this.placeLog(p_226225_, p_226226_, p_226227_, blockpos$mutableblockpos.set(p_226229_.getX(), j, p_226229_.getZ()), p_226230_)
+            if (this.placeLog(p_226225_, p_226226_, p_226227_, blockpos$mutableblockpos.set(p_226229_.getBigX(), j, p_226229_.getBigZ()), p_226230_)
                 && i < p_226228_ - 1
                 && p_226227_.nextFloat() < this.placeBranchPerLogProbability) {
                 Direction direction = Direction.Plane.HORIZONTAL.getRandomDirection(p_226227_);
@@ -78,7 +_,7 @@
             }
 
             if (i == p_226228_ - 1) {
-                list.add(new FoliagePlacer.FoliageAttachment(blockpos$mutableblockpos.set(p_226229_.getX(), j + 1, p_226229_.getZ()), 0, false));
+                list.add(new FoliagePlacer.FoliageAttachment(blockpos$mutableblockpos.set(p_226229_.getBigX(), j + 1, p_226229_.getBigZ()), 0, false));
             }
         }
 
