--- a/net/minecraft/client/Camera.java
+++ b/net/minecraft/client/Camera.java
@@ -26,7 +_,7 @@
     private boolean initialized;
     private BlockGetter level;
     private Entity entity;
-    private Vec3 position = Vec3.ZERO;
+    private me.alphamode.mcbig.core.BigVec3 position = me.alphamode.mcbig.core.BigVec3.ZERO;
     private final BlockPos.MutableBlockPos blockPosition = new BlockPos.MutableBlockPos();
     private final Vector3f forwards = new Vector3f(0.0F, 0.0F, 1.0F);
     private final Vector3f up = new Vector3f(0.0F, 1.0F, 0.0F);
@@ -48,9 +_,9 @@
         this.partialTickTime = p_90580_;
         this.setRotation(p_90577_.getViewYRot(p_90580_), p_90577_.getViewXRot(p_90580_));
         this.setPosition(
-            Mth.lerp((double)p_90580_, p_90577_.xo, p_90577_.getX()),
+            Mth.lerp((double)p_90580_, p_90577_.bigXo, p_90577_.getBigX()),
             Mth.lerp((double)p_90580_, p_90577_.yo, p_90577_.getY()) + (double)Mth.lerp(p_90580_, this.eyeHeightOld, this.eyeHeight),
-            Mth.lerp((double)p_90580_, p_90577_.zo, p_90577_.getZ())
+            Mth.lerp((double)p_90580_, p_90577_.bigZo, p_90577_.getBigZ())
         );
         if (p_90578_) {
             if (p_90579_) {
@@ -81,11 +_,11 @@
             f *= 0.1F;
             f1 *= 0.1F;
             f2 *= 0.1F;
-            Vec3 vec3 = this.position.add((double)f, (double)f1, (double)f2);
-            Vec3 vec31 = new Vec3(
-                this.position.x - (double)this.forwards.x() * p_90567_ + (double)f,
+            me.alphamode.mcbig.core.BigVec3 vec3 = this.position.add((double)f, (double)f1, (double)f2);
+            me.alphamode.mcbig.core.BigVec3 vec31 = new me.alphamode.mcbig.core.BigVec3(
+                this.position.x.subtract((double)this.forwards.x() * p_90567_ + (double)f),
                 this.position.y - (double)this.forwards.y() * p_90567_ + (double)f1,
-                this.position.z - (double)this.forwards.z() * p_90567_ + (double)f2
+                this.position.z.subtract((double)this.forwards.z() * p_90567_ + (double)f2)
             );
             HitResult hitresult = this.level.clip(new ClipContext(vec3, vec31, ClipContext.Block.VISUAL, ClipContext.Fluid.NONE, this.entity));
             if (hitresult.getType() != HitResult.Type.MISS) {
@@ -103,7 +_,7 @@
         double d0 = (double)this.forwards.x() * p_90569_ + (double)this.up.x() * p_90570_ + (double)this.left.x() * p_90571_;
         double d1 = (double)this.forwards.y() * p_90569_ + (double)this.up.y() * p_90570_ + (double)this.left.y() * p_90571_;
         double d2 = (double)this.forwards.z() * p_90569_ + (double)this.up.z() * p_90570_ + (double)this.left.z() * p_90571_;
-        this.setPosition(new Vec3(this.position.x + d0, this.position.y + d1, this.position.z + d2));
+        this.setPosition(new me.alphamode.mcbig.core.BigVec3(this.position.x.add(d0), this.position.y + d1, this.position.z.add(d2)));
     }
 
     protected void setRotation(float p_90573_, float p_90574_) {
@@ -115,16 +_,16 @@
         this.left.set(1.0F, 0.0F, 0.0F).rotate(this.rotation);
     }
 
-    protected void setPosition(double p_90585_, double p_90586_, double p_90587_) {
-        this.setPosition(new Vec3(p_90585_, p_90586_, p_90587_));
+    protected void setPosition(me.alphamode.mcbig.math.BigDecimal p_90585_, double p_90586_, me.alphamode.mcbig.math.BigDecimal p_90587_) {
+        this.setPosition(new me.alphamode.mcbig.core.BigVec3(p_90585_, p_90586_, p_90587_));
     }
 
-    protected void setPosition(Vec3 p_90582_) {
+    protected void setPosition(me.alphamode.mcbig.core.BigVec3 p_90582_) {
         this.position = p_90582_;
         this.blockPosition.set(p_90582_.x, p_90582_.y, p_90582_.z);
     }
 
-    public Vec3 getPosition() {
+    public me.alphamode.mcbig.core.BigVec3 getPosition() {
         return this.position;
     }
 
@@ -185,7 +_,7 @@
                     camera$nearplane.getBottomLeft(),
                     camera$nearplane.getBottomRight()
                 )) {
-                    Vec3 vec31 = this.position.add(vec3);
+                    me.alphamode.mcbig.core.BigVec3 vec31 = this.position.add(vec3);
                     BlockPos blockpos = BlockPos.containing(vec31);
                     FluidState fluidstate1 = this.level.getFluidState(blockpos);
                     if (fluidstate1.is(FluidTags.LAVA)) {
