--- a/net/minecraft/client/Camera.java
+++ b/net/minecraft/client/Camera.java
@@ -31,7 +_,7 @@
     private boolean initialized;
     private BlockGetter level;
     private Entity entity;
-    private Vec3 position = Vec3.ZERO;
+    private me.alphamode.mcbig.core.BigVec3 position = me.alphamode.mcbig.core.BigVec3.ZERO;
     private final BlockPos.MutableBlockPos blockPosition = new BlockPos.MutableBlockPos();
     private final Vector3f forwards = new Vector3f(FORWARDS);
     private final Vector3f up = new Vector3f(UP);
@@ -60,13 +_,13 @@
                 .subtract(p_90577_.getVehicleAttachmentPoint(minecart))
                 .add(new Vec3(0.0, (double)Mth.lerp(p_90580_, this.eyeHeightOld, this.eyeHeight), 0.0));
             this.setRotation(p_90577_.getViewYRot(p_90580_), p_90577_.getViewXRot(p_90580_));
-            this.setPosition(newminecartbehavior.getCartLerpPosition(p_90580_).add(vec3));
+            this.setPosition(newminecartbehavior.getCartLerpPosition(p_90580_).add(vec3).toBig());
         } else {
             this.setRotation(p_90577_.getViewYRot(p_90580_), p_90577_.getViewXRot(p_90580_));
             this.setPosition(
-                Mth.lerp((double)p_90580_, p_90577_.xo, p_90577_.getX()),
+                Mth.lerp(p_90580_, p_90577_.bigXo, p_90577_.getBigX()),
                 Mth.lerp((double)p_90580_, p_90577_.yo, p_90577_.getY()) + (double)Mth.lerp(p_90580_, this.eyeHeightOld, this.eyeHeight),
-                Mth.lerp((double)p_90580_, p_90577_.zo, p_90577_.getZ())
+                Mth.lerp(p_90580_, p_90577_.bigZo, p_90577_.getBigZ())
             );
         }
 
@@ -98,8 +_,8 @@
             float f1 = (float)((i & 1) * 2 - 1);
             float f2 = (float)((i >> 1 & 1) * 2 - 1);
             float f3 = (float)((i >> 2 & 1) * 2 - 1);
-            Vec3 vec3 = this.position.add((double)(f1 * 0.1F), (double)(f2 * 0.1F), (double)(f3 * 0.1F));
-            Vec3 vec31 = vec3.add(new Vec3(this.forwards).scale((double)(-p_350335_)));
+            me.alphamode.mcbig.core.BigVec3 vec3 = this.position.add((double)(f1 * 0.1F), (double)(f2 * 0.1F), (double)(f3 * 0.1F));
+            me.alphamode.mcbig.core.BigVec3 vec31 = vec3.add(new Vec3(this.forwards).scale((double)(-p_350335_)));
             HitResult hitresult = this.level.clip(new ClipContext(vec3, vec31, ClipContext.Block.VISUAL, ClipContext.Fluid.NONE, this.entity));
             if (hitresult.getType() != HitResult.Type.MISS) {
                 float f4 = (float)hitresult.getLocation().distanceToSqr(this.position);
@@ -114,7 +_,7 @@
 
     protected void move(float p_351037_, float p_351008_, float p_350869_) {
         Vector3f vector3f = new Vector3f(p_350869_, p_351008_, -p_351037_).rotate(this.rotation);
-        this.setPosition(new Vec3(this.position.x + (double)vector3f.x, this.position.y + (double)vector3f.y, this.position.z + (double)vector3f.z));
+        this.setPosition(new me.alphamode.mcbig.core.BigVec3(this.position.x.add((double)vector3f.x), this.position.y + (double)vector3f.y, this.position.z.add((double)vector3f.z)));
     }
 
     protected void setRotation(float p_90573_, float p_90574_) {
@@ -126,16 +_,16 @@
         LEFT.rotate(this.rotation, this.left);
     }
 
-    protected void setPosition(double p_90585_, double p_90586_, double p_90587_) {
-        this.setPosition(new Vec3(p_90585_, p_90586_, p_90587_));
+    protected void setPosition(me.alphamode.mcbig.math.BigDecimal p_90585_, double p_90586_, me.alphamode.mcbig.math.BigDecimal p_90587_) {
+        this.setPosition(new me.alphamode.mcbig.core.BigVec3(p_90585_, p_90586_, p_90587_));
     }
 
-    protected void setPosition(Vec3 p_90582_) {
+    protected void setPosition(me.alphamode.mcbig.core.BigVec3 p_90582_) {
         this.position = p_90582_;
         this.blockPosition.set(p_90582_.x, p_90582_.y, p_90582_.z);
     }
 
-    public Vec3 getPosition() {
+    public me.alphamode.mcbig.core.BigVec3 getPosition() {
         return this.position;
     }
 
@@ -196,7 +_,7 @@
                     camera$nearplane.getBottomLeft(),
                     camera$nearplane.getBottomRight()
                 )) {
-                    Vec3 vec31 = this.position.add(vec3);
+                    me.alphamode.mcbig.core.BigVec3 vec31 = this.position.add(vec3);
                     BlockPos blockpos = BlockPos.containing(vec31);
                     FluidState fluidstate1 = this.level.getFluidState(blockpos);
                     if (fluidstate1.is(FluidTags.LAVA)) {
