--- a/net/minecraft/world/entity/Entity.java
+++ b/net/minecraft/world/entity/Entity.java
@@ -52,7 +_,6 @@
 import net.minecraft.network.chat.Component;
 import net.minecraft.network.chat.HoverEvent;
 import net.minecraft.network.chat.MutableComponent;
-import net.minecraft.network.chat.Style;
 import net.minecraft.network.protocol.Packet;
 import net.minecraft.network.protocol.game.ClientGamePacketListener;
 import net.minecraft.network.protocol.game.ClientboundAddEntityPacket;
@@ -168,7 +_,10 @@
     public double xo;
     public double yo;
     public double zo;
-    private Vec3 position;
+    public me.alphamode.mcbig.math.BigDecimal bigXo = me.alphamode.mcbig.math.BigDecimal.ZERO;
+    public me.alphamode.mcbig.math.BigDecimal bigZo = me.alphamode.mcbig.math.BigDecimal.ZERO;
+    protected Vec3 position;
+    protected me.alphamode.mcbig.core.BigVec3 bigPosition;
     private BlockPos blockPosition;
     private ChunkPos chunkPosition;
     private Vec3 deltaMovement = Vec3.ZERO;
@@ -257,6 +_,7 @@
         this.level = p_19871_;
         this.dimensions = p_19870_.getDimensions();
         this.position = Vec3.ZERO;
+        this.bigPosition = me.alphamode.mcbig.core.BigVec3.ZERO;
         this.blockPosition = BlockPos.ZERO;
         this.chunkPosition = ChunkPos.ZERO;
         SynchedEntityData.Builder synchedentitydata$builder = new SynchedEntityData.Builder(this);
@@ -276,7 +_,7 @@
 
     public boolean isColliding(BlockPos p_20040_, BlockState p_20041_) {
         VoxelShape voxelshape = p_20041_.getCollisionShape(this.level(), p_20040_, CollisionContext.of(this));
-        VoxelShape voxelshape1 = voxelshape.move((double)p_20040_.getX(), (double)p_20040_.getY(), (double)p_20040_.getZ());
+        VoxelShape voxelshape1 = voxelshape.move((double)p_20040_.getBigX().doubleValue(), (double)p_20040_.getBigY().doubleValue(), (double)p_20040_.getBigZ().doubleValue());
         return Shapes.joinIsNotEmpty(voxelshape1, Shapes.create(this.getBoundingBox()), BooleanOp.AND);
     }
 
@@ -299,8 +_,8 @@
         }
     }
 
-    public void syncPacketPositionCodec(double p_217007_, double p_217008_, double p_217009_) {
-        this.packetPositionCodec.setBase(new Vec3(p_217007_, p_217008_, p_217009_));
+    public void syncPacketPositionCodec(me.alphamode.mcbig.math.BigDecimal p_217007_, double p_217008_, me.alphamode.mcbig.math.BigDecimal p_217009_) {
+        this.packetPositionCodec.setBase(new me.alphamode.mcbig.core.BigVec3(p_217007_, p_217008_, p_217009_));
     }
 
     public VecDeltaCodec getPositionCodec() {
@@ -399,11 +_,20 @@
         this.setPos(p_146885_.x(), p_146885_.y(), p_146885_.z());
     }
 
+    public final void setPos(me.alphamode.mcbig.core.BigVec3 p_146885_) {
+        this.setPos(p_146885_.x(), p_146885_.y(), p_146885_.z());
+    }
+
     public void setPos(double p_20210_, double p_20211_, double p_20212_) {
         this.setPosRaw(p_20210_, p_20211_, p_20212_);
         this.setBoundingBox(this.makeBoundingBox());
     }
 
+    public void setPos(me.alphamode.mcbig.math.BigDecimal p_20210_, double p_20211_, me.alphamode.mcbig.math.BigDecimal p_20212_) {
+        this.setPosRaw(p_20210_, p_20211_, p_20212_);
+        this.setBoundingBox(this.makeBoundingBox());
+    }
+
     protected final AABB makeBoundingBox() {
         return this.makeBoundingBox(this.position);
     }
@@ -414,6 +_,7 @@
 
     protected void reapplyPosition() {
         this.setPos(this.position.x, this.position.y, this.position.z);
+        this.setPos(this.bigPosition.x, this.bigPosition.y, this.bigPosition.z);
     }
 
     public void turn(double p_19885_, double p_19886_) {
@@ -629,7 +_,7 @@
 
     public void move(MoverType p_19973_, Vec3 p_19974_) {
         if (this.noPhysics) {
-            this.setPos(this.getX() + p_19974_.x, this.getY() + p_19974_.y, this.getZ() + p_19974_.z);
+            this.setPos(this.getBigX().add(p_19974_.x), this.getY() + p_19974_.y, this.getBigZ().add(p_19974_.z));
         } else {
             this.wasOnFire = this.isOnFire();
             if (p_19973_ == MoverType.PISTON) {
@@ -654,14 +_,14 @@
                 if (this.fallDistance != 0.0F && d0 >= 1.0) {
                     BlockHitResult blockhitresult = this.level()
                         .clip(
-                            new ClipContext(this.position(), this.position().add(vec3), ClipContext.Block.FALLDAMAGE_RESETTING, ClipContext.Fluid.WATER, this)
+                            new ClipContext(this.bigPosition(), this.bigPosition().add(vec3), ClipContext.Block.FALLDAMAGE_RESETTING, ClipContext.Fluid.WATER, this)
                         );
                     if (blockhitresult.getType() != HitResult.Type.MISS) {
                         this.resetFallDistance();
                     }
                 }
 
-                this.setPos(this.getX() + vec3.x, this.getY() + vec3.y, this.getZ() + vec3.z);
+                this.setPos(this.getBigX().add(vec3.x), this.getY() + vec3.y, this.getBigZ().add(vec3.z));
             }
 
             profilerfiller.pop();
@@ -802,7 +_,7 @@
                 }
 
                 if (p_286543_) {
-                    this.level().gameEvent(GameEvent.STEP, this.position(), GameEvent.Context.of(this, p_286549_));
+                    this.level().gameEvent(GameEvent.STEP, this.bigPosition(), GameEvent.Context.of(this, p_286549_));
                 }
 
                 return true;
@@ -864,9 +_,9 @@
                     : blockpos;
             }
         } else {
-            int i = Mth.floor(this.position.x);
-            int j = Mth.floor(this.position.y - (double)p_216987_);
-            int k = Mth.floor(this.position.z);
+            me.alphamode.mcbig.math.BigInteger i = Mth.bigFloor(this.position.x);
+            me.alphamode.mcbig.math.BigInteger j = Mth.bigFloor(this.position.y - (double)p_216987_);
+            me.alphamode.mcbig.math.BigInteger k = Mth.bigFloor(this.position.z);
             return new BlockPos(i, j, k);
         }
     }
@@ -1108,12 +_,12 @@
     public BlockPos adjustSpawnLocation(ServerLevel p_352356_, BlockPos p_352343_) {
         BlockPos blockpos = p_352356_.getSharedSpawnPos();
         Vec3 vec3 = blockpos.getCenter();
-        int i = p_352356_.getChunkAt(blockpos).getHeight(Heightmap.Types.MOTION_BLOCKING_NO_LEAVES, blockpos.getX(), blockpos.getZ()) + 1;
+        int i = p_352356_.getChunkAt(blockpos).getHeight(Heightmap.Types.MOTION_BLOCKING_NO_LEAVES, blockpos.getBigX(), blockpos.getBigZ()) + 1;
         return BlockPos.containing(vec3.x, (double)i, vec3.z);
     }
 
     public void gameEvent(Holder<GameEvent> p_316133_, @Nullable Entity p_146854_) {
-        this.level().gameEvent(p_146854_, p_316133_, this.position);
+        this.level().gameEvent(p_146854_, p_316133_, this.bigPosition);
     }
 
     public void gameEvent(Holder<GameEvent> p_316256_) {
@@ -1248,7 +_,7 @@
                 this.level()
                     .gameEvent(
                         GameEvent.HIT_GROUND,
-                        this.position,
+                        this.bigPosition,
                         GameEvent.Context.of(
                             this, this.mainSupportingBlockPos.<BlockState>map(p_286200_ -> this.level().getBlockState(p_286200_)).orElse(p_19913_)
                         )
@@ -1286,10 +_,12 @@
     private boolean isInRain() {
         BlockPos blockpos = this.blockPosition();
         return this.level().isRainingAt(blockpos)
-            || this.level().isRainingAt(BlockPos.containing((double)blockpos.getX(), this.getBoundingBox().maxY, (double)blockpos.getZ()));
+            || this.level().isRainingAt(BlockPos.containing((double)blockpos.getBigX().doubleValue(), this.getBoundingBox().maxY, (double)blockpos.getBigZ().doubleValue()));
     }
 
     private boolean isInBubbleColumn() {
+        if (me.alphamode.mcbig.core.BigConstants.BIG_MODE)
+            return false;
         return this.getInBlockState().is(Blocks.BUBBLE_COLUMN);
     }
 
@@ -1361,7 +_,7 @@
             return;
         }
 
-        BlockPos blockpos = BlockPos.containing(this.getX(), d0, this.getZ());
+        BlockPos blockpos = BlockPos.containing(this.getBigX(), d0, this.getBigZ());
         FluidState fluidstate = this.level().getFluidState(blockpos);
         double d1 = (double)((float)blockpos.getY() + fluidstate.getHeight(this.level(), blockpos));
         if (d1 > d0) {
@@ -1421,12 +_,12 @@
             BlockPos blockpos1 = this.blockPosition();
             double d0 = this.getX() + (this.random.nextDouble() - 0.5) * (double)this.dimensions.width();
             double d1 = this.getZ() + (this.random.nextDouble() - 0.5) * (double)this.dimensions.width();
-            if (blockpos1.getX() != blockpos.getX()) {
-                d0 = Mth.clamp(d0, (double)blockpos.getX(), (double)blockpos.getX() + 1.0);
+            if (!java.util.Objects.equals(blockpos1.getBigX(), blockpos.getBigX())) {
+                d0 = Mth.clamp(d0, (double)blockpos.getBigX().doubleValue(), (double)blockpos.getBigX().doubleValue() + 1.0);
             }
 
-            if (blockpos1.getZ() != blockpos.getZ()) {
-                d1 = Mth.clamp(d1, (double)blockpos.getZ(), (double)blockpos.getZ() + 1.0);
+            if (!java.util.Objects.equals(blockpos1.getBigZ(), blockpos.getBigZ())) {
+                d1 = Mth.clamp(d1, (double)blockpos.getBigZ().doubleValue(), (double)blockpos.getBigZ().doubleValue() + 1.0);
             }
 
             this.level().addParticle(new BlockParticleOption(ParticleTypes.BLOCK, blockstate), d0, this.getY() + 0.1, d1, vec3.x * -4.0, 1.5, vec3.z * -4.0);
@@ -1461,10 +_,15 @@
     @Deprecated
     public float getLightLevelDependentMagicValue() {
         return this.level().hasChunkAt(this.getBlockX(), this.getBlockZ())
-            ? this.level().getLightLevelDependentMagicValue(BlockPos.containing(this.getX(), this.getEyeY(), this.getZ()))
+            ? this.level().getLightLevelDependentMagicValue(BlockPos.containing(this.getBigX(), this.getEyeY(), this.getBigZ()))
             : 0.0F;
     }
 
+    public void absMoveTo(me.alphamode.mcbig.math.BigDecimal p_19891_, double p_19892_, me.alphamode.mcbig.math.BigDecimal p_19893_, float p_19894_, float p_19895_) {
+        this.absMoveTo(p_19891_, p_19892_, p_19893_);
+        this.absRotateTo(p_19894_, p_19895_);
+    }
+
     public void absMoveTo(double p_19891_, double p_19892_, double p_19893_, float p_19894_, float p_19895_) {
         this.absMoveTo(p_19891_, p_19892_, p_19893_);
         this.absRotateTo(p_19894_, p_19895_);
@@ -1477,9 +_,18 @@
         this.xRotO = this.getXRot();
     }
 
+    public void absMoveTo(me.alphamode.mcbig.math.BigDecimal p_20249_, double p_20250_, me.alphamode.mcbig.math.BigDecimal p_20251_) {
+        me.alphamode.mcbig.math.BigDecimal d0 = p_20249_;//Mth.clamp(p_20249_, -3.0E7, 3.0E7);
+        me.alphamode.mcbig.math.BigDecimal d1 = p_20251_;//Mth.clamp(p_20251_, -3.0E7, 3.0E7);
+        this.xo = d0.doubleValue();
+        this.yo = p_20250_;
+        this.zo = d1.doubleValue();
+        this.setPos(d0, p_20250_, d1);
+    }
+
     public void absMoveTo(double p_20249_, double p_20250_, double p_20251_) {
-        double d0 = Mth.clamp(p_20249_, -3.0E7, 3.0E7);
-        double d1 = Mth.clamp(p_20251_, -3.0E7, 3.0E7);
+        double d0 = p_20249_;//Mth.clamp(p_20249_, -3.0E7, 3.0E7);
+        double d1 = p_20251_;//Mth.clamp(p_20251_, -3.0E7, 3.0E7);
         this.xo = d0;
         this.yo = p_20250_;
         this.zo = d1;
@@ -1510,28 +_,40 @@
         this.reapplyPosition();
     }
 
+    public void moveTo(me.alphamode.mcbig.math.BigDecimal p_20108_, double p_20109_, me.alphamode.mcbig.math.BigDecimal p_20110_, float p_20111_, float p_20112_) {
+        this.setPosRaw(p_20108_, p_20109_, p_20110_);
+        this.setYRot(p_20111_);
+        this.setXRot(p_20112_);
+        this.setOldPosAndRot();
+        this.reapplyPosition();
+    }
+
     public final void setOldPosAndRot() {
         this.setOldPos();
         this.setOldRot();
     }
 
-    public final void setOldPosAndRot(Vec3 p_371927_, float p_371686_, float p_371638_) {
+    public final void setOldPosAndRot(me.alphamode.mcbig.core.BigVec3 p_371927_, float p_371686_, float p_371638_) {
         this.setOldPos(p_371927_);
         this.setOldRot(p_371686_, p_371638_);
     }
 
     protected void setOldPos() {
-        this.setOldPos(this.position);
+        this.setOldPos(this.bigPosition);
     }
 
     public void setOldRot() {
         this.setOldRot(this.getYRot(), this.getXRot());
     }
 
-    private void setOldPos(Vec3 p_371773_) {
-        this.xo = this.xOld = p_371773_.x;
+    protected void setOldPos(me.alphamode.mcbig.core.BigVec3 p_371773_) {
+        this.xo = this.xOld = p_371773_.x.doubleValue();
         this.yo = this.yOld = p_371773_.y;
-        this.zo = this.zOld = p_371773_.z;
+        this.zo = this.zOld = p_371773_.z.doubleValue();
+        if (this instanceof Player player) {
+            player.bigXo = p_371773_.x;
+            player.bigZo = p_371773_.z;
+        }
     }
 
     private void setOldRot(float p_371749_, float p_371505_) {
@@ -1568,6 +_,13 @@
         return d0 * d0 + d1 * d1 + d2 * d2;
     }
 
+    public double distanceToSqr(me.alphamode.mcbig.core.BigVec3 p_20239_) {
+        double d0 = this.getBigX().subtract(p_20239_.x).doubleValue();
+        double d1 = this.getY() - p_20239_.y;
+        double d2 = this.getBigZ().subtract(p_20239_.z).doubleValue();
+        return d0 * d0 + d1 * d1 + d2 * d2;
+    }
+
     public void playerTouch(Player p_20081_) {
     }
 
@@ -1701,7 +_,7 @@
         Vec3 vec3 = this.getEyePosition(p_19909_);
         Vec3 vec31 = this.getViewVector(p_19909_);
         Vec3 vec32 = vec3.add(vec31.x * p_19908_, vec31.y * p_19908_, vec31.z * p_19908_);
-        return this.level().clip(new ClipContext(vec3, vec32, ClipContext.Block.OUTLINE, p_19910_ ? ClipContext.Fluid.ANY : ClipContext.Fluid.NONE, this));
+        return this.level().clip(new ClipContext(vec3.toBig(), vec32.toBig(), ClipContext.Block.OUTLINE, p_19910_ ? ClipContext.Fluid.ANY : ClipContext.Fluid.NONE, this));
     }
 
     public boolean canBeHitByProjectile() {
@@ -1854,9 +_,9 @@
             this.hasImpulse = true;
             double d3 = 3.0000512E7;
             this.setPosRaw(
-                Mth.clamp(listtag.getDouble(0), -3.0000512E7, 3.0000512E7),
-                Mth.clamp(listtag.getDouble(1), -2.0E7, 2.0E7),
-                Mth.clamp(listtag.getDouble(2), -3.0000512E7, 3.0000512E7)
+                /*Mth.clamp(*/listtag.getDouble(0),/*, -3.0000512E7, 3.0000512E7)*/
+                /*Mth.clamp(*/listtag.getDouble(1),/*, -2.0E7, 2.0E7),*/
+                /*Mth.clamp(*/listtag.getDouble(2)/*, -3.0000512E7, 3.0000512E7)*/
             );
             this.setYRot(listtag2.getFloat(0));
             this.setXRot(listtag2.getFloat(1));
@@ -2005,7 +_,7 @@
                             && blockstate.isSuffocating(this.level(), p_201942_)
                             && Shapes.joinIsNotEmpty(
                                 blockstate.getCollisionShape(this.level(), p_201942_)
-                                    .move((double)p_201942_.getX(), (double)p_201942_.getY(), (double)p_201942_.getZ()),
+                                    .move((double)p_201942_.getBigX().doubleValue(), (double)p_201942_.getBigY().doubleValue(), (double)p_201942_.getBigZ().doubleValue()),
                                 Shapes.create(aabb),
                                 BooleanOp.AND
                             );
@@ -2201,6 +_,11 @@
     public void cancelLerp() {
     }
 
+    public void lerpTo(me.alphamode.mcbig.math.BigDecimal p_19896_, double p_19897_, me.alphamode.mcbig.math.BigDecimal p_19898_, float p_19899_, float p_19900_, int p_19901_) {
+        this.setPos(p_19896_, p_19897_, p_19898_);
+        this.setRot(p_19899_, p_19900_);
+    }
+
     public void lerpTo(double p_19896_, double p_19897_, double p_19898_, float p_19899_, float p_19900_, int p_19901_) {
         this.setPos(p_19896_, p_19897_, p_19898_);
         this.setRot(p_19899_, p_19900_);
@@ -2539,7 +_,7 @@
 
     protected void moveTowardsClosestSpace(double p_20315_, double p_20316_, double p_20317_) {
         BlockPos blockpos = BlockPos.containing(p_20315_, p_20316_, p_20317_);
-        Vec3 vec3 = new Vec3(p_20315_ - (double)blockpos.getX(), p_20316_ - (double)blockpos.getY(), p_20317_ - (double)blockpos.getZ());
+        Vec3 vec3 = new Vec3(p_20315_ - (double)blockpos.getBigX().doubleValue(), p_20316_ - (double)blockpos.getBigY().doubleValue(), p_20317_ - (double)blockpos.getBigZ().doubleValue());
         BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
         Direction direction = Direction.UP;
         double d0 = Double.MAX_VALUE;
@@ -2746,7 +_,7 @@
         float f = p_380120_.yRot() + (p_380120_.relatives().contains(Relative.Y_ROT) ? 0.0F : p_379847_.getYRot() - this.getYRot());
         float f1 = p_380120_.xRot() + (p_380120_.relatives().contains(Relative.X_ROT) ? 0.0F : p_379847_.getXRot() - this.getXRot());
         Vec3 vec3 = p_379847_.position().subtract(this.position());
-        Vec3 vec31 = p_380120_.position()
+        me.alphamode.mcbig.core.BigVec3 vec31 = p_380120_.position()
             .add(
                 p_380120_.relatives().contains(Relative.X) ? 0.0 : vec3.x(),
                 p_380120_.relatives().contains(Relative.Y) ? 0.0 : vec3.y(),
@@ -2923,9 +_,9 @@
 
     public boolean teleportTo(
         ServerLevel p_265257_,
-        double p_265407_,
+        me.alphamode.mcbig.math.BigDecimal p_265407_,
         double p_265727_,
-        double p_265410_,
+        me.alphamode.mcbig.math.BigDecimal p_265410_,
         Set<Relative> p_265083_,
         float p_265573_,
         float p_265094_,
@@ -2933,7 +_,7 @@
     ) {
         float f = Mth.clamp(p_265094_, -90.0F, 90.0F);
         Entity entity = this.teleport(
-            new TeleportTransition(p_265257_, new Vec3(p_265407_, p_265727_, p_265410_), Vec3.ZERO, p_265573_, f, p_265083_, TeleportTransition.DO_NOTHING)
+            new TeleportTransition(p_265257_, new me.alphamode.mcbig.core.BigVec3(p_265407_, p_265727_, p_265410_), Vec3.ZERO, p_265573_, f, p_265083_, TeleportTransition.DO_NOTHING)
         );
         return entity != null;
     }
@@ -2949,6 +_,13 @@
         }
     }
 
+    public void teleportTo(me.alphamode.mcbig.math.BigDecimal p_19887_, double p_19888_, me.alphamode.mcbig.math.BigDecimal p_19889_) {
+        if (this.level() instanceof ServerLevel) {
+            this.moveTo(p_19887_, p_19888_, p_19889_, this.getYRot(), this.getXRot());
+            this.teleportPassengers();
+        }
+    }
+
     private void teleportPassengers() {
         this.getSelfAndPassengers().forEach(p_185977_ -> {
             for (Entity entity : p_185977_.passengers) {
@@ -2958,7 +_,7 @@
     }
 
     public void teleportRelative(double p_249341_, double p_252229_, double p_252038_) {
-        this.teleportTo(this.getX() + p_249341_, this.getY() + p_252229_, this.getZ() + p_252038_);
+        this.teleportTo(this.getBigX().add(p_249341_), this.getY() + p_252229_, this.getBigZ().add(p_252038_));
     }
 
     public boolean shouldShowName() {
@@ -3261,7 +_,7 @@
     public CommandSourceStack createCommandSourceStackForNameResolution(ServerLevel p_376557_) {
         return new CommandSourceStack(
             CommandSource.NULL,
-            this.position(),
+            this.bigPosition(),
             this.getRotationVector(),
             p_376557_,
             0,
@@ -3273,7 +_,7 @@
     }
 
     public void lookAt(EntityAnchorArgument.Anchor p_20033_, Vec3 p_20034_) {
-        Vec3 vec3 = p_20033_.apply(this);
+        Vec3 vec3 = p_20033_.apply(this).toVanilla();
         double d0 = p_20034_.x - vec3.x;
         double d1 = p_20034_.y - vec3.y;
         double d2 = p_20034_.z - vec3.z;
@@ -3358,10 +_,10 @@
 
     public boolean touchingUnloadedChunk() {
         AABB aabb = this.getBoundingBox().inflate(1.0);
-        int i = Mth.floor(aabb.minX);
-        int j = Mth.ceil(aabb.maxX);
-        int k = Mth.floor(aabb.minZ);
-        int l = Mth.ceil(aabb.maxZ);
+        me.alphamode.mcbig.math.BigInteger i = Mth.bigFloor(aabb.minX);
+        me.alphamode.mcbig.math.BigInteger j = Mth.bigCeil(aabb.maxX);
+        me.alphamode.mcbig.math.BigInteger k = Mth.bigFloor(aabb.minZ);
+        me.alphamode.mcbig.math.BigInteger l = Mth.bigCeil(aabb.maxZ);
         return !this.level().hasChunksAt(i, k, j, l);
     }
 
@@ -3397,8 +_,12 @@
         return this.position;
     }
 
-    public Vec3 trackingPosition() {
-        return this.position();
+    public me.alphamode.mcbig.core.BigVec3 bigPosition() {
+        return this.bigPosition;
+    }
+
+    public me.alphamode.mcbig.core.BigVec3 trackingPosition() {
+        return this.bigPosition();
     }
 
     @Override
@@ -3434,14 +_,18 @@
         this.setDeltaMovement(new Vec3(p_20335_, p_20336_, p_20337_));
     }
 
-    public final int getBlockX() {
-        return this.blockPosition.getX();
+    public final me.alphamode.mcbig.math.BigInteger getBlockX() {
+        return this.blockPosition.getBigX();
     }
 
     public final double getX() {
         return this.position.x;
     }
 
+    public final me.alphamode.mcbig.math.BigDecimal getBigX() {
+        return this.bigPosition.x;
+    }
+
     public double getX(double p_20166_) {
         return this.position.x + (double)this.getBbWidth() * p_20166_;
     }
@@ -3454,6 +_,10 @@
         return this.blockPosition.getY();
     }
 
+    public final me.alphamode.mcbig.math.BigInteger getBigBlockY() {
+        return this.blockPosition.getBigY();
+    }
+
     public final double getY() {
         return this.position.y;
     }
@@ -3470,14 +_,18 @@
         return this.position.y + (double)this.eyeHeight;
     }
 
-    public final int getBlockZ() {
-        return this.blockPosition.getZ();
+    public final me.alphamode.mcbig.math.BigInteger getBlockZ() {
+        return this.blockPosition.getBigZ();
     }
 
     public final double getZ() {
         return this.position.z;
     }
 
+    public final me.alphamode.mcbig.math.BigDecimal getBigZ() {
+        return this.bigPosition.z;
+    }
+
     public double getZ(double p_20247_) {
         return this.position.z + (double)this.getBbWidth() * p_20247_;
     }
@@ -3486,16 +_,35 @@
         return this.getZ((2.0 * this.random.nextDouble() - 1.0) * p_20263_);
     }
 
-    public final void setPosRaw(double p_20344_, double p_20345_, double p_20346_) {
+    public void setPosRaw(double p_20344_, double p_20345_, double p_20346_) {
         if (this.position.x != p_20344_ || this.position.y != p_20345_ || this.position.z != p_20346_) {
             this.position = new Vec3(p_20344_, p_20345_, p_20346_);
-            int i = Mth.floor(p_20344_);
-            int j = Mth.floor(p_20345_);
-            int k = Mth.floor(p_20346_);
-            if (i != this.blockPosition.getX() || j != this.blockPosition.getY() || k != this.blockPosition.getZ()) {
-                this.blockPosition = new BlockPos(i, j, k);
-                this.inBlockState = null;
-                if (SectionPos.blockToSectionCoord(i) != this.chunkPosition.x || SectionPos.blockToSectionCoord(k) != this.chunkPosition.z) {
+            me.alphamode.mcbig.math.BigInteger i = Mth.bigFloor(p_20344_);
+            me.alphamode.mcbig.math.BigInteger j = Mth.bigFloor(p_20345_);
+            me.alphamode.mcbig.math.BigInteger k = Mth.bigFloor(p_20346_);
+            if (!i.equals(this.blockPosition.getBigX()) || !j.equals(this.blockPosition.getBigY()) || !k.equals(this.blockPosition.getBigZ())) {
+                this.blockPosition = new BlockPos(i, j, k);
+                this.inBlockState = null;
+                if (!SectionPos.blockToSectionCoord(i).equals(this.chunkPosition.x) || !SectionPos.blockToSectionCoord(k).equals(this.chunkPosition.z)) {
+                    this.chunkPosition = new ChunkPos(this.blockPosition);
+                }
+            }
+
+            this.levelCallback.onMove();
+        }
+    }
+
+    public void setPosRaw(me.alphamode.mcbig.math.BigDecimal p_20344_, double p_20345_, me.alphamode.mcbig.math.BigDecimal p_20346_) {
+        if (!this.bigPosition.x.equals(p_20344_) || this.bigPosition.y != p_20345_ || !this.bigPosition.z.equals(p_20346_)) {
+            this.bigPosition = new me.alphamode.mcbig.core.BigVec3(p_20344_, p_20345_, p_20346_);
+            this.position = this.bigPosition.toVanilla();
+            me.alphamode.mcbig.math.BigInteger i = Mth.bigFloor(p_20344_);
+            me.alphamode.mcbig.math.BigInteger j = Mth.bigFloor(p_20345_);
+            me.alphamode.mcbig.math.BigInteger k = Mth.bigFloor(p_20346_);
+            if (!i.equals(this.blockPosition.getBigX()) || !j.equals(this.blockPosition.getBigY()) || !k.equals(this.blockPosition.getBigZ())) {
+                this.blockPosition = new BlockPos(i, j, k);
+                this.inBlockState = null;
+                if (!SectionPos.blockToSectionCoord(i).equals(this.chunkPosition.x) || !SectionPos.blockToSectionCoord(k).equals(this.chunkPosition.z)) {
                     this.chunkPosition = new ChunkPos(this.blockPosition);
                 }
             }
@@ -3513,9 +_,9 @@
 
     public void recreateFromPacket(ClientboundAddEntityPacket p_146866_) {
         int i = p_146866_.getId();
-        double d0 = p_146866_.getX();
+        me.alphamode.mcbig.math.BigDecimal d0 = p_146866_.getX();
         double d1 = p_146866_.getY();
-        double d2 = p_146866_.getZ();
+        me.alphamode.mcbig.math.BigDecimal d2 = p_146866_.getZ();
         this.syncPacketPositionCodec(d0, d1, d2);
         this.moveTo(d0, d1, d2, p_146866_.getYRot(), p_146866_.getXRot());
         this.setId(i);
