--- a/net/minecraft/world/entity/Mob.java
+++ b/net/minecraft/world/entity/Mob.java
@@ -403,9 +_,9 @@
                 compoundtag2.putUUID("UUID", uuid);
             } else if (this.leashHolder instanceof HangingEntity) {
                 BlockPos blockpos = ((HangingEntity)this.leashHolder).getPos();
-                compoundtag2.putInt("X", blockpos.getX());
-                compoundtag2.putInt("Y", blockpos.getY());
-                compoundtag2.putInt("Z", blockpos.getZ());
+                compoundtag2.putByteArray("X", blockpos.getBigX().toByteArray());
+                compoundtag2.putByteArray("Y", blockpos.getBigY().toByteArray());
+                compoundtag2.putByteArray("Z", blockpos.getBigZ().toByteArray());
             }
 
             p_21484_.put("Leash", compoundtag2);
