--- a/net/minecraft/gametest/framework/GameTestHelper.java
+++ b/net/minecraft/gametest/framework/GameTestHelper.java
@@ -232,7 +_,7 @@
 
     public void useBlock(BlockPos p_250131_, Player p_251507_) {
         BlockPos blockpos = this.absolutePos(p_250131_);
-        this.useBlock(p_250131_, p_251507_, new BlockHitResult(Vec3.atCenterOf(blockpos), Direction.NORTH, blockpos, true));
+        this.useBlock(p_250131_, p_251507_, new BlockHitResult(me.alphamode.mcbig.core.BigVec3.atCenterOf(blockpos), Direction.NORTH, blockpos, true));
     }
 
     public void useBlock(BlockPos p_262023_, Player p_261901_, BlockHitResult p_262040_) {
@@ -683,7 +_,7 @@
 
     public void assertSameBlockStates(BoundingBox p_177225_, BlockPos p_177226_) {
         BlockPos.betweenClosedStream(p_177225_).forEach(p_177267_ -> {
-            BlockPos blockpos = p_177226_.offset(p_177267_.getX() - p_177225_.minX(), p_177267_.getY() - p_177225_.minY(), p_177267_.getZ() - p_177225_.minZ());
+            BlockPos blockpos = p_177226_.offset(p_177267_.getBigX().subtract(p_177225_.minX()), p_177267_.getBigY().subtract(p_177225_.minY()), p_177267_.getBigZ().subtract(p_177225_.minZ()));
             this.assertSameBlockState(p_177267_, blockpos);
         });
     }
@@ -901,7 +_,7 @@
 
     public void placeAt(Player p_261595_, ItemStack p_262007_, BlockPos p_261973_, Direction p_262008_) {
         BlockPos blockpos = this.absolutePos(p_261973_.relative(p_262008_));
-        BlockHitResult blockhitresult = new BlockHitResult(Vec3.atCenterOf(blockpos), p_262008_, blockpos, false);
+        BlockHitResult blockhitresult = new BlockHitResult(me.alphamode.mcbig.core.BigVec3.atCenterOf(blockpos), p_262008_, blockpos, false);
         UseOnContext useoncontext = new UseOnContext(p_261595_, InteractionHand.MAIN_HAND, blockhitresult);
         p_262007_.useOn(useoncontext);
     }
