--- a/net/minecraft/gametest/framework/GameTestBatchRunner.java
+++ b/net/minecraft/gametest/framework/GameTestBatchRunner.java
@@ -75,8 +_,8 @@
                 private void testCompleted() {
                     if (multipletesttracker.isDone()) {
                         gametestbatch.runAfterBatchFunction(GameTestBatchRunner.this.level);
-                        LongSet longset = new LongArraySet(GameTestBatchRunner.this.level.getForcedChunks());
-                        longset.forEach(p_309120_ -> GameTestBatchRunner.this.level.setChunkForced(ChunkPos.getX(p_309120_), ChunkPos.getZ(p_309120_), false));
+                        it.unimi.dsi.fastutil.objects.ObjectSet<ChunkPos> longset = new it.unimi.dsi.fastutil.objects.ObjectArraySet<>(GameTestBatchRunner.this.level.getForcedChunks());
+                        longset.forEach(p_309120_ -> GameTestBatchRunner.this.level.setChunkForced(p_309120_.x(), p_309120_.z(), false));
                         GameTestBatchRunner.this.runBatch(p_127571_ + 1);
                     }
                 }
