--- a/net/minecraft/world/level/levelgen/structure/structures/OceanRuinPieces.java
+++ b/net/minecraft/world/level/levelgen/structure/structures/OceanRuinPieces.java
@@ -171,10 +_,10 @@
         OceanRuinStructure p_228992_,
         StructurePieceAccessor p_228993_
     ) {
-        BlockPos blockpos = new BlockPos(p_228991_.getX(), 90, p_228991_.getZ());
+        BlockPos blockpos = new BlockPos(p_228991_.getBigX(), 90, p_228991_.getBigZ());
         BlockPos blockpos1 = StructureTemplate.transform(new BlockPos(15, 0, 15), Mirror.NONE, p_228990_, BlockPos.ZERO).offset(blockpos);
         BoundingBox boundingbox = BoundingBox.fromCorners(blockpos, blockpos1);
-        BlockPos blockpos2 = new BlockPos(Math.min(blockpos.getX(), blockpos1.getX()), blockpos.getY(), Math.min(blockpos.getZ(), blockpos1.getZ()));
+        BlockPos blockpos2 = new BlockPos(blockpos.getBigX().min(blockpos1.getBigX()), blockpos.getBigY(), blockpos.getBigZ().min(blockpos1.getBigZ()));
         List<BlockPos> list = allPositions(p_228989_, blockpos2);
         int i = Mth.nextInt(p_228989_, 4, 8);
 
@@ -342,8 +_,8 @@
             ChunkPos p_229034_,
             BlockPos p_229035_
         ) {
-            int i = p_229029_.getHeight(Heightmap.Types.OCEAN_FLOOR_WG, this.templatePosition.getX(), this.templatePosition.getZ());
-            this.templatePosition = new BlockPos(this.templatePosition.getX(), i, this.templatePosition.getZ());
+            int i = p_229029_.getHeight(Heightmap.Types.OCEAN_FLOOR_WG, this.templatePosition.getBigX(), this.templatePosition.getBigZ());
+            this.templatePosition = new BlockPos(this.templatePosition.getBigX(), i, this.templatePosition.getBigZ());
             BlockPos blockpos = StructureTemplate.transform(
                     new BlockPos(this.template.getSize().getX() - 1, 0, this.template.getSize().getZ() - 1),
                     Mirror.NONE,
@@ -352,7 +_,7 @@
                 )
                 .offset(this.templatePosition);
             this.templatePosition = new BlockPos(
-                this.templatePosition.getX(), this.getHeight(this.templatePosition, p_229029_, blockpos), this.templatePosition.getZ()
+                this.templatePosition.getBigX(), this.getHeight(this.templatePosition, p_229029_, blockpos), this.templatePosition.getBigZ()
             );
             super.postProcess(p_229029_, p_229030_, p_229031_, p_229032_, p_229033_, p_229034_, p_229035_);
         }
