--- a/net/minecraft/world/level/levelgen/structure/structures/ShipwreckStructure.java
+++ b/net/minecraft/world/level/levelgen/structure/structures/ShipwreckStructure.java
@@ -41,7 +_,7 @@
             BoundingBox boundingbox = shipwreckpieces$shipwreckpiece.getBoundingBox();
             int i;
             if (this.isBeached) {
-                int j = Structure.getLowestY(p_229397_, boundingbox.minX(), boundingbox.getXSpan(), boundingbox.minZ(), boundingbox.getZSpan());
+                int j = Structure.getLowestY(p_229397_, boundingbox.minX(), me.alphamode.mcbig.math.BigInteger.val(boundingbox.getXSpan()), boundingbox.minZ(), boundingbox.getZSpan());
                 i = shipwreckpieces$shipwreckpiece.calculateBeachedPosition(j, p_229397_.random());
             } else {
                 i = Structure.getMeanFirstOccupiedHeight(p_229397_, boundingbox.minX(), boundingbox.getXSpan(), boundingbox.minZ(), boundingbox.getZSpan());
