--- a/net/minecraft/world/level/levelgen/structure/structures/StrongholdStructure.java
+++ b/net/minecraft/world/level/levelgen/structure/structures/StrongholdStructure.java
@@ -26,7 +_,7 @@
         StrongholdPieces.StartPiece strongholdpieces$startpiece;
         do {
             p_229946_.clear();
-            p_229947_.random().setLargeFeatureSeed(p_229947_.seed() + (long)(i++), p_229947_.chunkPos().x, p_229947_.chunkPos().z);
+            p_229947_.random().setLargeFeatureSeed(p_229947_.seed() + (long)(i++), p_229947_.chunkPos().x().longValue(), p_229947_.chunkPos().z().longValue());
             StrongholdPieces.resetPieces();
             strongholdpieces$startpiece = new StrongholdPieces.StartPiece(
                 p_229947_.random(), p_229947_.chunkPos().getBlockX(2), p_229947_.chunkPos().getBlockZ(2)
