--- a/net/minecraft/world/level/levelgen/structure/structures/SwampHutPiece.java
+++ b/net/minecraft/world/level/levelgen/structure/structures/SwampHutPiece.java
@@ -26,7 +_,7 @@
     private boolean spawnedWitch;
     private boolean spawnedCat;
 
-    public SwampHutPiece(RandomSource p_229952_, int p_229953_, int p_229954_) {
+    public SwampHutPiece(RandomSource p_229952_, me.alphamode.mcbig.math.BigInteger p_229953_, me.alphamode.mcbig.math.BigInteger p_229954_) {
         super(StructurePieceType.SWAMPLAND_HUT, p_229953_, 64, p_229954_, 7, 7, 9, getRandomHorizontalDirection(p_229952_));
     }
 
@@ -101,7 +_,7 @@
                     Witch witch = EntityType.WITCH.create(p_229961_.getLevel());
                     if (witch != null) {
                         witch.setPersistenceRequired();
-                        witch.moveTo((double)blockpos.getX() + 0.5, (double)blockpos.getY(), (double)blockpos.getZ() + 0.5, 0.0F, 0.0F);
+                        witch.moveTo((double)blockpos.getBigX().doubleValue() + 0.5, (double)blockpos.getBigY().doubleValue(), (double)blockpos.getBigZ().doubleValue() + 0.5, 0.0F, 0.0F);
                         witch.finalizeSpawn(p_229961_, p_229961_.getCurrentDifficultyAt(blockpos), MobSpawnType.STRUCTURE, null);
                         p_229961_.addFreshEntityWithPassengers(witch);
                     }
@@ -120,7 +_,7 @@
                 Cat cat = EntityType.CAT.create(p_229958_.getLevel());
                 if (cat != null) {
                     cat.setPersistenceRequired();
-                    cat.moveTo((double)blockpos.getX() + 0.5, (double)blockpos.getY(), (double)blockpos.getZ() + 0.5, 0.0F, 0.0F);
+                    cat.moveTo((double)blockpos.getBigX().doubleValue() + 0.5, (double)blockpos.getBigY().doubleValue(), (double)blockpos.getBigZ().doubleValue() + 0.5, 0.0F, 0.0F);
                     cat.finalizeSpawn(p_229958_, p_229958_.getCurrentDifficultyAt(blockpos), MobSpawnType.STRUCTURE, null);
                     p_229958_.addFreshEntityWithPassengers(cat);
                 }
