--- a/net/minecraft/world/entity/projectile/AbstractArrow.java
+++ b/net/minecraft/world/entity/projectile/AbstractArrow.java
@@ -265,10 +_,10 @@
     private void stepMoveAndHit(BlockHitResult p_371761_) {
         while (this.isAlive()) {
             Vec3 vec3 = this.position();
-            EntityHitResult entityhitresult = this.findHitEntity(vec3, p_371761_.getLocation());
-            Vec3 vec31 = Objects.requireNonNullElse(entityhitresult, p_371761_).getLocation();
+            EntityHitResult entityhitresult = this.findHitEntity(vec3, p_371761_.getLocation().toVanilla());
+            me.alphamode.mcbig.core.BigVec3 vec31 = Objects.requireNonNullElse(entityhitresult, p_371761_).getLocation();
             this.setPos(vec31);
-            this.applyEffectsFromBlocks(vec3, vec31);
+            this.applyEffectsFromBlocks(vec3, vec31.toVanilla());
             if (this.portalProcess != null && this.portalProcess.isInsidePortalThisTick()) {
                 this.handlePortal();
             }
@@ -512,7 +_,7 @@
     }
 
     protected void hitBlockEnchantmentEffects(ServerLevel p_345462_, BlockHitResult p_345204_, ItemStack p_345083_) {
-        Vec3 vec3 = p_345204_.getBlockPos().clampLocationWithin(p_345204_.getLocation());
+        Vec3 vec3 = p_345204_.getBlockPos().clampLocationWithin(p_345204_.getLocation().toVanilla());
         EnchantmentHelper.onHitBlock(
             p_345462_,
             p_345083_,
