--- a/net/minecraft/world/entity/projectile/FireworkRocketEntity.java
+++ b/net/minecraft/world/entity/projectile/FireworkRocketEntity.java
@@ -227,7 +_,7 @@
 
                     for(int i = 0; i < 2; ++i) {
                         Vec3 vec31 = new Vec3(livingentity.getX(), livingentity.getY(0.5 * (double)i), livingentity.getZ());
-                        HitResult hitresult = this.level().clip(new ClipContext(vec3, vec31, ClipContext.Block.COLLIDER, ClipContext.Fluid.NONE, this));
+                        HitResult hitresult = this.level().clip(new ClipContext(vec3.toBig(), vec31.toBig(), ClipContext.Block.COLLIDER, ClipContext.Fluid.NONE, this));
                         if (hitresult.getType() == HitResult.Type.MISS) {
                             flag = true;
                             break;
