--- a/net/minecraft/world/entity/projectile/ThrownEnderpearl.java
+++ b/net/minecraft/world/entity/projectile/ThrownEnderpearl.java
@@ -140,7 +_,7 @@
 
                         ServerPlayer serverplayer1 = serverplayer.teleport(
                             new TeleportTransition(
-                                serverlevel, vec3, Vec3.ZERO, 0.0F, 0.0F, Relative.union(Relative.ROTATION, Relative.DELTA), TeleportTransition.DO_NOTHING
+                                serverlevel, vec3.toBig(), Vec3.ZERO, 0.0F, 0.0F, Relative.union(Relative.ROTATION, Relative.DELTA), TeleportTransition.DO_NOTHING
                             )
                         );
                         if (serverplayer1 != null) {
@@ -153,7 +_,7 @@
                     }
                 } else {
                     Entity entity1 = entity.teleport(
-                        new TeleportTransition(serverlevel, vec3, entity.getDeltaMovement(), entity.getYRot(), entity.getXRot(), TeleportTransition.DO_NOTHING)
+                        new TeleportTransition(serverlevel, vec3.toBig(), entity.getDeltaMovement(), entity.getYRot(), entity.getXRot(), TeleportTransition.DO_NOTHING)
                     );
                     if (entity1 != null) {
                         entity1.resetFallDistance();
