--- a/net/minecraft/world/level/pathfinder/AmphibiousNodeEvaluator.java
+++ b/net/minecraft/world/level/pathfinder/AmphibiousNodeEvaluator.java
@@ -92,7 +_,7 @@
     }
 
     @Override
-    public PathType getPathType(PathfindingContext p_330489_, int p_164667_, int p_164668_, int p_164669_) {
+    public PathType getPathType(PathfindingContext p_330489_, me.alphamode.mcbig.math.BigInteger p_164667_, int p_164668_, me.alphamode.mcbig.math.BigInteger p_164669_) {
         PathType pathtype = p_330489_.getPathTypeFromState(p_164667_, p_164668_, p_164669_);
         if (pathtype == PathType.WATER) {
             BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
@@ -100,7 +_,7 @@
             for (Direction direction : Direction.values()) {
                 blockpos$mutableblockpos.set(p_164667_, p_164668_, p_164669_).move(direction);
                 PathType pathtype1 = p_330489_.getPathTypeFromState(
-                    blockpos$mutableblockpos.getX(), blockpos$mutableblockpos.getY(), blockpos$mutableblockpos.getZ()
+                    blockpos$mutableblockpos.getBigX(), blockpos$mutableblockpos.getY(), blockpos$mutableblockpos.getBigZ()
                 );
                 if (pathtype1 == PathType.BLOCKED) {
                     return PathType.WATER_BORDER;
