--- a/net/minecraft/world/level/pathfinder/FlyNodeEvaluator.java
+++ b/net/minecraft/world/level/pathfinder/FlyNodeEvaluator.java
@@ -13,7 +_,7 @@
 import net.minecraft.world.phys.AABB;
 
 public class FlyNodeEvaluator extends WalkNodeEvaluator {
-    private final Long2ObjectMap<PathType> pathTypeByPosCache = new Long2ObjectOpenHashMap<>();
+    private final it.unimi.dsi.fastutil.objects.Object2ObjectMap<BlockPos, PathType> pathTypeByPosCache = new it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap<>();
     private static final float SMALL_MOB_SIZE = 1.0F;
     private static final float SMALL_MOB_INFLATED_START_NODE_BOUNDING_BOX = 1.1F;
     private static final int MAX_START_NODE_CANDIDATES = 10;
@@ -63,7 +_,7 @@
 
     @Override
     protected boolean canStartAt(BlockPos p_262645_) {
-        PathType pathtype = this.getCachedPathType(p_262645_.getX(), p_262645_.getY(), p_262645_.getZ());
+        PathType pathtype = this.getCachedPathType(p_262645_.getBigX(), p_262645_.getBigY(), p_262645_.getBigZ());
         return this.mob.getPathfindingMalus(pathtype) >= 0.0F;
     }
 
@@ -75,97 +_,97 @@
     @Override
     public int getNeighbors(Node[] p_77266_, Node p_77267_) {
         int i = 0;
-        Node node = this.findAcceptedNode(p_77267_.x, p_77267_.y, p_77267_.z + 1);
+        Node node = this.findAcceptedNode(p_77267_.x, p_77267_.y, p_77267_.z.add());
         if (this.isOpen(node)) {
             p_77266_[i++] = node;
         }
 
-        Node node1 = this.findAcceptedNode(p_77267_.x - 1, p_77267_.y, p_77267_.z);
+        Node node1 = this.findAcceptedNode(p_77267_.x.subtract(), p_77267_.y, p_77267_.z);
         if (this.isOpen(node1)) {
             p_77266_[i++] = node1;
         }
 
-        Node node2 = this.findAcceptedNode(p_77267_.x + 1, p_77267_.y, p_77267_.z);
+        Node node2 = this.findAcceptedNode(p_77267_.x.add(), p_77267_.y, p_77267_.z);
         if (this.isOpen(node2)) {
             p_77266_[i++] = node2;
         }
 
-        Node node3 = this.findAcceptedNode(p_77267_.x, p_77267_.y, p_77267_.z - 1);
+        Node node3 = this.findAcceptedNode(p_77267_.x, p_77267_.y, p_77267_.z.subtract());
         if (this.isOpen(node3)) {
             p_77266_[i++] = node3;
         }
 
-        Node node4 = this.findAcceptedNode(p_77267_.x, p_77267_.y + 1, p_77267_.z);
+        Node node4 = this.findAcceptedNode(p_77267_.x, p_77267_.y.add(), p_77267_.z);
         if (this.isOpen(node4)) {
             p_77266_[i++] = node4;
         }
 
-        Node node5 = this.findAcceptedNode(p_77267_.x, p_77267_.y - 1, p_77267_.z);
+        Node node5 = this.findAcceptedNode(p_77267_.x, p_77267_.y.subtract(), p_77267_.z);
         if (this.isOpen(node5)) {
             p_77266_[i++] = node5;
         }
 
-        Node node6 = this.findAcceptedNode(p_77267_.x, p_77267_.y + 1, p_77267_.z + 1);
+        Node node6 = this.findAcceptedNode(p_77267_.x, p_77267_.y.add(), p_77267_.z.add());
         if (this.isOpen(node6) && this.hasMalus(node) && this.hasMalus(node4)) {
             p_77266_[i++] = node6;
         }
 
-        Node node7 = this.findAcceptedNode(p_77267_.x - 1, p_77267_.y + 1, p_77267_.z);
+        Node node7 = this.findAcceptedNode(p_77267_.x.subtract(), p_77267_.y.add(), p_77267_.z);
         if (this.isOpen(node7) && this.hasMalus(node1) && this.hasMalus(node4)) {
             p_77266_[i++] = node7;
         }
 
-        Node node8 = this.findAcceptedNode(p_77267_.x + 1, p_77267_.y + 1, p_77267_.z);
+        Node node8 = this.findAcceptedNode(p_77267_.x.add(), p_77267_.y.add(), p_77267_.z);
         if (this.isOpen(node8) && this.hasMalus(node2) && this.hasMalus(node4)) {
             p_77266_[i++] = node8;
         }
 
-        Node node9 = this.findAcceptedNode(p_77267_.x, p_77267_.y + 1, p_77267_.z - 1);
+        Node node9 = this.findAcceptedNode(p_77267_.x, p_77267_.y.add(), p_77267_.z.subtract());
         if (this.isOpen(node9) && this.hasMalus(node3) && this.hasMalus(node4)) {
             p_77266_[i++] = node9;
         }
 
-        Node node10 = this.findAcceptedNode(p_77267_.x, p_77267_.y - 1, p_77267_.z + 1);
+        Node node10 = this.findAcceptedNode(p_77267_.x, p_77267_.y.subtract(), p_77267_.z.add());
         if (this.isOpen(node10) && this.hasMalus(node) && this.hasMalus(node5)) {
             p_77266_[i++] = node10;
         }
 
-        Node node11 = this.findAcceptedNode(p_77267_.x - 1, p_77267_.y - 1, p_77267_.z);
+        Node node11 = this.findAcceptedNode(p_77267_.x.subtract(), p_77267_.y.subtract(), p_77267_.z);
         if (this.isOpen(node11) && this.hasMalus(node1) && this.hasMalus(node5)) {
             p_77266_[i++] = node11;
         }
 
-        Node node12 = this.findAcceptedNode(p_77267_.x + 1, p_77267_.y - 1, p_77267_.z);
+        Node node12 = this.findAcceptedNode(p_77267_.x.add(), p_77267_.y.subtract(), p_77267_.z);
         if (this.isOpen(node12) && this.hasMalus(node2) && this.hasMalus(node5)) {
             p_77266_[i++] = node12;
         }
 
-        Node node13 = this.findAcceptedNode(p_77267_.x, p_77267_.y - 1, p_77267_.z - 1);
+        Node node13 = this.findAcceptedNode(p_77267_.x, p_77267_.y.subtract(), p_77267_.z.subtract());
         if (this.isOpen(node13) && this.hasMalus(node3) && this.hasMalus(node5)) {
             p_77266_[i++] = node13;
         }
 
-        Node node14 = this.findAcceptedNode(p_77267_.x + 1, p_77267_.y, p_77267_.z - 1);
+        Node node14 = this.findAcceptedNode(p_77267_.x.add(), p_77267_.y, p_77267_.z.subtract());
         if (this.isOpen(node14) && this.hasMalus(node3) && this.hasMalus(node2)) {
             p_77266_[i++] = node14;
         }
 
-        Node node15 = this.findAcceptedNode(p_77267_.x + 1, p_77267_.y, p_77267_.z + 1);
+        Node node15 = this.findAcceptedNode(p_77267_.x.add(), p_77267_.y, p_77267_.z.add());
         if (this.isOpen(node15) && this.hasMalus(node) && this.hasMalus(node2)) {
             p_77266_[i++] = node15;
         }
 
-        Node node16 = this.findAcceptedNode(p_77267_.x - 1, p_77267_.y, p_77267_.z - 1);
+        Node node16 = this.findAcceptedNode(p_77267_.x.subtract(), p_77267_.y, p_77267_.z.subtract());
         if (this.isOpen(node16) && this.hasMalus(node3) && this.hasMalus(node1)) {
             p_77266_[i++] = node16;
         }
 
-        Node node17 = this.findAcceptedNode(p_77267_.x - 1, p_77267_.y, p_77267_.z + 1);
+        Node node17 = this.findAcceptedNode(p_77267_.x.subtract(), p_77267_.y, p_77267_.z.add());
         if (this.isOpen(node17) && this.hasMalus(node) && this.hasMalus(node1)) {
             p_77266_[i++] = node17;
         }
 
-        Node node18 = this.findAcceptedNode(p_77267_.x + 1, p_77267_.y + 1, p_77267_.z - 1);
+        Node node18 = this.findAcceptedNode(p_77267_.x.add(), p_77267_.y.add(), p_77267_.z.subtract());
         if (this.isOpen(node18)
             && this.hasMalus(node14)
             && this.hasMalus(node3)
@@ -176,7 +_,7 @@
             p_77266_[i++] = node18;
         }
 
-        Node node19 = this.findAcceptedNode(p_77267_.x + 1, p_77267_.y + 1, p_77267_.z + 1);
+        Node node19 = this.findAcceptedNode(p_77267_.x.add(), p_77267_.y.add(), p_77267_.z.add());
         if (this.isOpen(node19)
             && this.hasMalus(node15)
             && this.hasMalus(node)
@@ -187,7 +_,7 @@
             p_77266_[i++] = node19;
         }
 
-        Node node20 = this.findAcceptedNode(p_77267_.x - 1, p_77267_.y + 1, p_77267_.z - 1);
+        Node node20 = this.findAcceptedNode(p_77267_.x.subtract(), p_77267_.y.add(), p_77267_.z.subtract());
         if (this.isOpen(node20)
             && this.hasMalus(node16)
             && this.hasMalus(node3)
@@ -198,7 +_,7 @@
             p_77266_[i++] = node20;
         }
 
-        Node node21 = this.findAcceptedNode(p_77267_.x - 1, p_77267_.y + 1, p_77267_.z + 1);
+        Node node21 = this.findAcceptedNode(p_77267_.x.subtract(), p_77267_.y.add(), p_77267_.z.add());
         if (this.isOpen(node21)
             && this.hasMalus(node17)
             && this.hasMalus(node)
@@ -209,7 +_,7 @@
             p_77266_[i++] = node21;
         }
 
-        Node node22 = this.findAcceptedNode(p_77267_.x + 1, p_77267_.y - 1, p_77267_.z - 1);
+        Node node22 = this.findAcceptedNode(p_77267_.x.add(), p_77267_.y.subtract(), p_77267_.z.subtract());
         if (this.isOpen(node22)
             && this.hasMalus(node14)
             && this.hasMalus(node3)
@@ -220,7 +_,7 @@
             p_77266_[i++] = node22;
         }
 
-        Node node23 = this.findAcceptedNode(p_77267_.x + 1, p_77267_.y - 1, p_77267_.z + 1);
+        Node node23 = this.findAcceptedNode(p_77267_.x.add(), p_77267_.y.subtract(), p_77267_.z.add());
         if (this.isOpen(node23)
             && this.hasMalus(node15)
             && this.hasMalus(node)
@@ -231,7 +_,7 @@
             p_77266_[i++] = node23;
         }
 
-        Node node24 = this.findAcceptedNode(p_77267_.x - 1, p_77267_.y - 1, p_77267_.z - 1);
+        Node node24 = this.findAcceptedNode(p_77267_.x.subtract(), p_77267_.y.subtract(), p_77267_.z.subtract());
         if (this.isOpen(node24)
             && this.hasMalus(node16)
             && this.hasMalus(node3)
@@ -242,7 +_,7 @@
             p_77266_[i++] = node24;
         }
 
-        Node node25 = this.findAcceptedNode(p_77267_.x - 1, p_77267_.y - 1, p_77267_.z + 1);
+        Node node25 = this.findAcceptedNode(p_77267_.x.subtract(), p_77267_.y.subtract(), p_77267_.z.add());
         if (this.isOpen(node25)
             && this.hasMalus(node17)
             && this.hasMalus(node)
@@ -265,7 +_,7 @@
     }
 
     @Nullable
-    protected Node findAcceptedNode(int p_262970_, int p_263018_, int p_262947_) {
+    protected Node findAcceptedNode(me.alphamode.mcbig.math.BigInteger p_262970_, me.alphamode.mcbig.math.BigInteger p_263018_, me.alphamode.mcbig.math.BigInteger p_262947_) {
         Node node = null;
         PathType pathtype = this.getCachedPathType(p_262970_, p_263018_, p_262947_);
         float f = this.mob.getPathfindingMalus(pathtype);
@@ -282,20 +_,20 @@
     }
 
     @Override
-    protected PathType getCachedPathType(int p_326867_, int p_326833_, int p_326788_) {
+    protected PathType getCachedPathType(me.alphamode.mcbig.math.BigInteger p_326867_, me.alphamode.mcbig.math.BigInteger p_326833_, me.alphamode.mcbig.math.BigInteger p_326788_) {
         return this.pathTypeByPosCache
             .computeIfAbsent(
-                BlockPos.asLong(p_326867_, p_326833_, p_326788_),
+                new BlockPos(p_326867_, p_326833_, p_326788_),
                 p_330153_ -> this.getPathTypeOfMob(this.currentContext, p_326867_, p_326833_, p_326788_, this.mob)
             );
     }
 
     @Override
-    public PathType getPathType(PathfindingContext p_331058_, int p_77246_, int p_77247_, int p_77248_) {
+    public PathType getPathType(PathfindingContext p_331058_, me.alphamode.mcbig.math.BigInteger p_77246_, me.alphamode.mcbig.math.BigInteger p_77247_, me.alphamode.mcbig.math.BigInteger p_77248_) {
         PathType pathtype = p_331058_.getPathTypeFromState(p_77246_, p_77247_, p_77248_);
-        if (pathtype == PathType.OPEN && p_77247_ >= p_331058_.level().getMinBuildHeight() + 1) {
-            BlockPos blockpos = new BlockPos(p_77246_, p_77247_ - 1, p_77248_);
-            PathType pathtype1 = p_331058_.getPathTypeFromState(blockpos.getX(), blockpos.getY(), blockpos.getZ());
+        if (pathtype == PathType.OPEN && p_77247_.compareTo(p_331058_.level().getMinBuildHeight() + 1) >= 0) {
+            BlockPos blockpos = new BlockPos(p_77246_, p_77247_.subtract(), p_77248_);
+            PathType pathtype1 = p_331058_.getPathTypeFromState(blockpos.getBigX(), blockpos.getBigY(), blockpos.getBigZ());
             if (pathtype1 == PathType.DAMAGE_FIRE || pathtype1 == PathType.LAVA) {
                 pathtype = PathType.DAMAGE_FIRE;
             } else if (pathtype1 == PathType.DAMAGE_OTHER) {
@@ -336,12 +_,12 @@
             return BlockPos.randomBetweenClosed(
                 p_263108_.getRandom(),
                 10,
-                Mth.floor(aabb1.minX),
-                Mth.floor(aabb1.minY),
-                Mth.floor(aabb1.minZ),
-                Mth.floor(aabb1.maxX),
-                Mth.floor(aabb1.maxY),
-                Mth.floor(aabb1.maxZ)
+                Mth.bigFloor(aabb1.minX),
+                Mth.bigFloor(aabb1.minY),
+                Mth.bigFloor(aabb1.minZ),
+                Mth.bigFloor(aabb1.maxX),
+                Mth.bigFloor(aabb1.maxY),
+                Mth.bigFloor(aabb1.maxZ)
             );
         }
     }
