--- a/net/minecraft/world/level/pathfinder/NodeEvaluator.java
+++ b/net/minecraft/world/level/pathfinder/NodeEvaluator.java
@@ -38,11 +_,11 @@
     }
 
     protected Node getNode(BlockPos p_77350_) {
-        return this.getNode(p_77350_.getX(), p_77350_.getY(), p_77350_.getZ());
+        return this.getNode(p_77350_.getBigX(), p_77350_.getBigY(), p_77350_.getBigZ());
     }
 
-    protected Node getNode(int p_77325_, int p_77326_, int p_77327_) {
-        return this.nodes.computeIfAbsent(Node.createHash(p_77325_, p_77326_, p_77327_), p_77332_ -> new Node(p_77325_, p_77326_, p_77327_));
+    protected Node getNode(me.alphamode.mcbig.math.BigInteger p_77325_, me.alphamode.mcbig.math.BigInteger p_77326_, me.alphamode.mcbig.math.BigInteger p_77327_) {
+        return this.nodes.computeIfAbsent(Node.createHash(p_77325_.intValue(), p_77326_.intValue(), p_77327_.intValue()), p_77332_ -> new Node(p_77325_, p_77326_, p_77327_));
     }
 
     public abstract Node getStart();
@@ -50,17 +_,17 @@
     public abstract Target getTarget(double p_326792_, double p_326849_, double p_326903_);
 
     protected Target getTargetNodeAt(double p_326819_, double p_326953_, double p_326931_) {
-        return new Target(this.getNode(Mth.floor(p_326819_), Mth.floor(p_326953_), Mth.floor(p_326931_)));
+        return new Target(this.getNode(Mth.bigFloor(p_326819_), Mth.bigFloor(p_326953_), Mth.bigFloor(p_326931_)));
     }
 
     public abstract int getNeighbors(Node[] p_77353_, Node p_77354_);
 
-    public abstract PathType getPathTypeOfMob(PathfindingContext p_330720_, int p_326783_, int p_326930_, int p_326884_, Mob p_326798_);
+    public abstract PathType getPathTypeOfMob(PathfindingContext p_330720_, me.alphamode.mcbig.math.BigInteger p_326783_, me.alphamode.mcbig.math.BigInteger p_326930_, me.alphamode.mcbig.math.BigInteger p_326884_, Mob p_326798_);
 
-    public abstract PathType getPathType(PathfindingContext p_330531_, int p_331795_, int p_332013_, int p_331675_);
+    public abstract PathType getPathType(PathfindingContext p_330531_, me.alphamode.mcbig.math.BigInteger p_331795_, me.alphamode.mcbig.math.BigInteger p_332013_, me.alphamode.mcbig.math.BigInteger p_331675_);
 
     public PathType getPathType(Mob p_331317_, BlockPos p_331263_) {
-        return this.getPathType(new PathfindingContext(p_331317_.level(), p_331317_), p_331263_.getX(), p_331263_.getY(), p_331263_.getZ());
+        return this.getPathType(new PathfindingContext(p_331317_.level(), p_331317_), p_331263_.getBigX(), p_331263_.getBigY(), p_331263_.getBigZ());
     }
 
     public void setCanPassDoors(boolean p_77352_) {
