--- a/net/minecraft/world/level/levelgen/carver/CaveWorldCarver.java
+++ b/net/minecraft/world/level/levelgen/carver/CaveWorldCarver.java
@@ -36,9 +_,9 @@
         int j = p_224889_.nextInt(p_224889_.nextInt(p_224889_.nextInt(this.getCaveBound()) + 1) + 1);
 
         for(int k = 0; k < j; ++k) {
-            double d0 = (double)p_224891_.getBlockX(p_224889_.nextInt(16));
+            double d0 = (double)p_224891_.getBlockX(p_224889_.nextInt(16)).doubleValue();
             double d1 = (double)p_224886_.y.sample(p_224889_, p_224885_);
-            double d2 = (double)p_224891_.getBlockZ(p_224889_.nextInt(16));
+            double d2 = (double)p_224891_.getBlockZ(p_224889_.nextInt(16)).intValue();
             double d3 = (double)p_224886_.horizontalRadiusMultiplier.sample(p_224889_);
             double d4 = (double)p_224886_.verticalRadiusMultiplier.sample(p_224889_);
             double d5 = (double)p_224886_.floorLevel.sample(p_224889_);
