--- a/net/minecraft/world/entity/animal/frog/Frog.java
+++ b/net/minecraft/world/entity/animal/frog/Frog.java
@@ -390,8 +_,8 @@
         }
 
         @Override
-        public PathType getPathType(PathfindingContext p_331446_, int p_326799_, int p_326899_, int p_326891_) {
-            this.belowPos.set(p_326799_, p_326899_ - 1, p_326891_);
+        public PathType getPathType(PathfindingContext p_331446_, me.alphamode.mcbig.math.BigInteger p_326799_, me.alphamode.mcbig.math.BigInteger p_326899_, me.alphamode.mcbig.math.BigInteger p_326891_) {
+            this.belowPos.set(p_326799_, p_326899_.subtract(), p_326891_);
             BlockState blockstate = p_331446_.getBlockState(this.belowPos);
             return blockstate.is(BlockTags.FROG_PREFER_JUMP_TO) ? PathType.OPEN : super.getPathType(p_331446_, p_326799_, p_326899_, p_326891_);
         }
