--- a/net/minecraft/world/level/levelgen/placement/PlacementContext.java
+++ b/net/minecraft/world/level/levelgen/placement/PlacementContext.java
@@ -24,12 +_,12 @@
         this.topFeature = p_191820_;
     }
 
-    public int getHeight(Heightmap.Types p_191825_, int p_191826_, int p_191827_) {
+    public int getHeight(Heightmap.Types p_191825_, me.alphamode.mcbig.math.BigInteger p_191826_, me.alphamode.mcbig.math.BigInteger p_191827_) {
         return this.level.getHeight(p_191825_, p_191826_, p_191827_);
     }
 
     public CarvingMask getCarvingMask(ChunkPos p_191822_, GenerationStep.Carving p_191823_) {
-        return ((ProtoChunk)this.level.getChunk(p_191822_.x, p_191822_.z)).getOrCreateCarvingMask(p_191823_);
+        return ((ProtoChunk)this.level.getChunk(p_191822_.x(), p_191822_.z())).getOrCreateCarvingMask(p_191823_);
     }
 
     public BlockState getBlockState(BlockPos p_191829_) {
