--- a/net/minecraft/world/level/levelgen/placement/SurfaceWaterDepthFilter.java
+++ b/net/minecraft/world/level/levelgen/placement/SurfaceWaterDepthFilter.java
@@ -24,8 +_,8 @@
 
     @Override
     protected boolean shouldPlace(PlacementContext p_226411_, RandomSource p_226412_, BlockPos p_226413_) {
-        int i = p_226411_.getHeight(Heightmap.Types.OCEAN_FLOOR, p_226413_.getX(), p_226413_.getZ());
-        int j = p_226411_.getHeight(Heightmap.Types.WORLD_SURFACE, p_226413_.getX(), p_226413_.getZ());
+        int i = p_226411_.getHeight(Heightmap.Types.OCEAN_FLOOR, p_226413_.getBigX(), p_226413_.getBigZ());
+        int j = p_226411_.getHeight(Heightmap.Types.WORLD_SURFACE, p_226413_.getBigX(), p_226413_.getBigZ());
         return j - i <= this.maxWaterDepth;
     }
 
