--- a/net/minecraft/world/entity/npc/WanderingTraderSpawner.java
+++ b/net/minecraft/world/entity/npc/WanderingTraderSpawner.java
@@ -136,8 +_,8 @@
         BlockPos blockpos = null;
 
         for(int i = 0; i < 10; ++i) {
-            int j = p_35930_.getX() + this.random.nextInt(p_35931_ * 2) - p_35931_;
-            int k = p_35930_.getZ() + this.random.nextInt(p_35931_ * 2) - p_35931_;
+            me.alphamode.mcbig.math.BigInteger j = p_35930_.getBigX().add(this.random.nextInt(p_35931_ * 2) - p_35931_);
+            me.alphamode.mcbig.math.BigInteger k = p_35930_.getBigZ().add(this.random.nextInt(p_35931_ * 2) - p_35931_);
             int l = p_35929_.getHeight(Heightmap.Types.WORLD_SURFACE, j, k);
             BlockPos blockpos1 = new BlockPos(j, l, k);
             if (NaturalSpawner.isSpawnPositionOk(SpawnPlacements.Type.ON_GROUND, p_35929_, blockpos1, EntityType.WANDERING_TRADER)) {
