--- a/net/minecraft/world/entity/vehicle/OldMinecartBehavior.java
+++ b/net/minecraft/world/entity/vehicle/OldMinecartBehavior.java
@@ -98,7 +_,7 @@
             if ($$4) {
                 this.moveAlongTrack(serverlevel);
                 if (blockstate.is(Blocks.ACTIVATOR_RAIL)) {
-                    this.minecart.activateMinecart(blockpos.getX(), blockpos.getY(), blockpos.getZ(), blockstate.getValue(PoweredRailBlock.POWERED));
+                    this.minecart.activateMinecart(blockpos.getBigX(), blockpos.getBigY(), blockpos.getBigZ(), blockstate.getValue(PoweredRailBlock.POWERED));
                 }
             } else {
                 this.minecart.comeOffTrack(serverlevel);
