--- a/net/minecraft/server/level/TicketType.java
+++ b/net/minecraft/server/level/TicketType.java
@@ -12,12 +_,12 @@
     private final long timeout;
     public static final TicketType<Unit> START = create("start", (p_9471_, p_9472_) -> 0);
     public static final TicketType<Unit> DRAGON = create("dragon", (p_9460_, p_9461_) -> 0);
-    public static final TicketType<ChunkPos> PLAYER = create("player", Comparator.comparingLong(ChunkPos::toLong));
-    public static final TicketType<ChunkPos> FORCED = create("forced", Comparator.comparingLong(ChunkPos::toLong));
-    public static final TicketType<ChunkPos> LIGHT = create("light", Comparator.comparingLong(ChunkPos::toLong));
-    public static final TicketType<BlockPos> PORTAL = create("portal", Vec3i::compareTo, 300);
+    public static final TicketType<ChunkPos> PLAYER = create("player", ChunkPos::compareTo);
+    public static final TicketType<ChunkPos> FORCED = create("forced", ChunkPos::compareTo);
+    public static final TicketType<ChunkPos> LIGHT = create("light", ChunkPos::compareTo);
+    public static final TicketType<BlockPos> PORTAL = create("portal", me.alphamode.mcbig.core.Vec3l::compareTo, 300);
     public static final TicketType<Integer> POST_TELEPORT = create("post_teleport", Integer::compareTo, 5);
-    public static final TicketType<ChunkPos> UNKNOWN = create("unknown", Comparator.comparingLong(ChunkPos::toLong), 1);
+    public static final TicketType<ChunkPos> UNKNOWN = create("unknown", ChunkPos::compareTo, 1);
 
     public static <T> TicketType<T> create(String p_9463_, Comparator<T> p_9464_) {
         return new TicketType<>(p_9463_, p_9464_, 0L);
