--- a/net/minecraft/commands/arguments/coordinates/ColumnPosArgument.java
+++ b/net/minecraft/commands/arguments/coordinates/ColumnPosArgument.java
@@ -28,7 +_,7 @@
 
     public static ColumnPos getColumnPos(CommandContext<CommandSourceStack> p_118993_, String p_118994_) {
         BlockPos blockpos = p_118993_.getArgument(p_118994_, Coordinates.class).getBlockPos(p_118993_.getSource());
-        return new ColumnPos(blockpos.getX(), blockpos.getZ());
+        return new ColumnPos(blockpos.getBigX(), blockpos.getBigZ());
     }
 
     public Coordinates parse(StringReader p_118991_) throws CommandSyntaxException {
@@ -40,7 +_,7 @@
             if (p_118991_.canRead() && p_118991_.peek() == ' ') {
                 p_118991_.skip();
                 WorldCoordinate worldcoordinate1 = WorldCoordinate.parseInt(p_118991_);
-                return new WorldCoordinates(worldcoordinate, new WorldCoordinate(true, 0.0), worldcoordinate1);
+                return new WorldCoordinates(worldcoordinate, new WorldCoordinate(true, me.alphamode.mcbig.math.BigDecimal.ZERO), worldcoordinate1);
             } else {
                 p_118991_.setCursor(i);
                 throw ERROR_NOT_COMPLETE.createWithContext(p_118991_);
