--- a/net/minecraft/client/renderer/debug/GameEventListenerRenderer.java
+++ b/net/minecraft/client/renderer/debug/GameEventListenerRenderer.java
@@ -148,7 +_,7 @@
     ) {
         Camera camera = Minecraft.getInstance().gameRenderer.getMainCamera();
         if (camera.isInitialized()) {
-            Vec3 vec3 = camera.getPosition().reverse();
+            Vec3 vec3 = camera.getPosition().reverse().toVanilla();
             DebugRenderer.renderFilledBox(p_270351_, p_270763_, p_270205_.move(vec3), p_270707_, p_270538_, p_270314_, p_270966_);
         }
     }
@@ -183,7 +_,7 @@
         }
 
         public Optional<Vec3> getPosition(Level p_173876_) {
-            return this.listenerSource.getPosition(p_173876_);
+            return this.listenerSource.getPosition(p_173876_).map(me.alphamode.mcbig.core.BigVec3::toVanilla);
         }
 
         @Override
@@ -197,7 +_,7 @@
         }
 
         @Override
-        public boolean handleGameEvent(ServerLevel p_234540_, Holder<GameEvent> p_316503_, GameEvent.Context p_250285_, Vec3 p_250758_) {
+        public boolean handleGameEvent(ServerLevel p_234540_, Holder<GameEvent> p_316503_, GameEvent.Context p_250285_, me.alphamode.mcbig.core.BigVec3 p_250758_) {
             return false;
         }
     }
