--- a/net/minecraft/client/renderer/debug/OctreeDebugRenderer.java
+++ b/net/minecraft/client/renderer/debug/OctreeDebugRenderer.java
@@ -46,14 +_,14 @@
         MutableInt p_366741_,
         boolean p_371203_
     ) {
-        AABB aabb = p_366527_.getAABB();
+        me.alphamode.mcbig.phys.BigAABB aabb = p_366527_.getAABB();
         double d0 = aabb.getXsize();
         long i = Math.round(d0 / 16.0);
         if (i == 1L) {
             p_366741_.add(1);
-            double d1 = aabb.getCenter().x;
+            double d1 = aabb.getCenter().x.doubleValue();
             double d2 = aabb.getCenter().y;
-            double d3 = aabb.getCenter().z;
+            double d3 = aabb.getCenter().z.doubleValue();
             int k = p_371203_ ? -16711936 : -1;
             DebugRenderer.renderFloatingText(p_366585_, p_366909_, String.valueOf(p_366741_.getValue()), d1, d2, d3, k, 0.3F);
         }
@@ -63,7 +_,7 @@
         ShapeRenderer.renderLineBox(
             p_366585_,
             vertexconsumer,
-            aabb.deflate(0.1 * (double)p_366591_).move(-p_366814_, -p_366595_, -p_366414_),
+            aabb.deflate(0.1 * (double)p_366591_).move(-p_366814_, -p_366595_, -p_366414_).toVanilla(),
             getColorComponent(j, 0.3F),
             getColorComponent(j, 0.8F),
             getColorComponent(j, 0.5F),
