--- a/net/minecraft/world/level/levelgen/structure/SinglePieceStructure.java
+++ b/net/minecraft/world/level/levelgen/structure/SinglePieceStructure.java
@@ -32,6 +_,6 @@
 
     @FunctionalInterface
     protected interface PieceConstructor {
-        StructurePiece construct(WorldgenRandom p_226550_, int p_226551_, int p_226552_);
+        StructurePiece construct(WorldgenRandom p_226550_, me.alphamode.mcbig.math.BigInteger p_226551_, me.alphamode.mcbig.math.BigInteger p_226552_);
     }
 }
