--- a/net/minecraft/world/level/levelgen/structure/StructureStart.java
+++ b/net/minecraft/world/level/levelgen/structure/StructureStart.java
@@ -49,7 +_,7 @@
                 LOGGER.error("Unknown stucture id: {}", s);
                 return null;
             } else {
-                ChunkPos chunkpos = new ChunkPos(p_226859_.getInt("ChunkX"), p_226859_.getInt("ChunkZ"));
+                ChunkPos chunkpos = new ChunkPos(p_226859_.getByteArray("ChunkX"), p_226859_.getByteArray("ChunkZ"));
                 int i = p_226859_.getInt("references");
                 ListTag listtag = p_226859_.getList("Children", 10);
 
@@ -85,7 +_,7 @@
         if (!list.isEmpty()) {
             BoundingBox boundingbox = list.get(0).boundingBox;
             BlockPos blockpos = boundingbox.getCenter();
-            BlockPos blockpos1 = new BlockPos(blockpos.getX(), boundingbox.minY(), blockpos.getZ());
+            BlockPos blockpos1 = new BlockPos(blockpos.getBigX(), boundingbox.minY(), blockpos.getBigZ());
 
             for(StructurePiece structurepiece : list) {
                 if (structurepiece.getBoundingBox().intersects(p_226855_)) {
@@ -101,8 +_,8 @@
         CompoundTag compoundtag = new CompoundTag();
         if (this.isValid()) {
             compoundtag.putString("id", p_192661_.registryAccess().registryOrThrow(Registries.STRUCTURE).getKey(this.structure).toString());
-            compoundtag.putInt("ChunkX", p_192662_.x);
-            compoundtag.putInt("ChunkZ", p_192662_.z);
+            compoundtag.putByteArray("ChunkX", p_192662_.x().toByteArray());
+            compoundtag.putByteArray("ChunkZ", p_192662_.z().toByteArray());
             compoundtag.putInt("references", this.references);
             compoundtag.put("Children", this.pieceContainer.save(p_192661_));
             return compoundtag;
