--- a/net/minecraft/world/level/levelgen/structure/Structure.java
+++ b/net/minecraft/world/level/levelgen/structure/Structure.java
@@ -107,8 +_,8 @@
         Structure.GenerationContext p_226586_, Heightmap.Types p_226587_, Consumer<StructurePiecesBuilder> p_226588_
     ) {
         ChunkPos chunkpos = p_226586_.chunkPos();
-        int i = chunkpos.getMiddleBlockX();
-        int j = chunkpos.getMiddleBlockZ();
+        me.alphamode.mcbig.math.BigInteger i = chunkpos.getMiddleBlockX();
+        me.alphamode.mcbig.math.BigInteger j = chunkpos.getMiddleBlockZ();
         int k = p_226586_.chunkGenerator().getFirstOccupiedHeight(i, j, p_226587_, p_226586_.heightAccessor(), p_226586_.randomState());
         return Optional.of(new Structure.GenerationStub(new BlockPos(i, k, j), p_226588_));
     }
@@ -120,9 +_,9 @@
                 p_263005_.chunkGenerator
                     .getBiomeSource()
                     .getNoiseBiome(
-                        QuartPos.fromBlock(blockpos.getX()),
-                        QuartPos.fromBlock(blockpos.getY()),
-                        QuartPos.fromBlock(blockpos.getZ()),
+                        QuartPos.fromBlock(blockpos.getBigX()),
+                        QuartPos.fromBlock(blockpos.getBigY()),
+                        QuartPos.fromBlock(blockpos.getBigZ()),
                         p_263005_.randomState.sampler()
                     )
             );
@@ -139,28 +_,28 @@
     ) {
     }
 
-    private static int[] getCornerHeights(Structure.GenerationContext p_226614_, int p_226615_, int p_226616_, int p_226617_, int p_226618_) {
+    private static int[] getCornerHeights(Structure.GenerationContext p_226614_, me.alphamode.mcbig.math.BigInteger p_226615_, int p_226616_, me.alphamode.mcbig.math.BigInteger p_226617_, int p_226618_) {
         ChunkGenerator chunkgenerator = p_226614_.chunkGenerator();
         LevelHeightAccessor levelheightaccessor = p_226614_.heightAccessor();
         RandomState randomstate = p_226614_.randomState();
         return new int[]{
             chunkgenerator.getFirstOccupiedHeight(p_226615_, p_226617_, Heightmap.Types.WORLD_SURFACE_WG, levelheightaccessor, randomstate),
-            chunkgenerator.getFirstOccupiedHeight(p_226615_, p_226617_ + p_226618_, Heightmap.Types.WORLD_SURFACE_WG, levelheightaccessor, randomstate),
-            chunkgenerator.getFirstOccupiedHeight(p_226615_ + p_226616_, p_226617_, Heightmap.Types.WORLD_SURFACE_WG, levelheightaccessor, randomstate),
+            chunkgenerator.getFirstOccupiedHeight(p_226615_, p_226617_.add(p_226618_), Heightmap.Types.WORLD_SURFACE_WG, levelheightaccessor, randomstate),
+            chunkgenerator.getFirstOccupiedHeight(p_226615_.add(p_226616_), p_226617_, Heightmap.Types.WORLD_SURFACE_WG, levelheightaccessor, randomstate),
             chunkgenerator.getFirstOccupiedHeight(
-                p_226615_ + p_226616_, p_226617_ + p_226618_, Heightmap.Types.WORLD_SURFACE_WG, levelheightaccessor, randomstate
+                p_226615_.add(p_226616_), p_226617_.add(p_226618_), Heightmap.Types.WORLD_SURFACE_WG, levelheightaccessor, randomstate
             )
         };
     }
 
     protected static int getLowestY(Structure.GenerationContext p_226573_, int p_226574_, int p_226575_) {
         ChunkPos chunkpos = p_226573_.chunkPos();
-        int i = chunkpos.getMinBlockX();
-        int j = chunkpos.getMinBlockZ();
+        me.alphamode.mcbig.math.BigInteger i = chunkpos.getBigMinBlockX();
+        me.alphamode.mcbig.math.BigInteger j = chunkpos.getBigMinBlockZ();
         return getLowestY(p_226573_, i, j, p_226574_, p_226575_);
     }
 
-    protected static int getLowestY(Structure.GenerationContext p_226577_, int p_226578_, int p_226579_, int p_226580_, int p_226581_) {
+    protected static int getLowestY(Structure.GenerationContext p_226577_, me.alphamode.mcbig.math.BigInteger p_226578_, me.alphamode.mcbig.math.BigInteger p_226579_, int p_226580_, int p_226581_) {
         int[] aint = getCornerHeights(p_226577_, p_226578_, p_226580_, p_226579_, p_226581_);
         return Math.min(Math.min(aint[0], aint[1]), Math.min(aint[2], aint[3]));
     }
@@ -179,8 +_,8 @@
         }
 
         ChunkPos chunkpos = p_226583_.chunkPos();
-        int k = chunkpos.getBlockX(7);
-        int l = chunkpos.getBlockZ(7);
+        me.alphamode.mcbig.math.BigInteger k = chunkpos.getBlockX(7);
+        me.alphamode.mcbig.math.BigInteger l = chunkpos.getBlockZ(7);
         return new BlockPos(k, getLowestY(p_226583_, k, l, i, j), l);
     }
 
@@ -220,7 +_,7 @@
 
         private static WorldgenRandom makeRandom(long p_226654_, ChunkPos p_226655_) {
             WorldgenRandom worldgenrandom = new WorldgenRandom(new LegacyRandomSource(0L));
-            worldgenrandom.setLargeFeatureSeed(p_226654_, p_226655_.x, p_226655_.z);
+            worldgenrandom.setLargeFeatureSeed(p_226654_, p_226655_.x().longValue(), p_226655_.z().longValue());
             return worldgenrandom;
         }
     }
