--- a/net/minecraft/world/level/portal/PortalForcer.java
+++ b/net/minecraft/world/level/portal/PortalForcer.java
@@ -80,7 +_,7 @@
         BlockPos.MutableBlockPos blockpos$mutableblockpos = p_77667_.mutable();
 
         for(BlockPos.MutableBlockPos blockpos$mutableblockpos1 : BlockPos.spiralAround(p_77667_, 16, Direction.EAST, Direction.SOUTH)) {
-            int j = Math.min(i, this.level.getHeight(Heightmap.Types.MOTION_BLOCKING, blockpos$mutableblockpos1.getX(), blockpos$mutableblockpos1.getZ()));
+            int j = Math.min(i, this.level.getHeight(Heightmap.Types.MOTION_BLOCKING, blockpos$mutableblockpos1.getBigX(), blockpos$mutableblockpos1.getBigZ()));
             int k = 1;
             if (worldborder.isWithinBounds(blockpos$mutableblockpos1) && worldborder.isWithinBounds(blockpos$mutableblockpos1.move(direction, 1))) {
                 blockpos$mutableblockpos1.move(direction.getOpposite(), 1);
@@ -131,7 +_,7 @@
                 return Optional.empty();
             }
 
-            blockpos = new BlockPos(p_77667_.getX(), Mth.clamp(p_77667_.getY(), k1, i2), p_77667_.getZ()).immutable();
+            blockpos = new BlockPos(p_77667_.getBigX(), Mth.clamp(p_77667_.getY(), k1, i2), p_77667_.getBigZ()).immutable();
             Direction direction1 = direction.getClockWise();
             if (!worldborder.isWithinBounds(blockpos)) {
                 return Optional.empty();
