--- a/net/minecraft/world/level/chunk/StructureAccess.java
+++ b/net/minecraft/world/level/chunk/StructureAccess.java
@@ -3,6 +_,8 @@
 import it.unimi.dsi.fastutil.longs.LongSet;
 import java.util.Map;
 import javax.annotation.Nullable;
+
+import net.minecraft.world.level.ChunkPos;
 import net.minecraft.world.level.levelgen.structure.Structure;
 import net.minecraft.world.level.levelgen.structure.StructureStart;
 
@@ -12,11 +_,11 @@
 
     void setStartForStructure(Structure p_223437_, StructureStart p_223438_);
 
-    LongSet getReferencesForStructure(Structure p_223439_);
-
-    void addReferenceForStructure(Structure p_223435_, long p_223436_);
-
-    Map<Structure, LongSet> getAllReferences();
-
-    void setAllReferences(Map<Structure, LongSet> p_223440_);
+    it.unimi.dsi.fastutil.objects.ObjectSet<ChunkPos> getReferencesForStructure(Structure p_223439_);
+
+    void addReferenceForStructure(Structure p_223435_, ChunkPos p_223436_);
+
+    Map<Structure, it.unimi.dsi.fastutil.objects.ObjectSet<ChunkPos>> getAllReferences();
+
+    void setAllReferences(Map<Structure, it.unimi.dsi.fastutil.objects.ObjectSet<ChunkPos>> p_223440_);
 }
