--- a/net/minecraft/world/level/block/CampfireBlock.java
+++ b/net/minecraft/world/level/block/CampfireBlock.java
@@ -177,9 +_,9 @@
         if (p_220918_.getValue(LIT)) {
             if (p_220921_.nextInt(10) == 0) {
                 p_220919_.playLocalSound(
-                    (double)p_220920_.getX() + 0.5,
+                    (double)p_220920_.getBigX().doubleValue() + 0.5,
                     (double)p_220920_.getY() + 0.5,
-                    (double)p_220920_.getZ() + 0.5,
+                    (double)p_220920_.getBigZ().doubleValue() + 0.5,
                     SoundEvents.CAMPFIRE_CRACKLE,
                     SoundSource.BLOCKS,
                     0.5F + p_220921_.nextFloat(),
@@ -192,9 +_,9 @@
                 for(int i = 0; i < p_220921_.nextInt(1) + 1; ++i) {
                     p_220919_.addParticle(
                         ParticleTypes.LAVA,
-                        (double)p_220920_.getX() + 0.5,
+                        (double)p_220920_.getBigX().doubleValue() + 0.5,
                         (double)p_220920_.getY() + 0.5,
-                        (double)p_220920_.getZ() + 0.5,
+                        (double)p_220920_.getBigZ().doubleValue() + 0.5,
                         (double)(p_220921_.nextFloat() / 2.0F),
                         5.0E-5,
                         (double)(p_220921_.nextFloat() / 2.0F)
