--- a/net/minecraft/world/level/block/ChiseledBookShelfBlock.java
+++ b/net/minecraft/world/level/block/ChiseledBookShelfBlock.java
@@ -124,7 +_,7 @@
             return Optional.empty();
         } else {
             BlockPos blockpos = p_261714_.getBlockPos().relative(direction);
-            Vec3 vec3 = p_261714_.getLocation().subtract((double)blockpos.getX(), (double)blockpos.getY(), (double)blockpos.getZ());
+            me.alphamode.mcbig.core.BigVec3 vec3 = p_261714_.getLocation().subtract((double)blockpos.getX(), (double)blockpos.getY(), (double)blockpos.getZ());
             double d0 = vec3.x();
             double d1 = vec3.y();
             double d2 = vec3.z();
