--- a/net/minecraft/world/level/block/TorchBlock.java
+++ b/net/minecraft/world/level/block/TorchBlock.java
@@ -43,9 +_,9 @@
 
     @Override
     public void animateTick(BlockState p_222593_, Level p_222594_, BlockPos p_222595_, RandomSource p_222596_) {
-        double d0 = (double)p_222595_.getX() + 0.5;
-        double d1 = (double)p_222595_.getY() + 0.7;
-        double d2 = (double)p_222595_.getZ() + 0.5;
+        double d0 = (double)p_222595_.getBigX().doubleValue() + 0.5;
+        double d1 = (double)p_222595_.getBigY().doubleValue() + 0.7;
+        double d2 = (double)p_222595_.getBigZ().doubleValue() + 0.5;
         p_222594_.addParticle(ParticleTypes.SMOKE, d0, d1, d2, 0.0, 0.0, 0.0);
         p_222594_.addParticle(this.flameParticle, d0, d1, d2, 0.0, 0.0, 0.0);
     }
