--- a/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -364,11 +_,11 @@
     }
 
     private static double clampHorizontal(double p_143610_) {
-        return Mth.clamp(p_143610_, -3.0E7, 3.0E7);
+        return /*Mth.clamp(*/p_143610_/*, -3.0E7, 3.0E7)*/;
     }
 
     private static double clampVertical(double p_143654_) {
-        return Mth.clamp(p_143654_, -2.0E7, 2.0E7);
+        return /*Mth.clamp(*/p_143654_/*, -2.0E7, 2.0E7)*/;
     }
 
     @Override
@@ -1094,7 +_,7 @@
                     int i = this.player.level().getMaxBuildHeight();
                     if (blockpos.getY() < i) {
                         if (this.awaitingPositionFromClient == null
-                            && this.player.distanceToSqr((double)blockpos.getX() + 0.5, (double)blockpos.getY() + 0.5, (double)blockpos.getZ() + 0.5) < 64.0
+                            && this.player.distanceToSqr((double)blockpos.getBigX().doubleValue() + 0.5, (double)blockpos.getBigY().doubleValue() + 0.5, (double)blockpos.getBigZ().doubleValue() + 0.5) < 64.0
                             && serverlevel.mayInteract(this.player, blockpos)) {
                             InteractionResult interactionresult = this.player
                                 .gameMode
