--- a/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -211,12 +_,12 @@
     private int ackBlockChangesUpTo = -1;
     private final TickThrottler chatSpamThrottler = new TickThrottler(20, 200);
     private final TickThrottler dropSpamThrottler = new TickThrottler(20, 1480);
-    private double firstGoodX;
+    private me.alphamode.mcbig.math.BigDecimal firstGoodX = me.alphamode.mcbig.math.BigDecimal.ZERO;
     private double firstGoodY;
-    private double firstGoodZ;
-    private double lastGoodX;
+    private me.alphamode.mcbig.math.BigDecimal firstGoodZ = me.alphamode.mcbig.math.BigDecimal.ZERO;
+    private me.alphamode.mcbig.math.BigDecimal lastGoodX = me.alphamode.mcbig.math.BigDecimal.ZERO;
     private double lastGoodY;
-    private double lastGoodZ;
+    private me.alphamode.mcbig.math.BigDecimal lastGoodZ = me.alphamode.mcbig.math.BigDecimal.ZERO;
     @Nullable
     private Entity lastVehicle;
     private double vehicleFirstGoodX;
@@ -226,7 +_,7 @@
     private double vehicleLastGoodY;
     private double vehicleLastGoodZ;
     @Nullable
-    private Vec3 awaitingPositionFromClient;
+    private me.alphamode.mcbig.core.BigVec3 awaitingPositionFromClient;
     private int awaitingTeleport;
     private int awaitingTeleportTime;
     private boolean clientIsFloating;
@@ -325,12 +_,12 @@
     }
 
     public void resetPosition() {
-        this.firstGoodX = this.player.getX();
+        this.firstGoodX = this.player.getBigX();
         this.firstGoodY = this.player.getY();
-        this.firstGoodZ = this.player.getZ();
-        this.lastGoodX = this.player.getX();
+        this.firstGoodZ = this.player.getBigZ();
+        this.lastGoodX = this.player.getBigX();
         this.lastGoodY = this.player.getY();
-        this.lastGoodZ = this.player.getZ();
+        this.lastGoodZ = this.player.getBigZ();
     }
 
     @Override
@@ -379,12 +_,12 @@
         return Double.isNaN(p_143664_) || Double.isNaN(p_143665_) || Double.isNaN(p_143666_) || !Floats.isFinite(p_143668_) || !Floats.isFinite(p_143667_);
     }
 
-    private static double clampHorizontal(double p_143610_) {
-        return Mth.clamp(p_143610_, -3.0E7, 3.0E7);
+    private static me.alphamode.mcbig.math.BigDecimal clampHorizontal(me.alphamode.mcbig.math.BigDecimal p_143610_) {
+        return /*Mth.clamp(*/p_143610_/*, -3.0E7, 3.0E7)*/;
     }
 
     private static double clampVertical(double p_143654_) {
-        return Mth.clamp(p_143654_, -2.0E7, 2.0E7);
+        return /*Mth.clamp(*/p_143654_/*, -2.0E7, 2.0E7)*/;
     }
 
     @Override
@@ -399,9 +_,9 @@
                 double d0 = entity.getX();
                 double d1 = entity.getY();
                 double d2 = entity.getZ();
-                double d3 = clampHorizontal(p_9876_.position().x());
+                double d3 = /*clampHorizontal(*/p_9876_.position().x()/*)*/;
                 double d4 = clampVertical(p_9876_.position().y());
-                double d5 = clampHorizontal(p_9876_.position().z());
+                double d5 = /*clampHorizontal(*/p_9876_.position().z()/*)*/;
                 float f = Mth.wrapDegrees(p_9876_.yRot());
                 float f1 = Mth.wrapDegrees(p_9876_.xRot());
                 double d6 = d3 - this.vehicleFirstGoodX;
@@ -900,7 +_,7 @@
     @Override
     public void handleMovePlayer(ServerboundMovePlayerPacket p_9874_) {
         PacketUtils.ensureRunningOnSameThread(p_9874_, this, this.player.serverLevel());
-        if (containsInvalidValues(p_9874_.getX(0.0), p_9874_.getY(0.0), p_9874_.getZ(0.0), p_9874_.getYRot(0.0F), p_9874_.getXRot(0.0F))) {
+        if (containsInvalidValues(p_9874_.getX(me.alphamode.mcbig.math.BigDecimal.ZERO).doubleValue(), p_9874_.getY(0.0), p_9874_.getZ(me.alphamode.mcbig.math.BigDecimal.ZERO).doubleValue(), p_9874_.getYRot(0.0F), p_9874_.getXRot(0.0F))) {
             this.disconnect(Component.translatable("multiplayer.disconnect.invalid_player_movement"));
         } else {
             ServerLevel serverlevel = this.player.serverLevel();
@@ -910,26 +_,26 @@
                 }
 
                 if (!this.updateAwaitingTeleport() && this.player.hasClientLoaded()) {
-                    double d0 = clampHorizontal(p_9874_.getX(this.player.getX()));
+                    me.alphamode.mcbig.math.BigDecimal d0 = clampHorizontal(p_9874_.getX(this.player.getBigX()));
                     double d1 = clampVertical(p_9874_.getY(this.player.getY()));
-                    double d2 = clampHorizontal(p_9874_.getZ(this.player.getZ()));
+                    me.alphamode.mcbig.math.BigDecimal d2 = clampHorizontal(p_9874_.getZ(this.player.getBigZ()));
                     float f = Mth.wrapDegrees(p_9874_.getYRot(this.player.getYRot()));
                     float f1 = Mth.wrapDegrees(p_9874_.getXRot(this.player.getXRot()));
                     if (this.player.isPassenger()) {
                         this.player.absMoveTo(this.player.getX(), this.player.getY(), this.player.getZ(), f, f1);
                         this.player.serverLevel().getChunkSource().move(this.player);
                     } else {
-                        double d3 = this.player.getX();
+                        me.alphamode.mcbig.math.BigDecimal d3 = this.player.getBigX();
                         double d4 = this.player.getY();
-                        double d5 = this.player.getZ();
-                        double d6 = d0 - this.firstGoodX;
+                        me.alphamode.mcbig.math.BigDecimal d5 = this.player.getBigZ();
+                        double d6 = d0.subtract(this.firstGoodX).doubleValue();
                         double d7 = d1 - this.firstGoodY;
-                        double d8 = d2 - this.firstGoodZ;
+                        double d8 = d2.subtract(this.firstGoodZ).doubleValue();
                         double d9 = this.player.getDeltaMovement().lengthSqr();
                         double d10 = d6 * d6 + d7 * d7 + d8 * d8;
                         if (this.player.isSleeping()) {
                             if (d10 > 1.0) {
-                                this.teleport(this.player.getX(), this.player.getY(), this.player.getZ(), f, f1);
+                                this.teleport(this.player.getBigX(), this.player.getY(), this.player.getBigZ(), f, f1);
                             }
                         } else {
                             boolean flag = this.player.isFallFlying();
@@ -945,16 +_,16 @@
                                     float f2 = flag ? 300.0F : 100.0F;
                                     if (d10 - d9 > (double)(f2 * (float)i)) {
                                         LOGGER.warn("{} moved too quickly! {},{},{}", this.player.getName().getString(), d6, d7, d8);
-                                        this.teleport(this.player.getX(), this.player.getY(), this.player.getZ(), this.player.getYRot(), this.player.getXRot());
+                                        this.teleport(this.player.getBigX(), this.player.getY(), this.player.getBigZ(), this.player.getYRot(), this.player.getXRot());
                                         return;
                                     }
                                 }
                             }
 
                             AABB aabb = this.player.getBoundingBox();
-                            d6 = d0 - this.lastGoodX;
+                            d6 = d0.subtract(this.lastGoodX).doubleValue();
                             d7 = d1 - this.lastGoodY;
-                            d8 = d2 - this.lastGoodZ;
+                            d8 = d2.subtract(this.lastGoodZ).doubleValue();
                             boolean flag4 = d7 > 0.0;
                             if (this.player.onGround() && !p_9874_.isOnGround() && flag4) {
                                 this.player.jumpFromGround();
@@ -962,13 +_,13 @@
 
                             boolean flag1 = this.player.verticalCollisionBelow;
                             this.player.move(MoverType.PLAYER, new Vec3(d6, d7, d8));
-                            d6 = d0 - this.player.getX();
+                            d6 = d0.subtract(this.player.getBigX()).doubleValue();
                             d7 = d1 - this.player.getY();
                             if (d7 > -0.5 || d7 < 0.5) {
                                 d7 = 0.0;
                             }
 
-                            d8 = d2 - this.player.getZ();
+                            d8 = d2.subtract(this.player.getBigZ()).doubleValue();
                             d10 = d6 * d6 + d7 * d7 + d8 * d8;
                             boolean flag2 = false;
                             if (!this.player.isChangingDimension()
@@ -996,10 +_,10 @@
                                     && !flag3
                                     && this.noBlocksAround(this.player);
                                 this.player.serverLevel().getChunkSource().move(this.player);
-                                Vec3 vec3 = new Vec3(this.player.getX() - d3, this.player.getY() - d4, this.player.getZ() - d5);
+                                Vec3 vec3 = new Vec3(this.player.getBigX().subtract(d3).doubleValue(), this.player.getY() - d4, this.player.getBigZ().subtract(d5).doubleValue());
                                 this.player.setOnGroundWithMovement(p_9874_.isOnGround(), p_9874_.horizontalCollision(), vec3);
                                 this.player.doCheckFallDamage(vec3.x, vec3.y, vec3.z, p_9874_.isOnGround());
-                                this.player.recordMovementThroughBlocks(new Vec3(d3, d4, d5), this.player.position());
+                                this.player.recordMovementThroughBlocks(new Vec3(d3.doubleValue(), d4, d5.doubleValue()), this.player.position());
                                 this.handlePlayerKnownMovement(vec3);
                                 if (flag4) {
                                     this.player.resetFallDistance();
@@ -1014,13 +_,13 @@
                                     this.player.tryResetCurrentImpulseContext();
                                 }
 
-                                this.player.checkMovementStatistics(this.player.getX() - d3, this.player.getY() - d4, this.player.getZ() - d5);
-                                this.lastGoodX = this.player.getX();
+                                this.player.checkMovementStatistics(this.player.getX() - d3.doubleValue(), this.player.getY() - d4, this.player.getZ() - d5.doubleValue());
+                                this.lastGoodX = this.player.getBigX();
                                 this.lastGoodY = this.player.getY();
-                                this.lastGoodZ = this.player.getZ();
+                                this.lastGoodZ = this.player.getBigZ();
                             } else {
                                 this.teleport(d3, d4, d5, f, f1);
-                                this.player.doCheckFallDamage(this.player.getX() - d3, this.player.getY() - d4, this.player.getZ() - d5, p_9874_.isOnGround());
+                                this.player.doCheckFallDamage(this.player.getX() - d3.doubleValue(), this.player.getY() - d4, this.player.getZ() - d5.doubleValue(), p_9874_.isOnGround());
                             }
                         }
                     }
@@ -1062,8 +_,8 @@
         }
     }
 
-    private boolean isPlayerCollidingWithAnythingNew(LevelReader p_289008_, AABB p_288986_, double p_288990_, double p_288991_, double p_288967_) {
-        AABB aabb = this.player.getBoundingBox().move(p_288990_ - this.player.getX(), p_288991_ - this.player.getY(), p_288967_ - this.player.getZ());
+    private boolean isPlayerCollidingWithAnythingNew(LevelReader p_289008_, AABB p_288986_, me.alphamode.mcbig.math.BigDecimal p_288990_, double p_288991_, me.alphamode.mcbig.math.BigDecimal p_288967_) {
+        AABB aabb = this.player.getBoundingBox().move(p_288990_.subtract(this.player.getBigX()).doubleValue(), p_288991_ - this.player.getY(), p_288967_.subtract(this.player.getBigZ()).doubleValue());
         Iterable<VoxelShape> iterable = p_289008_.getCollisions(this.player, aabb.deflate(1.0E-5F));
         VoxelShape voxelshape = Shapes.create(p_288986_.deflate(1.0E-5F));
 
@@ -1076,8 +_,8 @@
         return false;
     }
 
-    public void teleport(double p_9775_, double p_9776_, double p_9777_, float p_9778_, float p_9779_) {
-        this.teleport(new PositionMoveRotation(new Vec3(p_9775_, p_9776_, p_9777_), Vec3.ZERO, p_9778_, p_9779_), Collections.emptySet());
+    public void teleport(me.alphamode.mcbig.math.BigDecimal p_9775_, double p_9776_, me.alphamode.mcbig.math.BigDecimal p_9777_, float p_9778_, float p_9779_) {
+        this.teleport(new PositionMoveRotation(new me.alphamode.mcbig.core.BigVec3(p_9775_, p_9776_, p_9777_), Vec3.ZERO, p_9778_, p_9779_), Collections.emptySet());
     }
 
     public void teleport(PositionMoveRotation p_371571_, Set<Relative> p_9786_) {
@@ -1087,7 +_,7 @@
         }
 
         this.player.teleportSetPosition(p_371571_, p_9786_);
-        this.awaitingPositionFromClient = this.player.position();
+        this.awaitingPositionFromClient = this.player.bigPosition();
         this.player.connection.send(ClientboundPlayerPositionPacket.of(this.awaitingTeleport, p_371571_, p_9786_));
     }
 
@@ -1158,10 +_,10 @@
             ItemStack itemstack = this.player.getItemInHand(interactionhand);
             if (itemstack.isItemEnabled(serverlevel.enabledFeatures())) {
                 BlockHitResult blockhitresult = p_9930_.getHitResult();
-                Vec3 vec3 = blockhitresult.getLocation();
+                me.alphamode.mcbig.core.BigVec3 vec3 = blockhitresult.getLocation();
                 BlockPos blockpos = blockhitresult.getBlockPos();
                 if (this.player.canInteractWithBlock(blockpos, 1.0)) {
-                    Vec3 vec31 = vec3.subtract(Vec3.atCenterOf(blockpos));
+                    me.alphamode.mcbig.core.BigVec3 vec31 = vec3.subtract(me.alphamode.mcbig.core.BigVec3.atCenterOf(blockpos));
                     double d0 = 1.0000001;
                     if (Math.abs(vec31.x()) < 1.0000001 && Math.abs(vec31.y()) < 1.0000001 && Math.abs(vec31.z()) < 1.0000001) {
                         Direction direction = blockhitresult.getDirection();
@@ -1238,7 +_,7 @@
             for (ServerLevel serverlevel : this.server.getAllLevels()) {
                 Entity entity = p_9928_.getEntity(serverlevel);
                 if (entity != null) {
-                    this.player.teleportTo(serverlevel, entity.getX(), entity.getY(), entity.getZ(), Set.of(), entity.getYRot(), entity.getXRot(), true);
+                    this.player.teleportTo(serverlevel, entity.getBigX(), entity.getY(), entity.getBigZ(), Set.of(), entity.getYRot(), entity.getXRot(), true);
                     return;
                 }
             }
@@ -1515,7 +_,7 @@
                 case STOP_SLEEPING:
                     if (this.player.isSleeping()) {
                         this.player.stopSleepInBed(false, true);
-                        this.awaitingPositionFromClient = this.player.position();
+                        this.awaitingPositionFromClient = this.player.bigPosition();
                     }
                     break;
                 case START_RIDING_JUMP:
