--- a/net/minecraft/world/entity/boss/enderdragon/phases/DragonStrafePlayerPhase.java
+++ b/net/minecraft/world/entity/boss/enderdragon/phases/DragonStrafePlayerPhase.java
@@ -134,7 +_,7 @@
 
     private void navigateToNextPathNode() {
         if (this.currentPath != null && !this.currentPath.isDone()) {
-            Vec3i vec3i = this.currentPath.getNextNodePos();
+            me.alphamode.mcbig.core.Vec3l vec3i = this.currentPath.getNextNodePos();
             this.currentPath.advance();
             double d0 = (double)vec3i.getX();
             double d2 = (double)vec3i.getZ();
