--- a/net/minecraft/world/entity/boss/enderdragon/phases/DragonTakeoffPhase.java
+++ b/net/minecraft/world/entity/boss/enderdragon/phases/DragonTakeoffPhase.java
@@ -65,7 +_,7 @@
         if (this.currentPath != null) {
             this.currentPath.advance();
             if (!this.currentPath.isDone()) {
-                Vec3i vec3i = this.currentPath.getNextNodePos();
+                me.alphamode.mcbig.core.Vec3l vec3i = this.currentPath.getNextNodePos();
                 this.currentPath.advance();
 
                 double d0;
