--- a/net/minecraft/world/item/context/BlockPlaceContext.java
+++ b/net/minecraft/world/item/context/BlockPlaceContext.java
@@ -35,10 +_,10 @@
             p_43645_.getHand(),
             p_43645_.getItemInHand(),
             new BlockHitResult(
-                new Vec3(
-                    (double)p_43646_.getX() + 0.5 + (double)p_43647_.getStepX() * 0.5,
-                    (double)p_43646_.getY() + 0.5 + (double)p_43647_.getStepY() * 0.5,
-                    (double)p_43646_.getZ() + 0.5 + (double)p_43647_.getStepZ() * 0.5
+                new me.alphamode.mcbig.core.BigVec3(
+                    p_43646_.getBigX().toBigDecimal().add(me.alphamode.mcbig.core.BigConstants.AABB).add(me.alphamode.mcbig.math.BigDecimal.val(p_43647_.getStepX()).multiply(me.alphamode.mcbig.core.BigConstants.AABB)),
+                        (double)p_43646_.getY() + 0.5 + (double)p_43647_.getStepY() * 0.5,
+                    p_43646_.getBigZ().toBigDecimal().add(me.alphamode.mcbig.core.BigConstants.AABB).add(me.alphamode.mcbig.math.BigDecimal.val(p_43647_.getStepZ()).multiply(me.alphamode.mcbig.core.BigConstants.AABB))
                 ),
                 p_43647_,
                 p_43646_,
