--- a/net/minecraft/world/level/levelgen/feature/stateproviders/DualNoiseProvider.java
+++ b/net/minecraft/world/level/levelgen/feature/stateproviders/DualNoiseProvider.java
@@ -67,9 +_,9 @@
     protected double getSlowNoiseValue(BlockPos p_191407_) {
         return this.slowNoise
             .getValue(
-                (double)((float)p_191407_.getX() * this.slowScale),
-                (double)((float)p_191407_.getY() * this.slowScale),
-                (double)((float)p_191407_.getZ() * this.slowScale)
+                (double)((float)p_191407_.getBigX().floatValue() * this.slowScale),
+                (double)((float)p_191407_.getBigY().floatValue() * this.slowScale),
+                (double)((float)p_191407_.getBigZ().floatValue() * this.slowScale)
             );
     }
 }
