--- a/net/minecraft/client/particle/ParticleEngine.java
+++ b/net/minecraft/client/particle/ParticleEngine.java
@@ -503,9 +_,9 @@
                                 this.add(
                                     new TerrainParticle(
                                         this.level,
-                                        (double)p_107356_.getX() + d7,
-                                        (double)p_107356_.getY() + d8,
-                                        (double)p_107356_.getZ() + d9,
+                                        (double)p_107356_.getBigX().doubleValue() + d7,
+                                        (double)p_107356_.getBigY().doubleValue() + d8,
+                                        (double)p_107356_.getBigZ().doubleValue() + d9,
                                         d4 - 0.5,
                                         d5 - 0.5,
                                         d6 - 0.5,
@@ -524,9 +_,9 @@
     public void crack(BlockPos p_107368_, Direction p_107369_) {
         BlockState blockstate = this.level.getBlockState(p_107368_);
         if (blockstate.getRenderShape() != RenderShape.INVISIBLE && blockstate.shouldSpawnTerrainParticles()) {
-            int i = p_107368_.getX();
+            double i = p_107368_.getBigX().doubleValue();
             int j = p_107368_.getY();
-            int k = p_107368_.getZ();
+            double k = p_107368_.getBigZ().doubleValue();
             float f = 0.1F;
             AABB aabb = blockstate.getShape(this.level, p_107368_).bounds();
             double d0 = (double)i + this.random.nextDouble() * (aabb.maxX - aabb.minX - 0.2F) + 0.1F + aabb.minX;
