--- a/net/minecraft/client/particle/SingleQuadParticle.java
+++ b/net/minecraft/client/particle/SingleQuadParticle.java
@@ -40,7 +_,7 @@
     }
 
     protected void renderRotatedQuad(VertexConsumer p_345690_, Camera p_344809_, Quaternionf p_344798_, float p_345099_) {
-        Vec3 vec3 = p_344809_.getPosition();
+        me.alphamode.mcbig.core.BigVec3 vec3 = p_344809_.getPosition();
         float f = (float)(Mth.lerp((double)p_345099_, this.xo, this.x) - vec3.x());
         float f1 = (float)(Mth.lerp((double)p_345099_, this.yo, this.y) - vec3.y());
         float f2 = (float)(Mth.lerp((double)p_345099_, this.zo, this.z) - vec3.z());
