--- a/net/minecraft/client/particle/VibrationSignalParticle.java
+++ b/net/minecraft/client/particle/VibrationSignalParticle.java
@@ -25,7 +_,7 @@
         this.quadSize = 0.3F;
         this.target = p_234109_;
         this.lifetime = p_234110_;
-        Optional<Vec3> optional = p_234109_.getPosition(p_234105_);
+        Optional<Vec3> optional = p_234109_.getPosition(p_234105_).map(me.alphamode.mcbig.core.BigVec3::toVanilla);
         if (optional.isPresent()) {
             Vec3 vec3 = optional.get();
             double d0 = p_234106_ - vec3.x();
@@ -66,7 +_,7 @@
         if (this.age++ >= this.lifetime) {
             this.remove();
         } else {
-            Optional<Vec3> optional = this.target.getPosition(this.level);
+            Optional<Vec3> optional = this.target.getPosition(this.level).map(me.alphamode.mcbig.core.BigVec3::toVanilla);
             if (optional.isEmpty()) {
                 this.remove();
             } else {
