--- a/net/minecraft/client/multiplayer/ClientPacketListener.java
+++ b/net/minecraft/client/multiplayer/ClientPacketListener.java
@@ -540,7 +_,7 @@
         double d1 = p_104960_.getY();
         double d2 = p_104960_.getZ();
         Entity entity = new ExperienceOrb(this.level, d0, d1, d2, p_104960_.getValue());
-        entity.syncPacketPositionCodec(d0, d1, d2);
+        entity.syncPacketPositionCodec(me.alphamode.mcbig.math.BigDecimal.val(d0), d1, me.alphamode.mcbig.math.BigDecimal.val(d2));
         entity.setYRot(0.0F);
         entity.setXRot(0.0F);
         entity.setId(p_104960_.getId());
@@ -570,7 +_,7 @@
         PacketUtils.ensureRunningOnSameThread(p_380347_, this, this.minecraft);
         Entity entity = this.level.getEntity(p_380347_.id());
         if (entity != null) {
-            Vec3 vec3 = p_380347_.values().position();
+            me.alphamode.mcbig.core.BigVec3 vec3 = p_380347_.values().position();
             entity.getPositionCodec().setBase(vec3);
             if (!entity.isControlledByLocalInstance()) {
                 float f = p_380347_.values().yRot();
@@ -602,9 +_,9 @@
                 this.connection
                     .send(
                         new ServerboundMovePlayerPacket.PosRot(
-                            this.minecraft.player.getX(),
+                            this.minecraft.player.getBigX(),
                             this.minecraft.player.getY(),
-                            this.minecraft.player.getZ(),
+                            this.minecraft.player.getBigZ(),
                             this.minecraft.player.getYRot(),
                             this.minecraft.player.getXRot(),
                             false,
@@ -663,12 +_,12 @@
         if (entity != null) {
             if (entity.isControlledByLocalInstance()) {
                 VecDeltaCodec vecdeltacodec1 = entity.getPositionCodec();
-                Vec3 vec31 = vecdeltacodec1.decode((long)p_105036_.getXa(), (long)p_105036_.getYa(), (long)p_105036_.getZa());
+                me.alphamode.mcbig.core.BigVec3 vec31 = vecdeltacodec1.decode((long)p_105036_.getXa(), (long)p_105036_.getYa(), (long)p_105036_.getZa());
                 vecdeltacodec1.setBase(vec31);
             } else {
                 if (p_105036_.hasPosition()) {
                     VecDeltaCodec vecdeltacodec = entity.getPositionCodec();
-                    Vec3 vec3 = vecdeltacodec.decode((long)p_105036_.getXa(), (long)p_105036_.getYa(), (long)p_105036_.getZa());
+                    me.alphamode.mcbig.core.BigVec3 vec3 = vecdeltacodec.decode((long)p_105036_.getXa(), (long)p_105036_.getYa(), (long)p_105036_.getZa());
                     vecdeltacodec.setBase(vec3);
                     float f = p_105036_.hasRotation() ? p_105036_.getyRot() : entity.lerpTargetYRot();
                     float f1 = p_105036_.hasRotation() ? p_105036_.getxRot() : entity.lerpTargetXRot();
@@ -728,7 +_,7 @@
 
         this.connection.send(new ServerboundAcceptTeleportationPacket(p_105056_.id()));
         this.connection
-            .send(new ServerboundMovePlayerPacket.PosRot(player.getX(), player.getY(), player.getZ(), player.getYRot(), player.getXRot(), false, false));
+            .send(new ServerboundMovePlayerPacket.PosRot(player.getBigX(), player.getY(), player.getBigZ(), player.getYRot(), player.getXRot(), false, false));
     }
 
     private static boolean setValuesFromPositionPacket(PositionMoveRotation p_380237_, Set<Relative> p_380116_, Entity p_379700_, boolean p_379954_) {
@@ -777,8 +_,8 @@
     @Override
     public void handleLevelChunkWithLight(ClientboundLevelChunkWithLightPacket p_194241_) {
         PacketUtils.ensureRunningOnSameThread(p_194241_, this, this.minecraft);
-        int i = p_194241_.getX();
-        int j = p_194241_.getZ();
+        me.alphamode.mcbig.math.BigInteger i = p_194241_.getX();
+        me.alphamode.mcbig.math.BigInteger j = p_194241_.getZ();
         this.updateLevelChunk(i, j, p_194241_.getChunkData());
         ClientboundLightUpdatePacketData clientboundlightupdatepacketdata = p_194241_.getLightData();
         this.level.queueLightUpdate(() -> {
@@ -817,7 +_,7 @@
                         this.minecraft
                             .levelRenderer
                             .setSectionDirty(
-                                clientboundchunksbiomespacket$chunkbiomedata2.pos().x + i, k, clientboundchunksbiomespacket$chunkbiomedata2.pos().z + j
+                                clientboundchunksbiomespacket$chunkbiomedata2.pos().x.add(i), k, clientboundchunksbiomespacket$chunkbiomedata2.pos().z.add(j)
                             );
                     }
                 }
@@ -825,7 +_,7 @@
         }
     }
 
-    private void updateLevelChunk(int p_194199_, int p_194200_, ClientboundLevelChunkPacketData p_194201_) {
+    private void updateLevelChunk(me.alphamode.mcbig.math.BigInteger p_194199_, me.alphamode.mcbig.math.BigInteger p_194200_, ClientboundLevelChunkPacketData p_194201_) {
         this.level
             .getChunkSource()
             .replaceWithPacketData(
@@ -833,7 +_,7 @@
             );
     }
 
-    private void enableChunkLight(LevelChunk p_194213_, int p_194214_, int p_194215_) {
+    private void enableChunkLight(LevelChunk p_194213_, me.alphamode.mcbig.math.BigInteger p_194214_, me.alphamode.mcbig.math.BigInteger p_194215_) {
         LevelLightEngine levellightengine = this.level.getChunkSource().getLightEngine();
         LevelChunkSection[] alevelchunksection = p_194213_.getSections();
         ChunkPos chunkpos = p_194213_.getPos();
@@ -844,7 +_,7 @@
             levellightengine.updateSectionStatus(SectionPos.of(chunkpos, j), levelchunksection.hasOnlyAir());
         }
 
-        this.level.setSectionRangeDirty(p_194214_ - 1, this.level.getMinSectionY(), p_194215_ - 1, p_194214_ + 1, this.level.getMaxSectionY(), p_194215_ + 1);
+        this.level.setSectionRangeDirty(p_194214_.subtract(), this.level.getMinSectionY(), p_194215_.subtract(), p_194214_.add(), this.level.getMaxSectionY(), p_194215_.add());
     }
 
     @Override
@@ -2334,13 +_,13 @@
     @Override
     public void handleLightUpdatePacket(ClientboundLightUpdatePacket p_194243_) {
         PacketUtils.ensureRunningOnSameThread(p_194243_, this, this.minecraft);
-        int i = p_194243_.getX();
-        int j = p_194243_.getZ();
+        me.alphamode.mcbig.math.BigInteger i = p_194243_.getX();
+        me.alphamode.mcbig.math.BigInteger j = p_194243_.getZ();
         ClientboundLightUpdatePacketData clientboundlightupdatepacketdata = p_194243_.getLightData();
         this.level.queueLightUpdate(() -> this.applyLightData(i, j, clientboundlightupdatepacketdata, true));
     }
 
-    private void applyLightData(int p_194249_, int p_194250_, ClientboundLightUpdatePacketData p_194251_, boolean p_363269_) {
+    private void applyLightData(me.alphamode.mcbig.math.BigInteger p_194249_, me.alphamode.mcbig.math.BigInteger p_194250_, ClientboundLightUpdatePacketData p_194251_, boolean p_363269_) {
         LevelLightEngine levellightengine = this.level.getChunkSource().getLightEngine();
         BitSet bitset = p_194251_.getSkyYMask();
         BitSet bitset1 = p_194251_.getEmptySkyYMask();
@@ -2432,8 +_,8 @@
     }
 
     private void readSectionList(
-        int p_171735_,
-        int p_171736_,
+        me.alphamode.mcbig.math.BigInteger p_171735_,
+        me.alphamode.mcbig.math.BigInteger p_171736_,
         LevelLightEngine p_171737_,
         LightLayer p_171738_,
         BitSet p_171739_,
