--- a/net/minecraft/client/multiplayer/ClientChunkCache.java
+++ b/net/minecraft/client/multiplayer/ClientChunkCache.java
@@ -51,12 +_,12 @@
         return this.lightEngine;
     }
 
-    private static boolean isValidChunk(@Nullable LevelChunk p_104439_, int p_104440_, int p_104441_) {
+    private static boolean isValidChunk(@Nullable LevelChunk p_104439_, me.alphamode.mcbig.math.BigInteger p_104440_, me.alphamode.mcbig.math.BigInteger p_104441_) {
         if (p_104439_ == null) {
             return false;
         } else {
             ChunkPos chunkpos = p_104439_.getPos();
-            return chunkpos.x == p_104440_ && chunkpos.z == p_104441_;
+            return chunkpos.x.equals(p_104440_) && chunkpos.z.equals(p_104441_);
         }
     }
 
@@ -71,7 +_,7 @@
     }
 
     @Nullable
-    public LevelChunk getChunk(int p_104451_, int p_104452_, ChunkStatus p_330230_, boolean p_104454_) {
+    public LevelChunk getChunk(me.alphamode.mcbig.math.BigInteger p_104451_, me.alphamode.mcbig.math.BigInteger p_104452_, ChunkStatus p_104453_, boolean p_104454_) {
         if (this.storage.inRange(p_104451_, p_104452_)) {
             LevelChunk levelchunk = this.storage.getChunk(this.storage.getIndex(p_104451_, p_104452_));
             if (isValidChunk(levelchunk, p_104451_, p_104452_)) {
@@ -87,7 +_,7 @@
         return this.level;
     }
 
-    public void replaceBiomes(int p_275374_, int p_275226_, FriendlyByteBuf p_275745_) {
+    public void replaceBiomes(me.alphamode.mcbig.math.BigInteger p_275374_, me.alphamode.mcbig.math.BigInteger p_275226_, FriendlyByteBuf p_275745_) {
         if (!this.storage.inRange(p_275374_, p_275226_)) {
             LOGGER.warn("Ignoring chunk since it's not in the view range: {}, {}", p_275374_, p_275226_);
         } else {
@@ -103,8 +_,8 @@
 
     @Nullable
     public LevelChunk replaceWithPacketData(
-        int p_194117_,
-        int p_194118_,
+        me.alphamode.mcbig.math.BigInteger p_194117_,
+        me.alphamode.mcbig.math.BigInteger p_194118_,
         FriendlyByteBuf p_194119_,
         CompoundTag p_194120_,
         Consumer<ClientboundLevelChunkPacketData.BlockEntityTagOutput> p_194121_
@@ -134,7 +_,7 @@
     public void tick(BooleanSupplier p_202421_, boolean p_202422_) {
     }
 
-    public void updateViewCenter(int p_104460_, int p_104461_) {
+    public void updateViewCenter(me.alphamode.mcbig.math.BigInteger p_104460_, me.alphamode.mcbig.math.BigInteger p_104461_) {
         this.storage.viewCenterX = p_104460_;
         this.storage.viewCenterZ = p_104461_;
     }
@@ -177,26 +_,26 @@
 
     @Override
     public void onLightUpdate(LightLayer p_104436_, SectionPos p_104437_) {
-        Minecraft.getInstance().levelRenderer.setSectionDirty(p_104437_.x(), p_104437_.y(), p_104437_.z());
+        Minecraft.getInstance().levelRenderer.setSectionDirty(p_104437_.x(), p_104437_.getY(), p_104437_.z());
     }
 
-    public LongOpenHashSet getLoadedEmptySections() {
+    public it.unimi.dsi.fastutil.objects.ObjectOpenHashSet<SectionPos> getLoadedEmptySections() {
         return this.storage.loadedEmptySections;
     }
 
     @Override
-    public void onSectionEmptinessChanged(int p_366524_, int p_366407_, int p_366651_, boolean p_366887_) {
+    public void onSectionEmptinessChanged(me.alphamode.mcbig.math.BigInteger p_366524_, int p_366407_, me.alphamode.mcbig.math.BigInteger p_366651_, boolean p_366887_) {
         this.storage.onSectionEmptinessChanged(p_366524_, p_366407_, p_366651_, p_366887_);
     }
 
     @OnlyIn(Dist.CLIENT)
     final class Storage {
         final AtomicReferenceArray<LevelChunk> chunks;
-        final LongOpenHashSet loadedEmptySections = new LongOpenHashSet();
+        final it.unimi.dsi.fastutil.objects.ObjectOpenHashSet<SectionPos> loadedEmptySections = new it.unimi.dsi.fastutil.objects.ObjectOpenHashSet<>();
         final int chunkRadius;
         private final int viewRange;
-        volatile int viewCenterX;
-        volatile int viewCenterZ;
+        volatile me.alphamode.mcbig.math.BigInteger viewCenterX = me.alphamode.mcbig.math.BigInteger.ZERO;
+        volatile me.alphamode.mcbig.math.BigInteger viewCenterZ = me.alphamode.mcbig.math.BigInteger.ZERO;
         int chunkCount;
 
         Storage(int p_104474_) {
@@ -205,8 +_,8 @@
             this.chunks = new AtomicReferenceArray<>(this.viewRange * this.viewRange);
         }
 
-        int getIndex(int p_104482_, int p_104483_) {
-            return Math.floorMod(p_104483_, this.viewRange) * this.viewRange + Math.floorMod(p_104482_, this.viewRange);
+        int getIndex(me.alphamode.mcbig.math.BigInteger p_104482_, me.alphamode.mcbig.math.BigInteger p_104483_) {
+            return me.alphamode.mcbig.math.BigMath.floorMod(p_104483_, this.viewRange).intValue() * this.viewRange + me.alphamode.mcbig.math.BigMath.floorMod(p_104482_, this.viewRange).intValue();
         }
 
         void replace(int p_104485_, @Nullable LevelChunk p_104486_) {
@@ -232,9 +_,9 @@
             ClientChunkCache.this.level.unload(p_366475_);
         }
 
-        public void onSectionEmptinessChanged(int p_366606_, int p_366859_, int p_366870_, boolean p_366411_) {
+        public void onSectionEmptinessChanged(me.alphamode.mcbig.math.BigInteger p_366606_, int p_366859_, me.alphamode.mcbig.math.BigInteger p_366870_, boolean p_366411_) {
             if (this.inRange(p_366606_, p_366870_)) {
-                long i = SectionPos.asLong(p_366606_, p_366859_, p_366870_);
+                SectionPos i = SectionPos.of(p_366606_, p_366859_, p_366870_);
                 if (p_366411_) {
                     this.loadedEmptySections.add(i);
                 } else if (this.loadedEmptySections.remove(i)) {
@@ -248,7 +_,7 @@
 
             for (int i = 0; i < alevelchunksection.length; i++) {
                 ChunkPos chunkpos = p_366765_.getPos();
-                this.loadedEmptySections.remove(SectionPos.asLong(chunkpos.x, p_366765_.getSectionYFromSectionIndex(i), chunkpos.z));
+                this.loadedEmptySections.remove(SectionPos.of(chunkpos.x, p_366765_.getSectionYFromSectionIndex(i), chunkpos.z));
             }
         }
 
@@ -259,7 +_,7 @@
                 LevelChunkSection levelchunksection = alevelchunksection[i];
                 if (levelchunksection.hasOnlyAir()) {
                     ChunkPos chunkpos = p_366694_.getPos();
-                    this.loadedEmptySections.add(SectionPos.asLong(chunkpos.x, p_366694_.getSectionYFromSectionIndex(i), chunkpos.z));
+                    this.loadedEmptySections.add(SectionPos.of(chunkpos.x, p_366694_.getSectionYFromSectionIndex(i), chunkpos.z));
                 }
             }
         }
@@ -270,7 +_,7 @@
 
             for (int i = 0; i < alevelchunksection.length; i++) {
                 LevelChunkSection levelchunksection = alevelchunksection[i];
-                long j = SectionPos.asLong(chunkpos.x, p_386574_.getSectionYFromSectionIndex(i), chunkpos.z);
+                SectionPos j = SectionPos.of(chunkpos.x, p_386574_.getSectionYFromSectionIndex(i), chunkpos.z);
                 if (levelchunksection.hasOnlyAir()) {
                     this.loadedEmptySections.add(j);
                 } else if (this.loadedEmptySections.remove(j)) {
@@ -279,8 +_,8 @@
             }
         }
 
-        boolean inRange(int p_104501_, int p_104502_) {
-            return Math.abs(p_104501_ - this.viewCenterX) <= this.chunkRadius && Math.abs(p_104502_ - this.viewCenterZ) <= this.chunkRadius;
+        boolean inRange(me.alphamode.mcbig.math.BigInteger p_104501_, me.alphamode.mcbig.math.BigInteger p_104502_) {
+            return p_104501_.subtract(this.viewCenterX).abs().intValue() <= this.chunkRadius && p_104502_.subtract(this.viewCenterZ).abs().intValue() <= this.chunkRadius;
         }
 
         @Nullable
@@ -292,8 +_,8 @@
             try (FileOutputStream fileoutputstream = new FileOutputStream(p_171623_)) {
                 int i = ClientChunkCache.this.storage.chunkRadius;
 
-                for (int j = this.viewCenterZ - i; j <= this.viewCenterZ + i; j++) {
-                    for (int k = this.viewCenterX - i; k <= this.viewCenterX + i; k++) {
+                for(me.alphamode.mcbig.math.BigInteger j = this.viewCenterZ.subtract(i); j.compareTo(this.viewCenterZ.add(i)) <= 0; j = j.add()) {
+                    for(me.alphamode.mcbig.math.BigInteger k = this.viewCenterX.subtract(i); k.compareTo(this.viewCenterX.add(i)) <= 0; k = k.add()) {
                         LevelChunk levelchunk = ClientChunkCache.this.storage.chunks.get(ClientChunkCache.this.storage.getIndex(k, j));
                         if (levelchunk != null) {
                             ChunkPos chunkpos = levelchunk.getPos();
