--- a/net/minecraft/world/level/gameevent/GameEventListenerRegistry.java
+++ b/net/minecraft/world/level/gameevent/GameEventListenerRegistry.java
@@ -20,7 +_,7 @@
 
         @Override
         public boolean visitInRangeListeners(
-            Holder<GameEvent> p_316158_, Vec3 p_249086_, GameEvent.Context p_249012_, GameEventListenerRegistry.ListenerVisitor p_252106_
+                Holder<GameEvent> p_316158_, me.alphamode.mcbig.core.BigVec3 p_249086_, GameEvent.Context p_249012_, GameEventListenerRegistry.ListenerVisitor p_252106_
         ) {
             return false;
         }
@@ -32,10 +_,10 @@
 
     void unregister(GameEventListener p_248758_);
 
-    boolean visitInRangeListeners(Holder<GameEvent> p_316434_, Vec3 p_249144_, GameEvent.Context p_249328_, GameEventListenerRegistry.ListenerVisitor p_250123_);
+    boolean visitInRangeListeners(Holder<GameEvent> p_316434_, me.alphamode.mcbig.core.BigVec3 p_249144_, GameEvent.Context p_249328_, GameEventListenerRegistry.ListenerVisitor p_250123_);
 
     @FunctionalInterface
     public interface ListenerVisitor {
-        void visit(GameEventListener p_250787_, Vec3 p_251603_);
+        void visit(GameEventListener p_250787_, me.alphamode.mcbig.core.BigVec3 p_251603_);
     }
 }
