--- a/net/minecraft/world/level/levelgen/blending/Blender.java
+++ b/net/minecraft/world/level/levelgen/blending/Blender.java
@@ -36,9 +_,9 @@
 import org.apache.commons.lang3.mutable.MutableObject;
 
 public class Blender {
-    private static final Blender EMPTY = new Blender(new Long2ObjectOpenHashMap(), new Long2ObjectOpenHashMap()) {
+    private static final Blender EMPTY = new Blender(new it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap<>(), new it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap<>()) {
         @Override
-        public Blender.BlendingOutput blendOffsetAndFactor(int p_209724_, int p_209725_) {
+        public Blender.BlendingOutput blendOffsetAndFactor(me.alphamode.mcbig.math.BigInteger p_209724_, me.alphamode.mcbig.math.BigInteger p_209725_) {
             return new Blender.BlendingOutput(1.0, 0.0);
         }
 
@@ -58,8 +_,8 @@
     private static final int DENSITY_BLENDING_RANGE_CELLS = 2;
     private static final int DENSITY_BLENDING_RANGE_CHUNKS = QuartPos.toSection(5);
     private static final double OLD_CHUNK_XZ_RADIUS = 8.0;
-    private final Long2ObjectOpenHashMap<BlendingData> heightAndBiomeBlendingData;
-    private final Long2ObjectOpenHashMap<BlendingData> densityBlendingData;
+    private final it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap<ChunkPos, BlendingData> heightAndBiomeBlendingData;
+    private final it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap<ChunkPos, BlendingData> densityBlendingData;
 
     public static Blender empty() {
         return EMPTY;
@@ -73,23 +_,24 @@
             if (!p_190203_.isOldChunkAround(chunkpos, HEIGHT_BLENDING_RANGE_CHUNKS)) {
                 return EMPTY;
             } else {
-                Long2ObjectOpenHashMap<BlendingData> long2objectopenhashmap = new Long2ObjectOpenHashMap<>();
-                Long2ObjectOpenHashMap<BlendingData> long2objectopenhashmap1 = new Long2ObjectOpenHashMap<>();
+                it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap<ChunkPos, BlendingData> long2objectopenhashmap = new it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap<>();
+                it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap<ChunkPos, BlendingData> long2objectopenhashmap1 = new it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap<>();
                 int i = Mth.square(HEIGHT_BLENDING_RANGE_CHUNKS + 1);
 
                 for(int j = -HEIGHT_BLENDING_RANGE_CHUNKS; j <= HEIGHT_BLENDING_RANGE_CHUNKS; ++j) {
                     for(int k = -HEIGHT_BLENDING_RANGE_CHUNKS; k <= HEIGHT_BLENDING_RANGE_CHUNKS; ++k) {
                         if (j * j + k * k <= i) {
-                            int l = chunkpos.x + j;
-                            int i1 = chunkpos.z + k;
+                            me.alphamode.mcbig.math.BigInteger l = chunkpos.x().add(j);
+                            me.alphamode.mcbig.math.BigInteger i1 = chunkpos.z().add(k);
                             BlendingData blendingdata = BlendingData.getOrUpdateBlendingData(p_190203_, l, i1);
                             if (blendingdata != null) {
-                                long2objectopenhashmap.put(ChunkPos.asLong(l, i1), blendingdata);
+                                ChunkPos pos = new ChunkPos(l, i1);
+                                long2objectopenhashmap.put(pos, blendingdata);
                                 if (j >= -DENSITY_BLENDING_RANGE_CHUNKS
                                     && j <= DENSITY_BLENDING_RANGE_CHUNKS
                                     && k >= -DENSITY_BLENDING_RANGE_CHUNKS
                                     && k <= DENSITY_BLENDING_RANGE_CHUNKS) {
-                                    long2objectopenhashmap1.put(ChunkPos.asLong(l, i1), blendingdata);
+                                    long2objectopenhashmap1.put(pos, blendingdata);
                                 }
                             }
                         }
@@ -103,14 +_,14 @@
         }
     }
 
-    Blender(Long2ObjectOpenHashMap<BlendingData> p_202197_, Long2ObjectOpenHashMap<BlendingData> p_202198_) {
+    Blender(it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap<ChunkPos, BlendingData> p_202197_, it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap<ChunkPos, BlendingData> p_202198_) {
         this.heightAndBiomeBlendingData = p_202197_;
         this.densityBlendingData = p_202198_;
     }
 
-    public Blender.BlendingOutput blendOffsetAndFactor(int p_209719_, int p_209720_) {
-        int i = QuartPos.fromBlock(p_209719_);
-        int j = QuartPos.fromBlock(p_209720_);
+    public Blender.BlendingOutput blendOffsetAndFactor(me.alphamode.mcbig.math.BigInteger p_209719_, me.alphamode.mcbig.math.BigInteger p_209720_) {
+        me.alphamode.mcbig.math.BigInteger i = QuartPos.fromBlock(p_209719_);
+        me.alphamode.mcbig.math.BigInteger j = QuartPos.fromBlock(p_209720_);
         double d0 = this.getBlendingDataValue(i, 0, j, BlendingData::getHeight);
         if (d0 != Double.MAX_VALUE) {
             return new Blender.BlendingOutput(0.0, heightToOffset(d0));
@@ -121,10 +_,10 @@
             this.heightAndBiomeBlendingData
                 .forEach(
                     (p_202249_, p_202250_) -> p_202250_.iterateHeights(
-                            QuartPos.fromSection(ChunkPos.getX(p_202249_)),
-                            QuartPos.fromSection(ChunkPos.getZ(p_202249_)),
+                            QuartPos.fromSection(p_202249_.x()),
+                            QuartPos.fromSection(p_202249_.z()),
                             (p_190199_, p_190200_, p_190201_) -> {
-                                double d3 = Mth.length((double)(i - p_190199_), (double)(j - p_190200_));
+                                double d3 = Mth.length((double)(i.subtract(p_190199_).intValue()), (double)(j.subtract(p_190200_).intValue()));
                                 if (!(d3 > (double)HEIGHT_BLENDING_RANGE_CELLS)) {
                                     if (d3 < mutabledouble2.doubleValue()) {
                                         mutabledouble2.setValue(d3);
@@ -156,9 +_,9 @@
     }
 
     public double blendDensity(DensityFunction.FunctionContext p_209721_, double p_209722_) {
-        int i = QuartPos.fromBlock(p_209721_.blockX());
+        me.alphamode.mcbig.math.BigInteger i = QuartPos.fromBlock(p_209721_.blockX());
         int j = p_209721_.blockY() / 8;
-        int k = QuartPos.fromBlock(p_209721_.blockZ());
+        me.alphamode.mcbig.math.BigInteger k = QuartPos.fromBlock(p_209721_.blockZ());
         double d0 = this.getBlendingDataValue(i, j, k, BlendingData::getDensity);
         if (d0 != Double.MAX_VALUE) {
             return d0;
@@ -169,12 +_,12 @@
             this.densityBlendingData
                 .forEach(
                     (p_202241_, p_202242_) -> p_202242_.iterateDensities(
-                            QuartPos.fromSection(ChunkPos.getX(p_202241_)),
-                            QuartPos.fromSection(ChunkPos.getZ(p_202241_)),
+                            QuartPos.fromSection(p_202241_.x().intValue()),
+                            QuartPos.fromSection(p_202241_.z().intValue()),
                             j - 1,
                             j + 1,
                             (p_202230_, p_202231_, p_202232_, p_202233_) -> {
-                                double d3 = Mth.length((double)(i - p_202230_), (double)((j - p_202231_) * 2), (double)(k - p_202232_));
+                                double d3 = Mth.length((double)(i.subtract(p_202230_).intValue()), (double)((j - p_202231_) * 2), (double)(k.subtract(p_202232_).intValue()));
                                 if (!(d3 > 2.0)) {
                                     if (d3 < mutabledouble2.doubleValue()) {
                                         mutabledouble2.setValue(d3);
@@ -197,24 +_,24 @@
         }
     }
 
-    private double getBlendingDataValue(int p_190175_, int p_190176_, int p_190177_, Blender.CellValueGetter p_190178_) {
-        int i = QuartPos.toSection(p_190175_);
-        int j = QuartPos.toSection(p_190177_);
-        boolean flag = (p_190175_ & 3) == 0;
-        boolean flag1 = (p_190177_ & 3) == 0;
+    private double getBlendingDataValue(me.alphamode.mcbig.math.BigInteger p_190175_, int p_190176_, me.alphamode.mcbig.math.BigInteger p_190177_, Blender.CellValueGetter p_190178_) {
+        me.alphamode.mcbig.math.BigInteger i = QuartPos.toSection(p_190175_);
+        me.alphamode.mcbig.math.BigInteger j = QuartPos.toSection(p_190177_);
+        boolean flag = (p_190175_.and(me.alphamode.mcbig.core.BigConstants.Ints.THREE).intValue()) == 0;
+        boolean flag1 = (p_190177_.and(me.alphamode.mcbig.core.BigConstants.Ints.THREE).intValue()) == 0;
         double d0 = this.getBlendingDataValue(p_190178_, i, j, p_190175_, p_190176_, p_190177_);
         if (d0 == Double.MAX_VALUE) {
             if (flag && flag1) {
-                d0 = this.getBlendingDataValue(p_190178_, i - 1, j - 1, p_190175_, p_190176_, p_190177_);
+                d0 = this.getBlendingDataValue(p_190178_, i.subtract(), j.subtract(), p_190175_, p_190176_, p_190177_);
             }
 
             if (d0 == Double.MAX_VALUE) {
                 if (flag) {
-                    d0 = this.getBlendingDataValue(p_190178_, i - 1, j, p_190175_, p_190176_, p_190177_);
+                    d0 = this.getBlendingDataValue(p_190178_, i.subtract(), j, p_190175_, p_190176_, p_190177_);
                 }
 
                 if (d0 == Double.MAX_VALUE && flag1) {
-                    d0 = this.getBlendingDataValue(p_190178_, i, j - 1, p_190175_, p_190176_, p_190177_);
+                    d0 = this.getBlendingDataValue(p_190178_, i, j.subtract(), p_190175_, p_190176_, p_190177_);
                 }
             }
         }
@@ -222,10 +_,10 @@
         return d0;
     }
 
-    private double getBlendingDataValue(Blender.CellValueGetter p_190212_, int p_190213_, int p_190214_, int p_190215_, int p_190216_, int p_190217_) {
-        BlendingData blendingdata = this.heightAndBiomeBlendingData.get(ChunkPos.asLong(p_190213_, p_190214_));
+    private double getBlendingDataValue(Blender.CellValueGetter p_190212_, me.alphamode.mcbig.math.BigInteger p_190213_, me.alphamode.mcbig.math.BigInteger p_190214_, me.alphamode.mcbig.math.BigInteger p_190215_, int p_190216_, me.alphamode.mcbig.math.BigInteger p_190217_) {
+        BlendingData blendingdata = this.heightAndBiomeBlendingData.get(new ChunkPos(p_190213_, p_190214_));
         return blendingdata != null
-            ? p_190212_.get(blendingdata, p_190215_ - QuartPos.fromSection(p_190213_), p_190216_, p_190217_ - QuartPos.fromSection(p_190214_))
+            ? p_190212_.get(blendingdata, p_190215_.subtract(QuartPos.fromSection(p_190213_)).intValue(), p_190216_, p_190217_.subtract(QuartPos.fromSection(p_190214_)).intValue())
             : Double.MAX_VALUE;
     }
 
@@ -237,17 +_,17 @@
     }
 
     @Nullable
-    private Holder<Biome> blendBiome(int p_224707_, int p_224708_, int p_224709_) {
+    private Holder<Biome> blendBiome(me.alphamode.mcbig.math.BigInteger p_224707_, me.alphamode.mcbig.math.BigInteger p_224708_, me.alphamode.mcbig.math.BigInteger p_224709_) {
         MutableDouble mutabledouble = new MutableDouble(Double.POSITIVE_INFINITY);
         MutableObject<Holder<Biome>> mutableobject = new MutableObject<>();
         this.heightAndBiomeBlendingData
             .forEach(
                 (p_224716_, p_224717_) -> p_224717_.iterateBiomes(
-                        QuartPos.fromSection(ChunkPos.getX(p_224716_)),
-                        p_224708_,
-                        QuartPos.fromSection(ChunkPos.getZ(p_224716_)),
+                        QuartPos.fromSection(p_224716_.x().intValue()),
+                        p_224708_.intValue(),
+                        QuartPos.fromSection(p_224716_.z().intValue()),
                         (p_224723_, p_224724_, p_224725_) -> {
-                            double d2 = Mth.length((double)(p_224707_ - p_224723_), (double)(p_224709_ - p_224724_));
+                            double d2 = Mth.length((double)(p_224707_.subtract(p_224723_)).doubleValue(), (double)(p_224709_.subtract(p_224724_)).doubleValue());
                             if (!(d2 > (double)HEIGHT_BLENDING_RANGE_CELLS)) {
                                 if (d2 < mutabledouble.doubleValue()) {
                                     mutableobject.setValue(p_224725_);
@@ -260,7 +_,7 @@
         if (mutabledouble.doubleValue() == Double.POSITIVE_INFINITY) {
             return null;
         } else {
-            double d0 = SHIFT_NOISE.getValue((double)p_224707_, 0.0, (double)p_224709_) * 12.0;
+            double d0 = SHIFT_NOISE.getValue((double)p_224707_.doubleValue(), 0.0, (double)p_224709_.doubleValue()) * 12.0;
             double d1 = Mth.clamp((mutabledouble.doubleValue() + d0) / (double)(HEIGHT_BLENDING_RANGE_CELLS + 1), 0.0, 1.0);
             return d1 > 0.5 ? null : mutableobject.getValue();
         }
@@ -287,7 +_,7 @@
             }
 
             for(Direction direction : Direction.Plane.HORIZONTAL) {
-                if (p_197032_.getChunk(chunkpos.x + direction.getStepX(), chunkpos.z + direction.getStepZ()).isOldNoiseGeneration() != flag) {
+                if (p_197032_.getChunk(chunkpos.x().add(direction.getStepX()), chunkpos.z().add(direction.getStepZ())).isOldNoiseGeneration() != flag) {
                     int i1 = direction == Direction.EAST ? 15 : 0;
                     int j1 = direction == Direction.WEST ? 0 : 15;
                     int k1 = direction == Direction.SOUTH ? 15 : 0;
@@ -295,7 +_,7 @@
 
                     for(int i2 = i1; i2 <= j1; ++i2) {
                         for(int j2 = k1; j2 <= l1; ++j2) {
-                            int k2 = Math.min(j, p_197033_.getHeight(Heightmap.Types.MOTION_BLOCKING, i2, j2)) + 1;
+                            int k2 = Math.min(j, p_197033_.getHeight(Heightmap.Types.MOTION_BLOCKING, me.alphamode.mcbig.math.BigInteger.val(i2), me.alphamode.mcbig.math.BigInteger.val(j2))) + 1;
 
                             for(int l2 = i; l2 < k2; ++l2) {
                                 generateBorderTick(p_197033_, blockpos$mutableblockpos.setWithOffset(blockpos, i2, l2, j2));
@@ -324,8 +_,8 @@
         Builder<Direction8, BlendingData> builder = ImmutableMap.builder();
 
         for(Direction8 direction8 : Direction8.values()) {
-            int i = chunkpos.x + direction8.getStepX();
-            int j = chunkpos.z + direction8.getStepZ();
+            me.alphamode.mcbig.math.BigInteger i = chunkpos.x().add(direction8.getStepX());
+            me.alphamode.mcbig.math.BigInteger j = chunkpos.z().add(direction8.getStepZ());
             BlendingData blendingdata = p_197035_.getChunk(i, j).getBlendingData();
             if (blendingdata != null) {
                 builder.put(direction8, blendingdata);
