--- a/net/minecraft/world/level/levelgen/blockpredicates/InsideWorldBoundsPredicate.java
+++ b/net/minecraft/world/level/levelgen/blockpredicates/InsideWorldBoundsPredicate.java
@@ -9,12 +_,12 @@
 
 public class InsideWorldBoundsPredicate implements BlockPredicate {
     public static final Codec<InsideWorldBoundsPredicate> CODEC = RecordCodecBuilder.create(
-        p_190473_ -> p_190473_.group(Vec3i.offsetCodec(16).optionalFieldOf("offset", BlockPos.ZERO).forGetter(p_190475_ -> p_190475_.offset))
+        p_190473_ -> p_190473_.group(me.alphamode.mcbig.core.Vec3l.offsetCodec(16).optionalFieldOf("offset", BlockPos.ZERO).forGetter(p_190475_ -> p_190475_.offset))
                 .apply(p_190473_, InsideWorldBoundsPredicate::new)
     );
-    private final Vec3i offset;
+    private final me.alphamode.mcbig.core.Vec3l offset;
 
-    public InsideWorldBoundsPredicate(Vec3i p_190467_) {
+    public InsideWorldBoundsPredicate(me.alphamode.mcbig.core.Vec3l p_190467_) {
         this.offset = p_190467_;
     }
 
