--- a/net/minecraft/client/renderer/CloudRenderer.java
+++ b/net/minecraft/client/renderer/CloudRenderer.java
@@ -39,8 +_,8 @@
     private static final int SOUTH_OFFSET = 1;
     private static final int WEST_OFFSET = 0;
     private boolean needsRebuild = true;
-    private int prevCellX = Integer.MIN_VALUE;
-    private int prevCellZ = Integer.MIN_VALUE;
+    private me.alphamode.mcbig.math.BigInteger prevCellX = me.alphamode.mcbig.math.BigInteger.val(Integer.MIN_VALUE);
+    private me.alphamode.mcbig.math.BigInteger prevCellZ = me.alphamode.mcbig.math.BigInteger.val(Integer.MIN_VALUE);
     private CloudRenderer.RelativeCameraPos prevRelativeCameraPos = CloudRenderer.RelativeCameraPos.INSIDE_CLOUDS;
     @Nullable
     private CloudStatus prevType;
@@ -122,7 +_,7 @@
         return (p_363963_ >> 0 & 1L) != 0L;
     }
 
-    public void render(int p_363907_, CloudStatus p_364293_, float p_363260_, Matrix4f p_363292_, Matrix4f p_364300_, Vec3 p_363573_, float p_360711_) {
+    public void render(int p_363907_, CloudStatus p_364293_, float p_363260_, Matrix4f p_363292_, Matrix4f p_364300_, me.alphamode.mcbig.core.BigVec3 p_363573_, float p_360711_) {
         if (this.texture != null) {
             float f = (float)((double)p_363260_ - p_363573_.y);
             float f1 = f + 4.0F;
@@ -135,21 +_,21 @@
                 cloudrenderer$relativecamerapos = CloudRenderer.RelativeCameraPos.INSIDE_CLOUDS;
             }
 
-            double d0 = p_363573_.x + (double)(p_360711_ * 0.030000001F);
-            double d1 = p_363573_.z + 3.96F;
+            me.alphamode.mcbig.math.BigDecimal d0 = p_363573_.x.add((double)(p_360711_ * 0.030000001F));
+            me.alphamode.mcbig.math.BigDecimal d1 = p_363573_.z.add(3.96F);
             double d2 = (double)this.texture.width * 12.0;
             double d3 = (double)this.texture.height * 12.0;
-            d0 -= (double)Mth.floor(d0 / d2) * d2;
-            d1 -= (double)Mth.floor(d1 / d3) * d3;
-            int i = Mth.floor(d0 / 12.0);
-            int j = Mth.floor(d1 / 12.0);
-            float f2 = (float)(d0 - (double)((float)i * 12.0F));
-            float f3 = (float)(d1 - (double)((float)j * 12.0F));
+            d0 = d0.subtract(Mth.bigFloor(d0.divide(d2)).toBigDecimal().multiply(d2));
+            d1 = d1.subtract(Mth.bigFloor(d1.divide(d3)).toBigDecimal().multiply(d3));
+            me.alphamode.mcbig.math.BigInteger i = Mth.bigFloor(d0.divide(me.alphamode.mcbig.core.BigConstants.TWELVE));
+            me.alphamode.mcbig.math.BigInteger j = Mth.bigFloor(d1.divide(me.alphamode.mcbig.core.BigConstants.TWELVE));
+            float f2 = (float)(d0.floatValue() - (double)((float)i.floatValue() * 12.0F));
+            float f3 = (float)(d1.floatValue() - (double)((float)j.floatValue() * 12.0F));
             RenderType rendertype = p_364293_ == CloudStatus.FANCY ? RenderType.clouds() : RenderType.flatClouds();
             this.vertexBuffer.bind();
             if (this.needsRebuild
-                || i != this.prevCellX
-                || j != this.prevCellZ
+                || !i.equals(this.prevCellX)
+                || !j.equals(this.prevCellZ)
                 || cloudrenderer$relativecamerapos != this.prevRelativeCameraPos
                 || p_364293_ != this.prevType) {
                 this.needsRebuild = false;
@@ -192,7 +_,7 @@
 
     @Nullable
     private MeshData buildMesh(
-        Tesselator p_362597_, int p_362701_, int p_361589_, CloudStatus p_365402_, CloudRenderer.RelativeCameraPos p_364842_, RenderType p_364348_
+        Tesselator p_362597_, me.alphamode.mcbig.math.BigInteger p_362701_, me.alphamode.mcbig.math.BigInteger p_361589_, CloudStatus p_365402_, CloudRenderer.RelativeCameraPos p_364842_, RenderType p_364348_
     ) {
         float f = 0.8F;
         int i = ARGB.colorFromFloat(0.8F, 1.0F, 1.0F, 1.0F);
@@ -207,8 +_,8 @@
     private void buildMesh(
         CloudRenderer.RelativeCameraPos p_363221_,
         BufferBuilder p_364486_,
-        int p_361006_,
-        int p_362674_,
+        me.alphamode.mcbig.math.BigInteger p_361006_,
+        me.alphamode.mcbig.math.BigInteger p_362674_,
         int p_362100_,
         int p_360889_,
         int p_360776_,
@@ -223,8 +_,8 @@
 
             for (int l = -32; l <= 32; l++) {
                 for (int i1 = -32; i1 <= 32; i1++) {
-                    int j1 = Math.floorMod(p_361006_ + i1, j);
-                    int k1 = Math.floorMod(p_362674_ + l, k);
+                    int j1 = me.alphamode.mcbig.math.BigMath.floorMod(p_361006_.add(i1), j).intValue();
+                    int k1 = me.alphamode.mcbig.math.BigMath.floorMod(p_362674_.add(l), k).intValue();
                     long l1 = along[j1 + k1 * j];
                     if (l1 != 0L) {
                         int i2 = getColor(l1);
