--- a/net/minecraft/client/renderer/SectionOcclusionGraph.java
+++ b/net/minecraft/client/renderer/SectionOcclusionGraph.java
@@ -117,8 +_,8 @@
         }
     }
 
-    public void update(boolean p_294298_, Camera p_294529_, Frustum p_294426_, List<SectionRenderDispatcher.RenderSection> p_295280_, LongOpenHashSet p_366410_) {
-        Vec3 vec3 = p_294529_.getPosition();
+    public void update(boolean p_294298_, Camera p_294529_, Frustum p_294426_, List<SectionRenderDispatcher.RenderSection> p_295280_, it.unimi.dsi.fastutil.objects.ObjectOpenHashSet<SectionPos> p_366410_) {
+        me.alphamode.mcbig.core.BigVec3 vec3 = p_294529_.getPosition();
         if (this.needsFullUpdate && (this.fullUpdateTask == null || this.fullUpdateTask.isDone())) {
             this.scheduleFullUpdate(p_294298_, p_294529_, vec3, p_366410_);
         }
@@ -126,9 +_,9 @@
         this.runPartialUpdate(p_294298_, p_294426_, p_295280_, vec3, p_366410_);
     }
 
-    private void scheduleFullUpdate(boolean p_294514_, Camera p_295663_, Vec3 p_295096_, LongOpenHashSet p_366772_) {
+    private void scheduleFullUpdate(boolean p_294514_, Camera p_295663_, me.alphamode.mcbig.core.BigVec3 p_295096_, it.unimi.dsi.fastutil.objects.ObjectOpenHashSet<SectionPos> p_366772_) {
         this.needsFullUpdate = false;
-        LongOpenHashSet longopenhashset = p_366772_.clone();
+        it.unimi.dsi.fastutil.objects.ObjectOpenHashSet<SectionPos> longopenhashset = p_366772_.clone();
         this.fullUpdateTask = CompletableFuture.runAsync(() -> {
             SectionOcclusionGraph.GraphState sectionocclusiongraph$graphstate = new SectionOcclusionGraph.GraphState(this.viewArea);
             this.nextGraphEvents.set(sectionocclusiongraph$graphstate.events);
@@ -144,7 +_,7 @@
     }
 
     private void runPartialUpdate(
-        boolean p_294795_, Frustum p_294341_, List<SectionRenderDispatcher.RenderSection> p_294796_, Vec3 p_295915_, LongOpenHashSet p_366753_
+        boolean p_294795_, Frustum p_294341_, List<SectionRenderDispatcher.RenderSection> p_294796_, me.alphamode.mcbig.core.BigVec3 p_295915_, it.unimi.dsi.fastutil.objects.ObjectOpenHashSet<SectionPos> p_366753_
     ) {
         SectionOcclusionGraph.GraphState sectionocclusiongraph$graphstate = this.currentGraph.get();
         this.queueSectionsWithNewNeighbors(sectionocclusiongraph$graphstate);
@@ -174,10 +_,10 @@
     }
 
     private void queueSectionsWithNewNeighbors(SectionOcclusionGraph.GraphState p_296471_) {
-        LongIterator longiterator = p_296471_.events.chunksWhichReceivedNeighbors.iterator();
+        it.unimi.dsi.fastutil.objects.ObjectIterator<ChunkPos> longiterator = p_296471_.events.chunksWhichReceivedNeighbors.iterator();
 
         while (longiterator.hasNext()) {
-            long i = longiterator.nextLong();
+            ChunkPos i = longiterator.next();
             List<SectionRenderDispatcher.RenderSection> list = p_296471_.storage.chunksWaitingForNeighbors.get(i);
             if (list != null && list.get(0).hasAllNeighbors()) {
                 p_296471_.events.sectionsToPropagateFrom.addAll(list);
@@ -189,20 +_,20 @@
     }
 
     private void addNeighbors(SectionOcclusionGraph.GraphEvents p_295866_, ChunkPos p_295968_) {
-        p_295866_.chunksWhichReceivedNeighbors.add(ChunkPos.asLong(p_295968_.x - 1, p_295968_.z));
-        p_295866_.chunksWhichReceivedNeighbors.add(ChunkPos.asLong(p_295968_.x, p_295968_.z - 1));
-        p_295866_.chunksWhichReceivedNeighbors.add(ChunkPos.asLong(p_295968_.x + 1, p_295968_.z));
-        p_295866_.chunksWhichReceivedNeighbors.add(ChunkPos.asLong(p_295968_.x, p_295968_.z + 1));
-        p_295866_.chunksWhichReceivedNeighbors.add(ChunkPos.asLong(p_295968_.x - 1, p_295968_.z - 1));
-        p_295866_.chunksWhichReceivedNeighbors.add(ChunkPos.asLong(p_295968_.x - 1, p_295968_.z + 1));
-        p_295866_.chunksWhichReceivedNeighbors.add(ChunkPos.asLong(p_295968_.x + 1, p_295968_.z - 1));
-        p_295866_.chunksWhichReceivedNeighbors.add(ChunkPos.asLong(p_295968_.x + 1, p_295968_.z + 1));
+        p_295866_.chunksWhichReceivedNeighbors.add(new ChunkPos(p_295968_.x.subtract(), p_295968_.z));
+        p_295866_.chunksWhichReceivedNeighbors.add(new ChunkPos(p_295968_.x, p_295968_.z.subtract()));
+        p_295866_.chunksWhichReceivedNeighbors.add(new ChunkPos(p_295968_.x.add(), p_295968_.z));
+        p_295866_.chunksWhichReceivedNeighbors.add(new ChunkPos(p_295968_.x, p_295968_.z.add()));
+        p_295866_.chunksWhichReceivedNeighbors.add(new ChunkPos(p_295968_.x.subtract(), p_295968_.z.subtract()));
+        p_295866_.chunksWhichReceivedNeighbors.add(new ChunkPos(p_295968_.x.subtract(), p_295968_.z.add()));
+        p_295866_.chunksWhichReceivedNeighbors.add(new ChunkPos(p_295968_.x.add(), p_295968_.z.subtract()));
+        p_295866_.chunksWhichReceivedNeighbors.add(new ChunkPos(p_295968_.x.add(), p_295968_.z.add()));
     }
 
     private void initializeQueueForFullUpdate(Camera p_295148_, Queue<SectionOcclusionGraph.Node> p_294801_) {
         BlockPos blockpos = p_295148_.getBlockPosition();
-        long i = SectionPos.asLong(blockpos);
-        int j = SectionPos.y(i);
+        SectionPos i = SectionPos.of(blockpos);
+        int j = i.y().intValue();
         SectionRenderDispatcher.RenderSection sectionrenderdispatcher$rendersection = this.viewArea.getRenderSection(i);
         if (sectionrenderdispatcher$rendersection == null) {
             LevelHeightAccessor levelheightaccessor = this.viewArea.getLevelHeightAccessor();
@@ -210,13 +_,13 @@
             int k = flag ? levelheightaccessor.getMinSectionY() : levelheightaccessor.getMaxSectionY();
             int l = this.viewArea.getViewDistance();
             List<SectionOcclusionGraph.Node> list = Lists.newArrayList();
-            int i1 = SectionPos.x(i);
-            int j1 = SectionPos.z(i);
+            me.alphamode.mcbig.math.BigInteger i1 = i.x();
+            me.alphamode.mcbig.math.BigInteger j1 = i.z();
 
             for (int k1 = -l; k1 <= l; k1++) {
                 for (int l1 = -l; l1 <= l; l1++) {
                     SectionRenderDispatcher.RenderSection sectionrenderdispatcher$rendersection1 = this.viewArea
-                        .getRenderSection(SectionPos.asLong(k1 + i1, k, l1 + j1));
+                        .getRenderSection(SectionPos.of(i1.add(k1), k, j1.add(l1)));
                     if (sectionrenderdispatcher$rendersection1 != null && this.isInViewDistance(i, sectionrenderdispatcher$rendersection1.getSectionNode())) {
                         Direction direction = flag ? Direction.UP : Direction.DOWN;
                         SectionOcclusionGraph.Node sectionocclusiongraph$node = new SectionOcclusionGraph.Node(
@@ -249,15 +_,15 @@
 
     private void runUpdates(
         SectionOcclusionGraph.GraphStorage p_295507_,
-        Vec3 p_294343_,
+        me.alphamode.mcbig.core.BigVec3 p_294343_,
         Queue<SectionOcclusionGraph.Node> p_295598_,
         boolean p_295668_,
         Consumer<SectionRenderDispatcher.RenderSection> p_295393_,
-        LongOpenHashSet p_366468_
+        it.unimi.dsi.fastutil.objects.ObjectOpenHashSet<SectionPos> p_366468_
     ) {
         int i = 16;
-        BlockPos blockpos = new BlockPos(Mth.floor(p_294343_.x / 16.0) * 16, Mth.floor(p_294343_.y / 16.0) * 16, Mth.floor(p_294343_.z / 16.0) * 16);
-        long j = SectionPos.asLong(blockpos);
+        BlockPos blockpos = new BlockPos(Mth.bigFloor(p_294343_.x.divide(me.alphamode.mcbig.core.BigConstants.SIXTEEN)).multiply(me.alphamode.mcbig.core.BigConstants.Ints.SIXTEEN), Mth.bigFloor(p_294343_.y / 16.0).multiply(me.alphamode.mcbig.core.BigConstants.Ints.SIXTEEN), Mth.bigFloor(p_294343_.z.divide(me.alphamode.mcbig.core.BigConstants.SIXTEEN)).multiply(me.alphamode.mcbig.core.BigConstants.Ints.SIXTEEN));
+        SectionPos j = SectionPos.of(blockpos);
         BlockPos blockpos1 = blockpos.offset(8, 8, 8);
 
         while (!p_295598_.isEmpty()) {
@@ -302,15 +_,15 @@
                     if (p_295668_ && flag) {
                         BlockPos blockpos2 = sectionrenderdispatcher$rendersection1.getOrigin();
                         BlockPos blockpos3 = blockpos2.offset(
-                            (direction.getAxis() == Direction.Axis.X ? blockpos1.getX() <= blockpos2.getX() : blockpos1.getX() >= blockpos2.getX()) ? 0 : 16,
-                            (direction.getAxis() == Direction.Axis.Y ? blockpos1.getY() <= blockpos2.getY() : blockpos1.getY() >= blockpos2.getY()) ? 0 : 16,
-                            (direction.getAxis() == Direction.Axis.Z ? blockpos1.getZ() <= blockpos2.getZ() : blockpos1.getZ() >= blockpos2.getZ()) ? 0 : 16
+                            (direction.getAxis() == Direction.Axis.X ? blockpos1.getBigX().compareTo(blockpos2.getBigX()) <= 0 : blockpos1.getBigX().compareTo(blockpos2.getBigX()) >= 0) ? 0 : 16,
+                            (direction.getAxis() == Direction.Axis.Y ? blockpos1.getBigY().compareTo(blockpos2.getBigY()) <= 0 : blockpos1.getBigY().compareTo(blockpos2.getBigY()) >= 0) ? 0 : 16,
+                            (direction.getAxis() == Direction.Axis.Z ? blockpos1.getBigZ().compareTo(blockpos2.getBigZ()) <= 0 : blockpos1.getBigZ().compareTo(blockpos2.getBigZ()) >= 0) ? 0 : 16
                         );
-                        Vec3 vec31 = new Vec3((double)blockpos3.getX(), (double)blockpos3.getY(), (double)blockpos3.getZ());
-                        Vec3 vec3 = p_294343_.subtract(vec31).normalize().scale(CEILED_SECTION_DIAGONAL);
+                        me.alphamode.mcbig.core.BigVec3 vec31 = new me.alphamode.mcbig.core.BigVec3(blockpos3.getBigX().toBigDecimal(), (double)blockpos3.getBigY().doubleValue(), blockpos3.getBigZ().toBigDecimal());
+                        me.alphamode.mcbig.core.BigVec3 vec3 = p_294343_.subtract(vec31).normalize().scale(CEILED_SECTION_DIAGONAL);
                         boolean flag2 = true;
 
-                        while (p_294343_.subtract(vec31).lengthSqr() > 3600.0) {
+                        while(p_294343_.subtract(vec31).lengthSqr().compareTo(me.alphamode.mcbig.core.BigConstants.OCCLUSION) > 0) {
                             vec31 = vec31.add(vec3);
                             LevelHeightAccessor levelheightaccessor = this.viewArea.getLevelHeightAccessor();
                             if (vec31.y > (double)levelheightaccessor.getMaxY() || vec31.y < (double)levelheightaccessor.getMinY()) {
@@ -345,7 +_,7 @@
                         } else if (this.isInViewDistance(j, sectionrenderdispatcher$rendersection1.getSectionNode())) {
                             p_295507_.sectionToNodeMap.put(sectionrenderdispatcher$rendersection1, sectionocclusiongraph$node2);
                             p_295507_.chunksWaitingForNeighbors
-                                .computeIfAbsent(ChunkPos.asLong(sectionrenderdispatcher$rendersection1.getOrigin()), p_294377_ -> new ArrayList<>())
+                                .computeIfAbsent(new ChunkPos(sectionrenderdispatcher$rendersection1.getOrigin()), p_294377_ -> new ArrayList())
                                 .add(sectionrenderdispatcher$rendersection1);
                         }
                     }
@@ -354,19 +_,19 @@
         }
     }
 
-    private boolean isInViewDistance(long p_366544_, long p_366485_) {
+    private boolean isInViewDistance(SectionPos p_366544_, SectionPos p_366485_) {
         return ChunkTrackingView.isInViewDistance(
-            SectionPos.x(p_366544_), SectionPos.z(p_366544_), this.viewArea.getViewDistance(), SectionPos.x(p_366485_), SectionPos.z(p_366485_)
+            p_366544_.x(), p_366544_.z(), this.viewArea.getViewDistance(), p_366485_.x(), p_366485_.z()
         );
     }
 
     @Nullable
-    private SectionRenderDispatcher.RenderSection getRelativeFrom(long p_366669_, SectionRenderDispatcher.RenderSection p_295211_, Direction p_294491_) {
-        long i = p_295211_.getNeighborSectionNode(p_294491_);
+    private SectionRenderDispatcher.RenderSection getRelativeFrom(SectionPos p_366669_, SectionRenderDispatcher.RenderSection p_295211_, Direction p_294491_) {
+        SectionPos i = p_295211_.getNeighborSectionNode(p_294491_);
         if (!this.isInViewDistance(p_366669_, i)) {
             return null;
         } else {
-            return Mth.abs(SectionPos.y(p_366669_) - SectionPos.y(i)) > this.viewArea.getViewDistance() ? null : this.viewArea.getRenderSection(i);
+            return Mth.abs(p_366669_.y().intValue() - i.y().intValue()) > this.viewArea.getViewDistance() ? null : this.viewArea.getRenderSection(i);
         }
     }
 
@@ -381,9 +_,9 @@
     }
 
     @OnlyIn(Dist.CLIENT)
-    static record GraphEvents(LongSet chunksWhichReceivedNeighbors, BlockingQueue<SectionRenderDispatcher.RenderSection> sectionsToPropagateFrom) {
+    static record GraphEvents(it.unimi.dsi.fastutil.objects.ObjectSet<ChunkPos> chunksWhichReceivedNeighbors, BlockingQueue<SectionRenderDispatcher.RenderSection> sectionsToPropagateFrom) {
         GraphEvents() {
-            this(new LongOpenHashSet(), new LinkedBlockingQueue<>());
+            this(new it.unimi.dsi.fastutil.objects.ObjectOpenHashSet<>(), new LinkedBlockingQueue<>());
         }
     }
 
@@ -398,12 +_,12 @@
     static class GraphStorage {
         public final SectionOcclusionGraph.SectionToNodeMap sectionToNodeMap;
         public final Octree sectionTree;
-        public final Long2ObjectMap<List<SectionRenderDispatcher.RenderSection>> chunksWaitingForNeighbors;
+        public final it.unimi.dsi.fastutil.objects.Object2ObjectMap<ChunkPos, List<SectionRenderDispatcher.RenderSection>> chunksWaitingForNeighbors;
 
         public GraphStorage(ViewArea p_366828_) {
             this.sectionToNodeMap = new SectionOcclusionGraph.SectionToNodeMap(p_366828_.sections.length);
             this.sectionTree = new Octree(p_366828_.getCameraSectionPos(), p_366828_.getViewDistance(), p_366828_.sectionGridSizeY, p_366828_.level.getMinY());
-            this.chunksWaitingForNeighbors = new Long2ObjectOpenHashMap<>();
+            this.chunksWaitingForNeighbors = new it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap<>();
         }
     }
 
@@ -449,7 +_,7 @@
 
         @Override
         public int hashCode() {
-            return Long.hashCode(this.section.getSectionNode());
+            return Long.hashCode(this.section.getSectionNode().asLong());
         }
 
         @Override
