--- a/net/minecraft/client/renderer/ViewArea.java
+++ b/net/minecraft/client/renderer/ViewArea.java
@@ -40,7 +_,7 @@
                 for (int k = 0; k < this.sectionGridSizeY; k++) {
                     for (int l = 0; l < this.sectionGridSizeZ; l++) {
                         int i1 = this.getSectionIndex(j, k, l);
-                        this.sections[i1] = p_294825_.new RenderSection(i1, SectionPos.asLong(j, k + this.level.getMinSectionY(), l));
+                        this.sections[i1] = p_294825_.new RenderSection(i1, SectionPos.of(j, k + this.level.getMinSectionY(), l));
                     }
                 }
             }
@@ -75,19 +_,19 @@
 
     public void repositionCamera(SectionPos p_366723_) {
         for (int i = 0; i < this.sectionGridSizeX; i++) {
-            int j = p_366723_.x() - this.viewDistance;
-            int k = j + Math.floorMod(i - j, this.sectionGridSizeX);
+            me.alphamode.mcbig.math.BigInteger j = p_366723_.x().subtract(this.viewDistance);
+            me.alphamode.mcbig.math.BigInteger k = j.add(me.alphamode.mcbig.math.BigMath.floorMod(me.alphamode.mcbig.math.BigInteger.val(i).subtract(j), this.sectionGridSizeX));
 
             for (int l = 0; l < this.sectionGridSizeZ; l++) {
-                int i1 = p_366723_.z() - this.viewDistance;
-                int j1 = i1 + Math.floorMod(l - i1, this.sectionGridSizeZ);
+                me.alphamode.mcbig.math.BigInteger i1 = p_366723_.z().subtract(this.viewDistance);
+                me.alphamode.mcbig.math.BigInteger j1 = i1.add(me.alphamode.mcbig.math.BigMath.floorMod(me.alphamode.mcbig.math.BigInteger.val(l).subtract(i1), this.sectionGridSizeZ));
 
                 for (int k1 = 0; k1 < this.sectionGridSizeY; k1++) {
                     int l1 = this.level.getMinSectionY() + k1;
                     SectionRenderDispatcher.RenderSection sectionrenderdispatcher$rendersection = this.sections[this.getSectionIndex(i, k1, l)];
-                    long i2 = sectionrenderdispatcher$rendersection.getSectionNode();
-                    if (i2 != SectionPos.asLong(k, l1, j1)) {
-                        sectionrenderdispatcher$rendersection.setSectionNode(SectionPos.asLong(k, l1, j1));
+                    SectionPos i2 = sectionrenderdispatcher$rendersection.getSectionNode();
+                    if (!i2.equals(SectionPos.of(k, l1, j1))) {
+                        sectionrenderdispatcher$rendersection.setSectionNode(SectionPos.of(k, l1, j1));
                     }
                 }
             }
@@ -101,7 +_,7 @@
         return this.cameraSectionPos;
     }
 
-    public void setDirty(int p_110860_, int p_110861_, int p_110862_, boolean p_110863_) {
+    public void setDirty(me.alphamode.mcbig.math.BigInteger p_110860_, int p_110861_, me.alphamode.mcbig.math.BigInteger p_110862_, boolean p_110863_) {
         SectionRenderDispatcher.RenderSection sectionrenderdispatcher$rendersection = this.getRenderSection(p_110860_, p_110861_, p_110862_);
         if (sectionrenderdispatcher$rendersection != null) {
             sectionrenderdispatcher$rendersection.setDirty(p_110863_);
@@ -110,34 +_,34 @@
 
     @Nullable
     protected SectionRenderDispatcher.RenderSection getRenderSectionAt(BlockPos p_294239_) {
-        return this.getRenderSection(SectionPos.asLong(p_294239_));
-    }
-
-    @Nullable
-    protected SectionRenderDispatcher.RenderSection getRenderSection(long p_366845_) {
-        int i = SectionPos.x(p_366845_);
-        int j = SectionPos.y(p_366845_);
-        int k = SectionPos.z(p_366845_);
-        return this.getRenderSection(i, j, k);
-    }
-
-    @Nullable
-    private SectionRenderDispatcher.RenderSection getRenderSection(int p_366548_, int p_366588_, int p_366508_) {
+        return this.getRenderSection(SectionPos.of(p_294239_));
+    }
+
+    @Nullable
+    protected SectionRenderDispatcher.RenderSection getRenderSection(SectionPos p_366845_) {
+//        int i = SectionPos.x(p_366845_);
+//        int j = SectionPos.y(p_366845_);
+//        int k = SectionPos.z(p_366845_);
+        return this.getRenderSection(p_366845_.x(), p_366845_.getY(), p_366845_.z());
+    }
+
+    @Nullable
+    private SectionRenderDispatcher.RenderSection getRenderSection(me.alphamode.mcbig.math.BigInteger p_366548_, int p_366588_, me.alphamode.mcbig.math.BigInteger p_366508_) {
         if (!this.containsSection(p_366548_, p_366588_, p_366508_)) {
             return null;
         } else {
             int i = p_366588_ - this.level.getMinSectionY();
-            int j = Math.floorMod(p_366548_, this.sectionGridSizeX);
-            int k = Math.floorMod(p_366508_, this.sectionGridSizeZ);
-            return this.sections[this.getSectionIndex(j, i, k)];
+            me.alphamode.mcbig.math.BigInteger j = me.alphamode.mcbig.math.BigMath.floorMod(p_366548_, this.sectionGridSizeX);
+            me.alphamode.mcbig.math.BigInteger k = me.alphamode.mcbig.math.BigMath.floorMod(p_366508_, this.sectionGridSizeZ);
+            return this.sections[this.getSectionIndex(j.intValue(), i, k.intValue())];
         }
     }
 
-    private boolean containsSection(int p_366483_, int p_366902_, int p_366901_) {
+    private boolean containsSection(me.alphamode.mcbig.math.BigInteger p_366483_, int p_366902_, me.alphamode.mcbig.math.BigInteger p_366901_) {
         if (p_366902_ >= this.level.getMinSectionY() && p_366902_ <= this.level.getMaxSectionY()) {
-            return p_366483_ < this.cameraSectionPos.x() - this.viewDistance || p_366483_ > this.cameraSectionPos.x() + this.viewDistance
+            return p_366483_.compareTo(this.cameraSectionPos.x().subtract(this.viewDistance)) < 0 || p_366483_.compareTo(this.cameraSectionPos.x().add(this.viewDistance)) > 0
                 ? false
-                : p_366901_ >= this.cameraSectionPos.z() - this.viewDistance && p_366901_ <= this.cameraSectionPos.z() + this.viewDistance;
+                : p_366901_.compareTo(this.cameraSectionPos.z().subtract(this.viewDistance)) >= 0 && p_366901_.compareTo(this.cameraSectionPos.z().add(this.viewDistance)) <= 0;
         } else {
             return false;
         }
