--- a/net/minecraft/commands/arguments/selector/EntitySelector.java
+++ b/net/minecraft/commands/arguments/selector/EntitySelector.java
@@ -25,7 +_,7 @@
 
 public class EntitySelector {
     public static final int INFINITE = Integer.MAX_VALUE;
-    public static final BiConsumer<Vec3, List<? extends Entity>> ORDER_ARBITRARY = (p_261404_, p_261405_) -> {
+    public static final BiConsumer<me.alphamode.mcbig.core.BigVec3, List<? extends Entity>> ORDER_ARBITRARY = (p_261404_, p_261405_) -> {
     };
     private static final EntityTypeTest<Entity, ?> ANY_TYPE = new EntityTypeTest<Entity, Entity>() {
         public Entity tryCast(Entity p_175109_) {
@@ -42,10 +_,10 @@
     private final boolean worldLimited;
     private final List<Predicate<Entity>> contextFreePredicates;
     private final MinMaxBounds.Doubles range;
-    private final Function<Vec3, Vec3> position;
+    private final Function<me.alphamode.mcbig.core.BigVec3, me.alphamode.mcbig.core.BigVec3> position;
     @Nullable
     private final AABB aabb;
-    private final BiConsumer<Vec3, List<? extends Entity>> order;
+    private final BiConsumer<me.alphamode.mcbig.core.BigVec3, List<? extends Entity>> order;
     private final boolean currentEntity;
     @Nullable
     private final String playerName;
@@ -60,9 +_,9 @@
         boolean p_121127_,
         List<Predicate<Entity>> p_352913_,
         MinMaxBounds.Doubles p_121129_,
-        Function<Vec3, Vec3> p_121130_,
+        Function<me.alphamode.mcbig.core.BigVec3, me.alphamode.mcbig.core.BigVec3> p_121130_,
         @Nullable AABB p_121131_,
-        BiConsumer<Vec3, List<? extends Entity>> p_121132_,
+        BiConsumer<me.alphamode.mcbig.core.BigVec3, List<? extends Entity>> p_121132_,
         boolean p_121133_,
         @Nullable String p_121134_,
         @Nullable UUID p_121135_,
@@ -142,7 +_,7 @@
 
             return List.of();
         } else {
-            Vec3 vec3 = this.position.apply(p_121161_.getPosition());
+            me.alphamode.mcbig.core.BigVec3 vec3 = this.position.apply(p_121161_.getPosition());
             AABB aabb = this.getAbsoluteAabb(vec3);
             if (this.currentEntity) {
                 Predicate<Entity> predicate1 = this.getPredicate(vec3, aabb, null);
@@ -197,7 +_,7 @@
             ServerPlayer serverplayer1 = p_121167_.getServer().getPlayerList().getPlayer(this.entityUUID);
             return serverplayer1 == null ? List.of() : List.of(serverplayer1);
         } else {
-            Vec3 vec3 = this.position.apply(p_121167_.getPosition());
+            me.alphamode.mcbig.core.BigVec3 vec3 = this.position.apply(p_121167_.getPosition());
             AABB aabb = this.getAbsoluteAabb(vec3);
             Predicate<Entity> predicate = this.getPredicate(vec3, aabb, null);
             if (this.currentEntity) {
@@ -230,11 +_,11 @@
     }
 
     @Nullable
-    private AABB getAbsoluteAabb(Vec3 p_352924_) {
-        return this.aabb != null ? this.aabb.move(p_352924_) : null;
+    private AABB getAbsoluteAabb(me.alphamode.mcbig.core.BigVec3 p_352924_) {
+        return this.aabb != null ? this.aabb.move(p_352924_.toVanilla()) : null;
     }
 
-    private Predicate<Entity> getPredicate(Vec3 p_121145_, @Nullable AABB p_352908_, @Nullable FeatureFlagSet p_352911_) {
+    private Predicate<Entity> getPredicate(me.alphamode.mcbig.core.BigVec3 p_121145_, @Nullable AABB p_352908_, @Nullable FeatureFlagSet p_352911_) {
         boolean flag = p_352911_ != null;
         boolean flag1 = p_352908_ != null;
         boolean flag2 = !this.range.isAny();
@@ -263,7 +_,7 @@
         return Util.allOf(list);
     }
 
-    private <T extends Entity> List<T> sortAndLimit(Vec3 p_121150_, List<T> p_121151_) {
+    private <T extends Entity> List<T> sortAndLimit(me.alphamode.mcbig.core.BigVec3 p_121150_, List<T> p_121151_) {
         if (p_121151_.size() > 1) {
             this.order.accept(p_121150_, p_121151_);
         }
