--- a/net/minecraft/world/Container.java
+++ b/net/minecraft/world/Container.java
@@ -87,7 +_,7 @@
         } else if (level.getBlockEntity(blockpos) != p_272877_) {
             return false;
         } else {
-            return p_272670_.distanceToSqr((double)blockpos.getX() + 0.5, (double)blockpos.getY() + 0.5, (double)blockpos.getZ() + 0.5)
+            return p_272670_.distanceToSqr((double)blockpos.getBigX().doubleValue() + 0.5, (double)blockpos.getBigY().doubleValue() + 0.5, (double)blockpos.getBigZ().doubleValue() + 0.5)
                 <= (double)(p_273411_ * p_273411_);
         }
     }
