--- a/net/minecraft/world/level/levelgen/placement/PlacementContext.java
+++ b/net/minecraft/world/level/levelgen/placement/PlacementContext.java
@@ -23,7 +_,7 @@
         this.topFeature = p_191820_;
     }
 
-    public int getHeight(Heightmap.Types p_191825_, int p_191826_, int p_191827_) {
+    public int getHeight(Heightmap.Types p_191825_, me.alphamode.mcbig.math.BigInteger p_191826_, me.alphamode.mcbig.math.BigInteger p_191827_) {
         return this.level.getHeight(p_191825_, p_191826_, p_191827_);
     }
 
