--- a/net/minecraft/network/protocol/game/ClientboundLevelChunkWithLightPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundLevelChunkWithLightPacket.java
@@ -14,8 +_,8 @@
     public static final StreamCodec<RegistryFriendlyByteBuf, ClientboundLevelChunkWithLightPacket> STREAM_CODEC = Packet.codec(
         ClientboundLevelChunkWithLightPacket::write, ClientboundLevelChunkWithLightPacket::new
     );
-    private final int x;
-    private final int z;
+    private final me.alphamode.mcbig.math.BigInteger x;
+    private final me.alphamode.mcbig.math.BigInteger z;
     private final ClientboundLevelChunkPacketData chunkData;
     private final ClientboundLightUpdatePacketData lightData;
 
@@ -28,15 +_,15 @@
     }
 
     private ClientboundLevelChunkWithLightPacket(RegistryFriendlyByteBuf p_319989_) {
-        this.x = p_319989_.readInt();
-        this.z = p_319989_.readInt();
+        this.x = p_319989_.readBigInteger();
+        this.z = p_319989_.readBigInteger();
         this.chunkData = new ClientboundLevelChunkPacketData(p_319989_, this.x, this.z);
         this.lightData = new ClientboundLightUpdatePacketData(p_319989_, this.x, this.z);
     }
 
     private void write(RegistryFriendlyByteBuf p_320036_) {
-        p_320036_.writeInt(this.x);
-        p_320036_.writeInt(this.z);
+        p_320036_.writeBigInteger(this.x);
+        p_320036_.writeBigInteger(this.z);
         this.chunkData.write(p_320036_);
         this.lightData.write(p_320036_);
     }
@@ -50,11 +_,11 @@
         p_195716_.handleLevelChunkWithLight(this);
     }
 
-    public int getX() {
+    public me.alphamode.mcbig.math.BigInteger getX() {
         return this.x;
     }
 
-    public int getZ() {
+    public me.alphamode.mcbig.math.BigInteger getZ() {
         return this.z;
     }
 
