--- a/net/minecraft/network/protocol/game/ClientboundSetChunkCacheCenterPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundSetChunkCacheCenterPacket.java
@@ -9,22 +_,22 @@
     public static final StreamCodec<FriendlyByteBuf, ClientboundSetChunkCacheCenterPacket> STREAM_CODEC = Packet.codec(
         ClientboundSetChunkCacheCenterPacket::write, ClientboundSetChunkCacheCenterPacket::new
     );
-    private final int x;
-    private final int z;
+    private final me.alphamode.mcbig.math.BigInteger x;
+    private final me.alphamode.mcbig.math.BigInteger z;
 
-    public ClientboundSetChunkCacheCenterPacket(int p_133086_, int p_133087_) {
+    public ClientboundSetChunkCacheCenterPacket(me.alphamode.mcbig.math.BigInteger p_133086_, me.alphamode.mcbig.math.BigInteger p_133087_) {
         this.x = p_133086_;
         this.z = p_133087_;
     }
 
     private ClientboundSetChunkCacheCenterPacket(FriendlyByteBuf p_179282_) {
-        this.x = p_179282_.readVarInt();
-        this.z = p_179282_.readVarInt();
+        this.x = p_179282_.readBigInteger();
+        this.z = p_179282_.readBigInteger();
     }
 
     private void write(FriendlyByteBuf p_133096_) {
-        p_133096_.writeVarInt(this.x);
-        p_133096_.writeVarInt(this.z);
+        p_133096_.writeBigInteger(this.x);
+        p_133096_.writeBigInteger(this.z);
     }
 
     @Override
@@ -36,11 +_,11 @@
         p_133093_.handleSetChunkCacheCenter(this);
     }
 
-    public int getX() {
+    public me.alphamode.mcbig.math.BigInteger getX() {
         return this.x;
     }
 
-    public int getZ() {
+    public me.alphamode.mcbig.math.BigInteger getZ() {
         return this.z;
     }
 }
