--- a/net/minecraft/network/protocol/game/ClientboundLightUpdatePacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundLightUpdatePacket.java
@@ -13,25 +_,25 @@
     public static final StreamCodec<FriendlyByteBuf, ClientboundLightUpdatePacket> STREAM_CODEC = Packet.codec(
         ClientboundLightUpdatePacket::write, ClientboundLightUpdatePacket::new
     );
-    private final int x;
-    private final int z;
+    private final me.alphamode.mcbig.math.BigInteger x;
+    private final me.alphamode.mcbig.math.BigInteger z;
     private final ClientboundLightUpdatePacketData lightData;
 
     public ClientboundLightUpdatePacket(ChunkPos p_285255_, LevelLightEngine p_285409_, @Nullable BitSet p_285387_, @Nullable BitSet p_285074_) {
-        this.x = p_285255_.x;
-        this.z = p_285255_.z;
+        this.x = p_285255_.x();
+        this.z = p_285255_.z();
         this.lightData = new ClientboundLightUpdatePacketData(p_285255_, p_285409_, p_285387_, p_285074_);
     }
 
     private ClientboundLightUpdatePacket(FriendlyByteBuf p_178918_) {
-        this.x = p_178918_.readVarInt();
-        this.z = p_178918_.readVarInt();
+        this.x = p_178918_.readBigInteger();
+        this.z = p_178918_.readBigInteger();
         this.lightData = new ClientboundLightUpdatePacketData(p_178918_, this.x, this.z);
     }
 
     private void write(FriendlyByteBuf p_132351_) {
-        p_132351_.writeVarInt(this.x);
-        p_132351_.writeVarInt(this.z);
+        p_132351_.writeBigInteger(this.x);
+        p_132351_.writeBigInteger(this.z);
         this.lightData.write(p_132351_);
     }
 
@@ -44,11 +_,11 @@
         p_132348_.handleLightUpdatePacket(this);
     }
 
-    public int getX() {
+    public me.alphamode.mcbig.math.BigInteger getX() {
         return this.x;
     }
 
-    public int getZ() {
+    public me.alphamode.mcbig.math.BigInteger getZ() {
         return this.z;
     }
 
