--- a/net/minecraft/network/protocol/game/ServerboundMovePlayerPacket.java
+++ b/net/minecraft/network/protocol/game/ServerboundMovePlayerPacket.java
@@ -8,9 +_,9 @@
 public abstract class ServerboundMovePlayerPacket implements Packet<ServerGamePacketListener> {
     private static final int FLAG_ON_GROUND = 1;
     private static final int FLAG_HORIZONTAL_COLLISION = 2;
-    protected final double x;
+    protected final me.alphamode.mcbig.math.BigDecimal x;
     protected final double y;
-    protected final double z;
+    protected final me.alphamode.mcbig.math.BigDecimal z;
     protected final float yRot;
     protected final float xRot;
     protected final boolean onGround;
@@ -40,9 +_,9 @@
     }
 
     protected ServerboundMovePlayerPacket(
-        double p_179675_,
+        me.alphamode.mcbig.math.BigDecimal p_179675_,
         double p_179676_,
-        double p_179677_,
+        me.alphamode.mcbig.math.BigDecimal p_179677_,
         float p_179678_,
         float p_179679_,
         boolean p_179680_,
@@ -68,7 +_,7 @@
         p_134138_.handleMovePlayer(this);
     }
 
-    public double getX(double p_134130_) {
+    public me.alphamode.mcbig.math.BigDecimal getX(me.alphamode.mcbig.math.BigDecimal p_134130_) {
         return this.hasPos ? this.x : p_134130_;
     }
 
@@ -76,7 +_,7 @@
         return this.hasPos ? this.y : p_134141_;
     }
 
-    public double getZ(double p_134147_) {
+    public me.alphamode.mcbig.math.BigDecimal getZ(me.alphamode.mcbig.math.BigDecimal p_134147_) {
         return this.hasPos ? this.z : p_134147_;
     }
 
@@ -109,14 +_,14 @@
             ServerboundMovePlayerPacket.Pos::write, ServerboundMovePlayerPacket.Pos::read
         );
 
-        public Pos(double p_134150_, double p_134151_, double p_134152_, boolean p_134153_, boolean p_361686_) {
+        public Pos(me.alphamode.mcbig.math.BigDecimal p_134150_, double p_134151_, me.alphamode.mcbig.math.BigDecimal p_134152_, boolean p_134153_, boolean p_361686_) {
             super(p_134150_, p_134151_, p_134152_, 0.0F, 0.0F, p_134153_, p_361686_, true, false);
         }
 
         private static ServerboundMovePlayerPacket.Pos read(FriendlyByteBuf p_179686_) {
-            double d0 = p_179686_.readDouble();
+            me.alphamode.mcbig.math.BigDecimal d0 = p_179686_.readBigDecimal();
             double d1 = p_179686_.readDouble();
-            double d2 = p_179686_.readDouble();
+            me.alphamode.mcbig.math.BigDecimal d2 = p_179686_.readBigDecimal();
             short short1 = p_179686_.readUnsignedByte();
             boolean flag = ServerboundMovePlayerPacket.unpackOnGround(short1);
             boolean flag1 = ServerboundMovePlayerPacket.unpackHorizontalCollision(short1);
@@ -124,9 +_,9 @@
         }
 
         private void write(FriendlyByteBuf p_134159_) {
-            p_134159_.writeDouble(this.x);
+            p_134159_.writeBigDecimal(this.x);
             p_134159_.writeDouble(this.y);
-            p_134159_.writeDouble(this.z);
+            p_134159_.writeBigDecimal(this.z);
             p_134159_.writeByte(ServerboundMovePlayerPacket.packFlags(this.onGround, this.horizontalCollision));
         }
 
@@ -141,14 +_,14 @@
             ServerboundMovePlayerPacket.PosRot::write, ServerboundMovePlayerPacket.PosRot::read
         );
 
-        public PosRot(double p_134162_, double p_134163_, double p_134164_, float p_134165_, float p_134166_, boolean p_134167_, boolean p_362765_) {
+        public PosRot(me.alphamode.mcbig.math.BigDecimal p_134162_, double p_134163_, me.alphamode.mcbig.math.BigDecimal p_134164_, float p_134165_, float p_134166_, boolean p_134167_, boolean p_362765_) {
             super(p_134162_, p_134163_, p_134164_, p_134165_, p_134166_, p_134167_, p_362765_, true, true);
         }
 
         private static ServerboundMovePlayerPacket.PosRot read(FriendlyByteBuf p_179688_) {
-            double d0 = p_179688_.readDouble();
+            me.alphamode.mcbig.math.BigDecimal d0 = p_179688_.readBigDecimal();
             double d1 = p_179688_.readDouble();
-            double d2 = p_179688_.readDouble();
+            me.alphamode.mcbig.math.BigDecimal d2 = p_179688_.readBigDecimal();
             float f = p_179688_.readFloat();
             float f1 = p_179688_.readFloat();
             short short1 = p_179688_.readUnsignedByte();
@@ -158,9 +_,9 @@
         }
 
         private void write(FriendlyByteBuf p_134173_) {
-            p_134173_.writeDouble(this.x);
+            p_134173_.writeBigDecimal(this.x);
             p_134173_.writeDouble(this.y);
-            p_134173_.writeDouble(this.z);
+            p_134173_.writeBigDecimal(this.z);
             p_134173_.writeFloat(this.yRot);
             p_134173_.writeFloat(this.xRot);
             p_134173_.writeByte(ServerboundMovePlayerPacket.packFlags(this.onGround, this.horizontalCollision));
@@ -178,7 +_,7 @@
         );
 
         public Rot(float p_134176_, float p_134177_, boolean p_134178_, boolean p_363477_) {
-            super(0.0, 0.0, 0.0, p_134176_, p_134177_, p_134178_, p_363477_, false, true);
+            super(me.alphamode.mcbig.math.BigDecimal.ZERO, 0.0, me.alphamode.mcbig.math.BigDecimal.ZERO, p_134176_, p_134177_, p_134178_, p_363477_, false, true);
         }
 
         private static ServerboundMovePlayerPacket.Rot read(FriendlyByteBuf p_179690_) {
@@ -208,7 +_,7 @@
         );
 
         public StatusOnly(boolean p_179692_, boolean p_363234_) {
-            super(0.0, 0.0, 0.0, 0.0F, 0.0F, p_179692_, p_363234_, false, false);
+            super(me.alphamode.mcbig.math.BigDecimal.ZERO, 0.0, me.alphamode.mcbig.math.BigDecimal.ZERO, 0.0F, 0.0F, p_179692_, p_363234_, false, false);
         }
 
         private static ServerboundMovePlayerPacket.StatusOnly read(FriendlyByteBuf p_179698_) {
