--- a/net/minecraft/world/entity/raid/Raid.java
+++ b/net/minecraft/world/entity/raid/Raid.java
@@ -541,7 +_,7 @@
             p_37715_.setCanJoinRaid(true);
             p_37715_.setTicksOutsideRaid(0);
             if (!p_37717_ && p_37716_ != null) {
-                p_37715_.setPos((double)p_37716_.getX() + 0.5, (double)p_37716_.getY() + 1.0, (double)p_37716_.getZ() + 0.5);
+                p_37715_.setPos((double)p_37716_.getBigX().doubleValue() + 0.5, (double)p_37716_.getBigY().doubleValue() + 1.0, (double)p_37716_.getBigZ().doubleValue() + 0.5);
                 p_37715_.finalizeSpawn(this.level, this.level.getCurrentDifficultyAt(p_37716_), EntitySpawnReason.EVENT, null);
                 p_37715_.applyRaidBuffs(this.level, p_37714_, false);
                 p_37715_.setOnGround(true);
@@ -636,10 +_,10 @@
                     int j1 = 10;
                     if (this.level
                             .hasChunksAt(
-                                blockpos$mutableblockpos.getX() - 10,
-                                blockpos$mutableblockpos.getZ() - 10,
-                                blockpos$mutableblockpos.getX() + 10,
-                                blockpos$mutableblockpos.getZ() + 10
+                                blockpos$mutableblockpos.getBigX().subtract(10),
+                                blockpos$mutableblockpos.getBigZ().subtract(10),
+                                blockpos$mutableblockpos.getBigX().add(10),
+                                blockpos$mutableblockpos.getBigZ().add(10)
                             )
                         && this.level.isPositionEntityTicking(blockpos$mutableblockpos)
                         && (
