--- a/net/minecraft/world/entity/raid/Raid.java
+++ b/net/minecraft/world/entity/raid/Raid.java
@@ -541,7 +_,7 @@
             p_37715_.setCanJoinRaid(true);
             p_37715_.setTicksOutsideRaid(0);
             if (!p_37717_ && p_37716_ != null) {
-                p_37715_.setPos((double)p_37716_.getX() + 0.5, (double)p_37716_.getY() + 1.0, (double)p_37716_.getZ() + 0.5);
+                p_37715_.setPos((double)p_37716_.getBigX().doubleValue() + 0.5, (double)p_37716_.getBigY().doubleValue() + 1.0, (double)p_37716_.getBigZ().doubleValue() + 0.5);
                 p_37715_.finalizeSpawn(this.level, this.level.getCurrentDifficultyAt(p_37716_), MobSpawnType.EVENT, null, null);
                 p_37715_.applyRaidBuffs(p_37714_, false);
                 p_37715_.setOnGround(true);
@@ -626,18 +_,18 @@
 
         for(int i1 = 0; i1 < p_37709_; ++i1) {
             float f = this.level.random.nextFloat() * (float) (Math.PI * 2);
-            int j = this.center.getX() + Mth.floor(Mth.cos(f) * 32.0F * (float)i) + this.level.random.nextInt(5);
-            int l = this.center.getZ() + Mth.floor(Mth.sin(f) * 32.0F * (float)i) + this.level.random.nextInt(5);
+            me.alphamode.mcbig.math.BigInteger j = this.center.getBigX().add(Mth.floor(Mth.cos(f) * 32.0F * (float)i) + this.level.random.nextInt(5));
+            me.alphamode.mcbig.math.BigInteger l = this.center.getBigZ().add(Mth.floor(Mth.sin(f) * 32.0F * (float)i) + this.level.random.nextInt(5));
             int k = this.level.getHeight(Heightmap.Types.WORLD_SURFACE, j, l);
             blockpos$mutableblockpos.set(j, k, l);
             if (!this.level.isVillage(blockpos$mutableblockpos) || p_37708_ >= 2) {
                 int j1 = 10;
                 if (this.level
                         .hasChunksAt(
-                            blockpos$mutableblockpos.getX() - 10,
-                            blockpos$mutableblockpos.getZ() - 10,
-                            blockpos$mutableblockpos.getX() + 10,
-                            blockpos$mutableblockpos.getZ() + 10
+                            blockpos$mutableblockpos.getBigX().subtract(me.alphamode.mcbig.core.BigConstants.Ints.TEN),
+                            blockpos$mutableblockpos.getBigZ().subtract(me.alphamode.mcbig.core.BigConstants.Ints.TEN),
+                            blockpos$mutableblockpos.getBigX().add(me.alphamode.mcbig.core.BigConstants.Ints.TEN),
+                            blockpos$mutableblockpos.getBigZ().add(me.alphamode.mcbig.core.BigConstants.Ints.TEN)
                         )
                     && this.level.isPositionEntityTicking(blockpos$mutableblockpos)
                     && (
