--- a/net/minecraft/world/level/levelgen/PatrolSpawner.java
+++ b/net/minecraft/world/level/levelgen/PatrolSpawner.java
@@ -52,10 +_,10 @@
                             BlockPos.MutableBlockPos blockpos$mutableblockpos = player.blockPosition().mutable().move(k, 0, l);
                             int i1 = 10;
                             if (!p_64570_.hasChunksAt(
-                                blockpos$mutableblockpos.getX() - 10,
-                                blockpos$mutableblockpos.getZ() - 10,
-                                blockpos$mutableblockpos.getX() + 10,
-                                blockpos$mutableblockpos.getZ() + 10
+                                blockpos$mutableblockpos.getBigX().subtract(me.alphamode.mcbig.core.BigConstants.Ints.TEN),
+                                blockpos$mutableblockpos.getBigZ().subtract(me.alphamode.mcbig.core.BigConstants.Ints.TEN),
+                                blockpos$mutableblockpos.getBigX().add(me.alphamode.mcbig.core.BigConstants.Ints.TEN),
+                                blockpos$mutableblockpos.getBigZ().add(me.alphamode.mcbig.core.BigConstants.Ints.TEN)
                             )) {
                                 return 0;
                             } else {
@@ -79,8 +_,8 @@
                                             this.spawnPatrolMember(p_64570_, blockpos$mutableblockpos, randomsource, false);
                                         }
 
-                                        blockpos$mutableblockpos.setX(blockpos$mutableblockpos.getX() + randomsource.nextInt(5) - randomsource.nextInt(5));
-                                        blockpos$mutableblockpos.setZ(blockpos$mutableblockpos.getZ() + randomsource.nextInt(5) - randomsource.nextInt(5));
+                                        blockpos$mutableblockpos.setX(blockpos$mutableblockpos.getBigX().add(randomsource.nextInt(5)).subtract(randomsource.nextInt(5)));
+                                        blockpos$mutableblockpos.setZ(blockpos$mutableblockpos.getBigZ().add(randomsource.nextInt(5)).subtract(randomsource.nextInt(5)));
                                     }
 
                                     return j1;
@@ -107,7 +_,7 @@
                     patrollingmonster.findPatrolTarget();
                 }
 
-                patrollingmonster.setPos((double)p_224534_.getX(), (double)p_224534_.getY(), (double)p_224534_.getZ());
+                patrollingmonster.setPos((double)p_224534_.getBigX().doubleValue(), (double)p_224534_.getBigY().doubleValue(), (double)p_224534_.getBigZ().doubleValue());
                 patrollingmonster.finalizeSpawn(p_224533_, p_224533_.getCurrentDifficultyAt(p_224534_), MobSpawnType.PATROL, null, null);
                 p_224533_.addFreshEntityWithPassengers(patrollingmonster);
                 return true;
