--- a/net/minecraft/server/players/PlayerList.java
+++ b/net/minecraft/server/players/PlayerList.java
@@ -219,7 +_,7 @@
         }
 
         this.broadcastSystemMessage(mutablecomponent.withStyle(ChatFormatting.YELLOW), false);
-        servergamepacketlistenerimpl.teleport(p_11263_.getX(), p_11263_.getY(), p_11263_.getZ(), p_11263_.getYRot(), p_11263_.getXRot());
+        servergamepacketlistenerimpl.teleport(p_11263_.getBigX(), p_11263_.getY(), p_11263_.getBigZ(), p_11263_.getYRot(), p_11263_.getXRot());
         ServerStatus serverstatus = this.server.getStatus();
         if (serverstatus != null && !p_301988_.transferred()) {
             p_11263_.sendServerStatus(serverstatus);
@@ -430,7 +_,7 @@
             serverplayer.addTag(s);
         }
 
-        Vec3 vec3 = teleporttransition.position();
+        me.alphamode.mcbig.core.BigVec3 vec3 = teleporttransition.position();
         serverplayer.moveTo(vec3.x, vec3.y, vec3.z, teleporttransition.yRot(), teleporttransition.xRot());
         if (teleporttransition.missingRespawnBlock()) {
             serverplayer.connection.send(new ClientboundGameEventPacket(ClientboundGameEventPacket.NO_RESPAWN_BLOCK_AVAILABLE, 0.0F));
@@ -440,7 +_,7 @@
         ServerLevel serverlevel1 = serverplayer.serverLevel();
         LevelData leveldata = serverlevel1.getLevelData();
         serverplayer.connection.send(new ClientboundRespawnPacket(serverplayer.createCommonSpawnInfo(serverlevel1), b0));
-        serverplayer.connection.teleport(serverplayer.getX(), serverplayer.getY(), serverplayer.getZ(), serverplayer.getYRot(), serverplayer.getXRot());
+        serverplayer.connection.teleport(serverplayer.getBigX(), serverplayer.getY(), serverplayer.getBigZ(), serverplayer.getYRot(), serverplayer.getXRot());
         serverplayer.connection.send(new ClientboundSetDefaultSpawnPositionPacket(serverlevel.getSharedSpawnPos(), serverlevel.getSharedSpawnAngle()));
         serverplayer.connection.send(new ClientboundChangeDifficultyPacket(leveldata.getDifficulty(), leveldata.isDifficultyLocked()));
         serverplayer.connection
