--- a/net/minecraft/world/item/BoatItem.java
+++ b/net/minecraft/world/item/BoatItem.java
@@ -79,7 +_,7 @@
     private AbstractBoat getBoat(Level p_220017_, HitResult p_220018_, ItemStack p_305852_, Player p_306127_) {
         AbstractBoat abstractboat = this.entityType.create(p_220017_, EntitySpawnReason.SPAWN_ITEM_USE);
         if (abstractboat != null) {
-            Vec3 vec3 = p_220018_.getLocation();
+            Vec3 vec3 = p_220018_.getLocation().toVanilla();
             abstractboat.setInitialPos(vec3.x, vec3.y, vec3.z);
             if (p_220017_ instanceof ServerLevel serverlevel) {
                 EntityType.<AbstractBoat>createDefaultStackConfig(serverlevel, p_305852_, p_306127_).accept(abstractboat);
