--- a/net/minecraft/world/item/HangingEntityItem.java
+++ b/net/minecraft/world/item/HangingEntityItem.java
@@ -68,7 +_,7 @@
             if (hangingentity.survives()) {
                 if (!level.isClientSide) {
                     hangingentity.playPlacementSound();
-                    level.gameEvent(player, GameEvent.ENTITY_PLACE, hangingentity.position());
+                    level.gameEvent(player, GameEvent.ENTITY_PLACE, hangingentity.bigPosition());
                     level.addFreshEntity(hangingentity);
                 }
 
