--- a/net/minecraft/world/item/MapItem.java
+++ b/net/minecraft/world/item/MapItem.java
@@ -43,7 +_,7 @@
         super(p_42847_);
     }
 
-    public static ItemStack create(Level p_42887_, int p_42888_, int p_42889_, byte p_42890_, boolean p_42891_, boolean p_42892_) {
+    public static ItemStack create(Level p_42887_, me.alphamode.mcbig.math.BigInteger p_42888_, me.alphamode.mcbig.math.BigInteger p_42889_, byte p_42890_, boolean p_42891_, boolean p_42892_) {
         ItemStack itemstack = new ItemStack(Items.FILLED_MAP);
         MapId mapid = createNewSavedData(p_42887_, p_42888_, p_42889_, p_42890_, p_42891_, p_42892_, p_42887_.dimension());
         itemstack.set(DataComponents.MAP_ID, mapid);
@@ -62,9 +_,9 @@
     }
 
     private static MapId createNewSavedData(
-        Level p_151121_, int p_151122_, int p_151123_, int p_151124_, boolean p_151125_, boolean p_151126_, ResourceKey<Level> p_151127_
+        Level p_151121_, me.alphamode.mcbig.math.BigInteger p_151122_, me.alphamode.mcbig.math.BigInteger p_151123_, int p_151124_, boolean p_151125_, boolean p_151126_, ResourceKey<Level> p_151127_
     ) {
-        MapItemSavedData mapitemsaveddata = MapItemSavedData.createFresh((double)p_151122_, (double)p_151123_, (byte)p_151124_, p_151125_, p_151126_, p_151127_);
+        MapItemSavedData mapitemsaveddata = MapItemSavedData.createFresh((double)p_151122_.doubleValue(), (double)p_151123_.doubleValue(), (byte)p_151124_, p_151125_, p_151126_, p_151127_);
         MapId mapid = p_151121_.getFreeMapId();
         p_151121_.setMapData(mapid, mapitemsaveddata);
         return mapid;
@@ -119,7 +_,7 @@
                                         for(int j3 = 0; j3 < i; ++j3) {
                                             blockpos$mutableblockpos.set(j2 + i4, 0, k2 + j3);
                                             int k3 = levelchunk.getHeight(
-                                                    Heightmap.Types.WORLD_SURFACE, blockpos$mutableblockpos.getX(), blockpos$mutableblockpos.getZ()
+                                                    Heightmap.Types.WORLD_SURFACE, blockpos$mutableblockpos.getBigX(), blockpos$mutableblockpos.getBigZ()
                                                 )
                                                 + 1;
                                             BlockState blockstate;
@@ -153,7 +_,7 @@
                                                 }
                                             }
 
-                                            p_42896_.checkBanners(p_42894_, blockpos$mutableblockpos.getX(), blockpos$mutableblockpos.getZ());
+                                            p_42896_.checkBanners(p_42894_, blockpos$mutableblockpos.getBigX(), blockpos$mutableblockpos.getBigZ());
                                             d1 += (double)k3 / (double)(i * i);
                                             multiset.add(blockstate.getMapColor(p_42894_, blockpos$mutableblockpos));
                                         }
