--- a/net/minecraft/world/item/SpawnEggItem.java
+++ b/net/minecraft/world/item/SpawnEggItem.java
@@ -116,7 +_,7 @@
                     }
 
                     p_43226_.awardStat(Stats.ITEM_USED.get(this));
-                    p_43225_.gameEvent(p_43226_, GameEvent.ENTITY_PLACE, entity.position());
+                    p_43225_.gameEvent(p_43226_, GameEvent.ENTITY_PLACE, entity.bigPosition());
                     return InteractionResultHolder.consume(itemstack);
                 }
             } else {
