--- a/net/minecraft/client/renderer/entity/PaintingRenderer.java
+++ b/net/minecraft/client/renderer/entity/PaintingRenderer.java
@@ -72,9 +_,9 @@
             for (int l = 0; l < i; l++) {
                 float f2 = (float)l + f + 0.5F;
                 float f3 = (float)k + f1 + 0.5F;
-                int i1 = p_360735_.getBlockX();
+                int i1 = p_360735_.getBlockX().intValue();
                 int j1 = Mth.floor(p_360735_.getY() + (double)f3);
-                int k1 = p_360735_.getBlockZ();
+                int k1 = p_360735_.getBlockZ().intValue();
                 switch (direction) {
                     case NORTH:
                         i1 = Mth.floor(p_360735_.getX() + (double)f2);
