--- a/net/minecraft/world/level/levelgen/feature/LargeDripstoneFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/LargeDripstoneFeature.java
@@ -185,7 +_,7 @@
                             BlockPos.MutableBlockPos blockpos$mutableblockpos = this.root.offset(i, 0, j).mutable();
                             boolean flag = false;
                             int l = this.pointingUp
-                                ? p_225146_.getHeight(Heightmap.Types.WORLD_SURFACE_WG, blockpos$mutableblockpos.getX(), blockpos$mutableblockpos.getZ())
+                                ? p_225146_.getHeight(Heightmap.Types.WORLD_SURFACE_WG, blockpos$mutableblockpos.getBigX(), blockpos$mutableblockpos.getBigZ())
                                 : Integer.MAX_VALUE;
 
                             for(int i1 = 0; i1 < k && blockpos$mutableblockpos.getY() < l; ++i1) {
