--- a/net/minecraft/world/level/levelgen/feature/VoidStartPlatformFeature.java
+++ b/net/minecraft/world/level/levelgen/feature/VoidStartPlatformFeature.java
@@ -17,8 +_,8 @@
         super(p_67354_);
     }
 
-    private static int checkerboardDistance(int p_67356_, int p_67357_, int p_67358_, int p_67359_) {
-        return Math.max(Math.abs(p_67356_ - p_67358_), Math.abs(p_67357_ - p_67359_));
+    private static int checkerboardDistance(me.alphamode.mcbig.math.BigInteger p_67356_, me.alphamode.mcbig.math.BigInteger p_67357_, me.alphamode.mcbig.math.BigInteger p_67358_, me.alphamode.mcbig.math.BigInteger p_67359_) {
+        return p_67356_.subtract(p_67358_).abs().max(p_67357_.subtract(p_67359_).abs()).intValue();
     }
 
     @Override
@@ -31,10 +_,10 @@
             BlockPos blockpos = PLATFORM_OFFSET.atY(p_160633_.origin().getY() + PLATFORM_OFFSET.getY());
             BlockPos.MutableBlockPos blockpos$mutableblockpos = new BlockPos.MutableBlockPos();
 
-            for (int i = chunkpos.getMinBlockZ(); i <= chunkpos.getMaxBlockZ(); i++) {
-                for (int j = chunkpos.getMinBlockX(); j <= chunkpos.getMaxBlockX(); j++) {
-                    if (checkerboardDistance(blockpos.getX(), blockpos.getZ(), j, i) <= 16) {
-                        blockpos$mutableblockpos.set(j, blockpos.getY(), i);
+            for(me.alphamode.mcbig.math.BigInteger i = chunkpos.getMinBlockZ(); i.compareTo(chunkpos.getMaxBlockZ()) <= 0; i = i.add()) {
+                for(me.alphamode.mcbig.math.BigInteger j = chunkpos.getMinBlockX(); j.compareTo(chunkpos.getMaxBlockX()) <= 0; j = j.add()) {
+                    if (checkerboardDistance(blockpos.getBigX(), blockpos.getBigZ(), j, i) <= 16) {
+                        blockpos$mutableblockpos.set(j, blockpos.getBigY(), i);
                         if (blockpos$mutableblockpos.equals(blockpos)) {
                             worldgenlevel.setBlock(blockpos$mutableblockpos, Blocks.COBBLESTONE.defaultBlockState(), 2);
                         } else {
