--- a/net/minecraft/world/level/storage/loot/predicates/LocationCheck.java
+++ b/net/minecraft/world/level/storage/loot/predicates/LocationCheck.java
@@ -17,9 +_,9 @@
 public record LocationCheck(Optional<LocationPredicate> predicate, BlockPos offset) implements LootItemCondition {
     private static final MapCodec<BlockPos> OFFSET_CODEC = RecordCodecBuilder.mapCodec(
         p_338168_ -> p_338168_.group(
-                    Codec.INT.optionalFieldOf("offsetX", Integer.valueOf(0)).forGetter(Vec3i::getX),
-                    Codec.INT.optionalFieldOf("offsetY", Integer.valueOf(0)).forGetter(Vec3i::getY),
-                    Codec.INT.optionalFieldOf("offsetZ", Integer.valueOf(0)).forGetter(Vec3i::getZ)
+                    Codec.INT.optionalFieldOf("offsetX", Integer.valueOf(0)).forGetter(me.alphamode.mcbig.core.Vec3l::getX),
+                    Codec.INT.optionalFieldOf("offsetY", Integer.valueOf(0)).forGetter(me.alphamode.mcbig.core.Vec3l::getY),
+                    Codec.INT.optionalFieldOf("offsetZ", Integer.valueOf(0)).forGetter(me.alphamode.mcbig.core.Vec3l::getZ)
                 )
                 .apply(p_338168_, BlockPos::new)
     );
