--- a/net/minecraft/world/entity/ai/village/VillageSiege.java
+++ b/net/minecraft/world/entity/ai/village/VillageSiege.java
@@ -22,9 +_,9 @@
     private VillageSiege.State siegeState = VillageSiege.State.SIEGE_DONE;
     private int zombiesToSpawn;
     private int nextSpawnTime;
-    private int spawnX;
-    private int spawnY;
-    private int spawnZ;
+    private me.alphamode.mcbig.math.BigInteger spawnX;
+    private me.alphamode.mcbig.math.BigInteger spawnY;
+    private me.alphamode.mcbig.math.BigInteger spawnZ;
 
     @Override
     public int tick(ServerLevel p_27013_, boolean p_27014_, boolean p_27015_) {
@@ -74,9 +_,9 @@
                 if (p_27008_.isVillage(blockpos) && !p_27008_.getBiome(blockpos).is(BiomeTags.WITHOUT_ZOMBIE_SIEGES)) {
                     for(int i = 0; i < 10; ++i) {
                         float f = p_27008_.random.nextFloat() * (float) (Math.PI * 2);
-                        this.spawnX = blockpos.getX() + Mth.floor(Mth.cos(f) * 32.0F);
-                        this.spawnY = blockpos.getY();
-                        this.spawnZ = blockpos.getZ() + Mth.floor(Mth.sin(f) * 32.0F);
+                        this.spawnX = blockpos.getBigX().add(Mth.floor(Mth.cos(f) * 32.0F));
+                        this.spawnY = blockpos.getBigY();
+                        this.spawnZ = blockpos.getBigZ().add(Mth.floor(Mth.sin(f) * 32.0F));
                         if (this.findRandomSpawnPos(p_27008_, new BlockPos(this.spawnX, this.spawnY, this.spawnZ)) != null) {
                             this.nextSpawnTime = 0;
                             this.zombiesToSpawn = 20;
@@ -112,8 +_,8 @@
     @Nullable
     private Vec3 findRandomSpawnPos(ServerLevel p_27010_, BlockPos p_27011_) {
         for(int i = 0; i < 10; ++i) {
-            int j = p_27011_.getX() + p_27010_.random.nextInt(16) - 8;
-            int k = p_27011_.getZ() + p_27010_.random.nextInt(16) - 8;
+            me.alphamode.mcbig.math.BigInteger j = p_27011_.getBigX().add(p_27010_.random.nextInt(16) - 8);
+            me.alphamode.mcbig.math.BigInteger k = p_27011_.getBigZ().add(p_27010_.random.nextInt(16) - 8);
             int l = p_27010_.getHeight(Heightmap.Types.WORLD_SURFACE, j, k);
             BlockPos blockpos = new BlockPos(j, l, k);
             if (p_27010_.isVillage(blockpos) && Monster.checkMonsterSpawnRules(EntityType.ZOMBIE, p_27010_, MobSpawnType.EVENT, blockpos, p_27010_.random)) {
