--- a/net/minecraft/client/renderer/debug/HeightMapRenderer.java
+++ b/net/minecraft/client/renderer/debug/HeightMapRenderer.java
@@ -44,8 +_,8 @@
 
                     for(int k = 0; k < 16; ++k) {
                         for(int l = 0; l < 16; ++l) {
-                            int i1 = SectionPos.sectionToBlockCoord(chunkpos.x, k);
-                            int j1 = SectionPos.sectionToBlockCoord(chunkpos.z, l);
+                            int i1 = SectionPos.sectionToBlockCoord(chunkpos.x(), k).intValue();
+                            int j1 = SectionPos.sectionToBlockCoord(chunkpos.z(), l).intValue();
                             float f = (float)(
                                 (double)((float)levelaccessor.getHeight(heightmap$types, i1, j1) + (float)heightmap$types.ordinal() * 0.09375F) - p_113579_
                             );
