--- a/net/minecraft/network/FriendlyByteBuf.java
+++ b/net/minecraft/network/FriendlyByteBuf.java
@@ -404,29 +_,52 @@
     }
 
     public BlockPos readBlockPos() {
-        return BlockPos.of(this.readLong());
+        return new BlockPos(this.readBigInteger(), this.readBigInteger(), this.readBigInteger());
     }
 
     public FriendlyByteBuf writeBlockPos(BlockPos p_130065_) {
-        this.writeLong(p_130065_.asLong());
-        return this;
+        this.writeBigInteger(p_130065_.getBigX());
+        this.writeBigInteger(p_130065_.getBigY());
+        this.writeBigInteger(p_130065_.getBigZ());
+        return this;
+    }
+
+    public FriendlyByteBuf writeBigInteger(me.alphamode.mcbig.math.BigInteger value) {
+        this.writeByteArray(value.toByteArray());
+        return this;
+    }
+
+    public me.alphamode.mcbig.math.BigInteger readBigInteger() {
+        return new me.alphamode.mcbig.math.BigInteger(this.readByteArray());
+    }
+
+    public FriendlyByteBuf writeBigDecimal(me.alphamode.mcbig.math.BigDecimal value) {
+        this.writeUtf(value.toString());
+        return this;
+    }
+
+    public me.alphamode.mcbig.math.BigDecimal readBigDecimal() {
+        return new me.alphamode.mcbig.math.BigDecimal(this.readUtf());
     }
 
     public ChunkPos readChunkPos() {
-        return new ChunkPos(this.readLong());
+        return new ChunkPos(this.readBigInteger(), this.readBigInteger());
     }
 
     public FriendlyByteBuf writeChunkPos(ChunkPos p_178342_) {
-        this.writeLong(p_178342_.toLong());
+        this.writeBigInteger(p_178342_.x());
+        this.writeBigInteger(p_178342_.z());
         return this;
     }
 
     public SectionPos readSectionPos() {
-        return SectionPos.of(this.readLong());
+        return SectionPos.of(this.readBigInteger(), this.readBigInteger(), this.readBigInteger());
     }
 
     public FriendlyByteBuf writeSectionPos(SectionPos p_178344_) {
-        this.writeLong(p_178344_.asLong());
+        this.writeBigInteger(p_178344_.x());
+        this.writeBigInteger(p_178344_.y());
+        this.writeBigInteger(p_178344_.z());
         return this;
     }
 
@@ -672,7 +_,7 @@
         float f2 = this.readFloat();
         boolean flag = this.readBoolean();
         return new BlockHitResult(
-            new Vec3((double)blockpos.getX() + (double)f, (double)blockpos.getY() + (double)f1, (double)blockpos.getZ() + (double)f2),
+            new me.alphamode.mcbig.core.BigVec3(blockpos.getBigX().toBigDecimal().add((double)f), (double)blockpos.getBigY().doubleValue() + (double)f1, blockpos.getBigZ().toBigDecimal().add(f2)),
             direction,
             blockpos,
             flag
@@ -683,10 +_,10 @@
         BlockPos blockpos = p_130063_.getBlockPos();
         this.writeBlockPos(blockpos);
         this.writeEnum(p_130063_.getDirection());
-        Vec3 vec3 = p_130063_.getLocation();
-        this.writeFloat((float)(vec3.x - (double)blockpos.getX()));
+        me.alphamode.mcbig.core.BigVec3 vec3 = p_130063_.getLocation();
+        this.writeFloat((vec3.x.subtract(blockpos.getBigX().toBigDecimal())).floatValue());
         this.writeFloat((float)(vec3.y - (double)blockpos.getY()));
-        this.writeFloat((float)(vec3.z - (double)blockpos.getZ()));
+        this.writeFloat((vec3.z.subtract(blockpos.getBigZ().toBigDecimal())).floatValue());
         this.writeBoolean(p_130063_.isInside());
     }
 
