--- a/net/minecraft/network/FriendlyByteBuf.java
+++ b/net/minecraft/network/FriendlyByteBuf.java
@@ -360,7 +_,7 @@
     }
 
     public static BlockPos readBlockPos(ByteBuf p_320037_) {
-        return BlockPos.of(p_320037_.readLong());
+        return new BlockPos(new me.alphamode.mcbig.math.BigInteger(readByteArray(p_320037_)), new me.alphamode.mcbig.math.BigInteger(readByteArray(p_320037_)), new me.alphamode.mcbig.math.BigInteger(readByteArray(p_320037_)));
     }
 
     public FriendlyByteBuf writeBlockPos(BlockPos p_130065_) {
@@ -369,24 +_,29 @@
     }
 
     public static void writeBlockPos(ByteBuf p_320943_, BlockPos p_320546_) {
-        p_320943_.writeLong(p_320546_.asLong());
+        writeByteArray(p_320943_, p_320546_.getBigX().toByteArray());
+        writeByteArray(p_320943_, p_320546_.getBigY().toByteArray());
+        writeByteArray(p_320943_, p_320546_.getBigZ().toByteArray());
     }
 
     public ChunkPos readChunkPos() {
-        return new ChunkPos(this.readLong());
+        return new ChunkPos(this.readBigInteger(), this.readBigInteger());
     }
 
     public FriendlyByteBuf writeChunkPos(ChunkPos p_178342_) {
-        this.writeLong(p_178342_.toLong());
+        this.writeBigInteger(p_178342_.x());
+        this.writeBigInteger(p_178342_.z());
         return this;
     }
 
     public SectionPos readSectionPos() {
-        return SectionPos.of(this.readLong());
+        return SectionPos.of(this.readBigInteger(), this.readBigInteger(), this.readBigInteger());
     }
 
     public FriendlyByteBuf writeSectionPos(SectionPos p_178344_) {
-        this.writeLong(p_178344_.asLong());
+        this.writeBigInteger(p_178344_.x());
+        this.writeBigInteger(p_178344_.y());
+        this.writeBigInteger(p_178344_.z());
         return this;
     }
 
@@ -401,6 +_,24 @@
         this.writeBlockPos(p_236815_.pos());
     }
 
+    public FriendlyByteBuf writeBigInteger(me.alphamode.mcbig.math.BigInteger value) {
+        this.writeByteArray(value.toByteArray());
+        return this;
+    }
+
+    public me.alphamode.mcbig.math.BigInteger readBigInteger() {
+        return new me.alphamode.mcbig.math.BigInteger(this.readByteArray());
+    }
+
+    public FriendlyByteBuf writeBigDecimal(me.alphamode.mcbig.math.BigDecimal value) {
+        this.writeUtf(value.toString());
+        return this;
+    }
+
+    public me.alphamode.mcbig.math.BigDecimal readBigDecimal() {
+        return new me.alphamode.mcbig.math.BigDecimal(this.readUtf());
+    }
+
     public Vector3f readVector3f() {
         return readVector3f(this);
     }
@@ -627,7 +_,7 @@
         float f2 = this.readFloat();
         boolean flag = this.readBoolean();
         return new BlockHitResult(
-            new Vec3((double)blockpos.getX() + (double)f, (double)blockpos.getY() + (double)f1, (double)blockpos.getZ() + (double)f2),
+            new me.alphamode.mcbig.core.BigVec3(blockpos.getBigX().toBigDecimal().add((double)f), (double)blockpos.getBigY().doubleValue() + (double)f1, blockpos.getBigZ().toBigDecimal().add(f2)),
             direction,
             blockpos,
             flag
@@ -638,10 +_,10 @@
         BlockPos blockpos = p_130063_.getBlockPos();
         this.writeBlockPos(blockpos);
         this.writeEnum(p_130063_.getDirection());
-        Vec3 vec3 = p_130063_.getLocation();
-        this.writeFloat((float)(vec3.x - (double)blockpos.getX()));
+        me.alphamode.mcbig.core.BigVec3 vec3 = p_130063_.getLocation();
+        this.writeFloat((vec3.x.subtract(blockpos.getBigX().toBigDecimal())).floatValue());
         this.writeFloat((float)(vec3.y - (double)blockpos.getY()));
-        this.writeFloat((float)(vec3.z - (double)blockpos.getZ()));
+        this.writeFloat((vec3.z.subtract(blockpos.getBigZ().toBigDecimal())).floatValue());
         this.writeBoolean(p_130063_.isInside());
     }
 
