--- a/net/minecraft/world/level/block/entity/BlockEntity.java
+++ b/net/minecraft/world/level/block/entity/BlockEntity.java
@@ -31,7 +_,7 @@
     }
 
     public static BlockPos getPosFromTag(CompoundTag p_187473_) {
-        return new BlockPos(p_187473_.getInt("x"), p_187473_.getInt("y"), p_187473_.getInt("z"));
+        return new BlockPos(p_187473_.getByteArray("x"), p_187473_.getByteArray("y"), p_187473_.getByteArray("z"));
     }
 
     @Nullable
@@ -90,9 +_,9 @@
 
     private void saveMetadata(CompoundTag p_187479_) {
         this.saveId(p_187479_);
-        p_187479_.putInt("x", this.worldPosition.getX());
-        p_187479_.putInt("y", this.worldPosition.getY());
-        p_187479_.putInt("z", this.worldPosition.getZ());
+        p_187479_.putByteArray("x", this.worldPosition.getBigX().toByteArray());
+        p_187479_.putByteArray("y", this.worldPosition.getBigY().toByteArray());
+        p_187479_.putByteArray("z", this.worldPosition.getBigZ().toByteArray());
     }
 
     @Nullable
