--- a/net/minecraft/world/level/block/entity/SculkCatalystBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/SculkCatalystBlockEntity.java
@@ -82,13 +_,13 @@
         }
 
         @Override
-        public boolean handleGameEvent(ServerLevel p_283470_, Holder<GameEvent> p_316661_, GameEvent.Context p_283014_, Vec3 p_282350_) {
+        public boolean handleGameEvent(ServerLevel p_283470_, Holder<GameEvent> p_316661_, GameEvent.Context p_283014_, me.alphamode.mcbig.core.BigVec3 p_282350_) {
             if (p_316661_.is(GameEvent.ENTITY_DIE) && p_283014_.sourceEntity() instanceof LivingEntity livingentity) {
                 if (!livingentity.wasExperienceConsumed()) {
                     DamageSource damagesource = livingentity.getLastDamageSource();
                     int i = livingentity.getExperienceReward(p_283470_, Optionull.map(damagesource, DamageSource::getEntity));
                     if (livingentity.shouldDropExperience() && i > 0) {
-                        this.sculkSpreader.addCursors(BlockPos.containing(p_282350_.relative(Direction.UP, 0.5)), i);
+                        this.sculkSpreader.addCursors(BlockPos.containing(p_282350_.relative(Direction.UP, me.alphamode.mcbig.core.BigConstants.POINT_FIVE)), i);
                         this.tryAwardItSpreadsAdvancement(p_283470_, livingentity);
                     }
 
