--- a/net/minecraft/world/level/block/entity/TheEndGatewayBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/TheEndGatewayBlockEntity.java
@@ -204,7 +_,7 @@
             for (int j = -p_59945_; j <= p_59945_; j++) {
                 if (i != 0 || j != 0 || p_59946_) {
                     for (int k = p_59943_.getMaxY(); k > (blockpos == null ? p_59943_.getMinY() : blockpos.getY()); k--) {
-                        BlockPos blockpos1 = new BlockPos(p_59944_.getX() + i, k, p_59944_.getZ() + j);
+                        BlockPos blockpos1 = new BlockPos(p_59944_.getBigX().add(i), k, p_59944_.getBigZ().add(j));
                         BlockState blockstate = p_59943_.getBlockState(blockpos1);
                         if (blockstate.isCollisionShapeFullBlock(p_59943_, blockpos1) && (p_59946_ || !blockstate.is(Blocks.BEDROCK))) {
                             blockpos = blockpos1;
@@ -219,7 +_,7 @@
     }
 
     private static LevelChunk getChunk(Level p_59948_, Vec3 p_59949_) {
-        return p_59948_.getChunk(Mth.floor(p_59949_.x / 16.0), Mth.floor(p_59949_.z / 16.0));
+        return p_59948_.getChunk(Mth.bigFloor(p_59949_.x / 16.0), Mth.bigFloor(p_59949_.z / 16.0));
     }
 
     @Nullable
@@ -238,7 +_,7 @@
             if (blockstate.is(Blocks.END_STONE)
                 && !p_59954_.getBlockState(blockpos4).isCollisionShapeFullBlock(p_59954_, blockpos4)
                 && !p_59954_.getBlockState(blockpos5).isCollisionShapeFullBlock(p_59954_, blockpos5)) {
-                double d1 = blockpos3.distToCenterSqr(0.0, 0.0, 0.0);
+                double d1 = blockpos3.distToCenterSqr(me.alphamode.mcbig.math.BigDecimal.ZERO, 0, me.alphamode.mcbig.math.BigDecimal.ZERO);
                 if (blockpos2 == null || d1 < d0) {
                     blockpos2 = blockpos3;
                     d0 = d1;
