--- a/net/minecraft/world/level/chunk/storage/ChunkStorage.java
+++ b/net/minecraft/world/level/chunk/storage/ChunkStorage.java
@@ -98,7 +_,7 @@
 
     protected void handleLegacyStructureIndex(ChunkPos p_321604_) {
         if (this.legacyStructureHandler != null) {
-            this.legacyStructureHandler.removeIndex(p_321604_.toLong());
+            this.legacyStructureHandler.removeIndex(p_321604_);
         }
     }
 
