--- a/net/minecraft/world/entity/monster/PatrollingMonster.java
+++ b/net/minecraft/world/entity/monster/PatrollingMonster.java
@@ -185,7 +_,7 @@
                     BlockPos blockpos = BlockPos.containing(vec33);
                     blockpos = this.mob.level().getHeightmapPos(Heightmap.Types.MOTION_BLOCKING_NO_LEAVES, blockpos);
                     if (!pathnavigation.moveTo(
-                        (double)blockpos.getX(), (double)blockpos.getY(), (double)blockpos.getZ(), flag ? this.leaderSpeedModifier : this.speedModifier
+                        (double)blockpos.getBigX().doubleValue(), (double)blockpos.getY(), (double)blockpos.getBigZ().doubleValue(), flag ? this.leaderSpeedModifier : this.speedModifier
                     )) {
                         this.moveRandomly();
                         this.cooldownUntil = this.mob.level().getGameTime() + 200L;
@@ -213,7 +_,7 @@
                 .getHeightmapPos(
                     Heightmap.Types.MOTION_BLOCKING_NO_LEAVES, this.mob.blockPosition().offset(-8 + randomsource.nextInt(16), 0, -8 + randomsource.nextInt(16))
                 );
-            return this.mob.getNavigation().moveTo((double)blockpos.getX(), (double)blockpos.getY(), (double)blockpos.getZ(), this.speedModifier);
+            return this.mob.getNavigation().moveTo((double)blockpos.getBigX().doubleValue(), (double)blockpos.getY(), (double)blockpos.getBigZ().doubleValue(), this.speedModifier);
         }
     }
 }
