--- a/net/minecraft/world/entity/monster/Ravager.java
+++ b/net/minecraft/world/entity/monster/Ravager.java
@@ -138,7 +_,7 @@
                 AABB aabb = this.getBoundingBox().inflate(0.2);
 
                 for(BlockPos blockpos : BlockPos.betweenClosed(
-                    Mth.floor(aabb.minX), Mth.floor(aabb.minY), Mth.floor(aabb.minZ), Mth.floor(aabb.maxX), Mth.floor(aabb.maxY), Mth.floor(aabb.maxZ)
+                    Mth.bigFloor(aabb.minX), Mth.bigFloor(aabb.minY), Mth.bigFloor(aabb.minZ), Mth.bigFloor(aabb.maxX), Mth.bigFloor(aabb.maxY), Mth.bigFloor(aabb.maxZ)
                 )) {
                     BlockState blockstate = this.level().getBlockState(blockpos);
                     Block block = blockstate.getBlock();
