--- a/net/minecraft/world/entity/monster/Vex.java
+++ b/net/minecraft/world/entity/monster/Vex.java
@@ -109,7 +_,7 @@
     public void readAdditionalSaveData(CompoundTag p_34008_) {
         super.readAdditionalSaveData(p_34008_);
         if (p_34008_.contains("BoundX")) {
-            this.boundOrigin = new BlockPos(p_34008_.getInt("BoundX"), p_34008_.getInt("BoundY"), p_34008_.getInt("BoundZ"));
+            this.boundOrigin = new BlockPos(p_34008_.getByteArray("BoundX"), p_34008_.getByteArray("BoundY"), p_34008_.getByteArray("BoundZ"));
         }
 
         if (p_34008_.contains("LifeTicks")) {
@@ -129,9 +_,9 @@
     public void addAdditionalSaveData(CompoundTag p_34015_) {
         super.addAdditionalSaveData(p_34015_);
         if (this.boundOrigin != null) {
-            p_34015_.putInt("BoundX", this.boundOrigin.getX());
-            p_34015_.putInt("BoundY", this.boundOrigin.getY());
-            p_34015_.putInt("BoundZ", this.boundOrigin.getZ());
+            p_34015_.putByteArray("BoundX", this.boundOrigin.getBigX().toByteArray());
+            p_34015_.putByteArray("BoundY", this.boundOrigin.getBigY().toByteArray());
+            p_34015_.putByteArray("BoundZ", this.boundOrigin.getBigZ().toByteArray());
         }
 
         if (this.hasLimitedLife) {
@@ -356,10 +_,10 @@
                 BlockPos blockpos1 = blockpos.offset(Vex.this.random.nextInt(15) - 7, Vex.this.random.nextInt(11) - 5, Vex.this.random.nextInt(15) - 7);
                 if (Vex.this.level().isEmptyBlock(blockpos1)) {
                     Vex.this.moveControl
-                        .setWantedPosition((double)blockpos1.getX() + 0.5, (double)blockpos1.getY() + 0.5, (double)blockpos1.getZ() + 0.5, 0.25);
+                        .setWantedPosition((double)blockpos1.getBigX().doubleValue() + 0.5, (double)blockpos1.getBigY().doubleValue() + 0.5, (double)blockpos1.getBigZ().doubleValue() + 0.5, 0.25);
                     if (Vex.this.getTarget() == null) {
                         Vex.this.getLookControl()
-                            .setLookAt((double)blockpos1.getX() + 0.5, (double)blockpos1.getY() + 0.5, (double)blockpos1.getZ() + 0.5, 180.0F, 20.0F);
+                            .setLookAt((double)blockpos1.getBigX().doubleValue() + 0.5, (double)blockpos1.getBigY().doubleValue() + 0.5, (double)blockpos1.getBigZ().doubleValue() + 0.5, 180.0F, 20.0F);
                     }
                     break;
                 }
