--- a/net/minecraft/util/LinearCongruentialGenerator.java
+++ b/net/minecraft/util/LinearCongruentialGenerator.java
@@ -5,7 +_,7 @@
     private static final long INCREMENT = 1442695040888963407L;
 
     public static long next(long p_13973_, long p_13974_) {
-        p_13973_ *= p_13973_ * 6364136223846793005L + 1442695040888963407L;
+        p_13973_ *= p_13973_ * MULTIPLIER + INCREMENT;
         return p_13973_ + p_13974_;
     }
 }
