--- a/net/minecraft/SharedConstants.java
+++ b/net/minecraft/SharedConstants.java
@@ -164,8 +_,8 @@
     }
 
     public static boolean debugVoidTerrain(ChunkPos p_183708_) {
-        int i = p_183708_.getMinBlockX();
-        int j = p_183708_.getMinBlockZ();
+        int i = p_183708_.getMinBlockX().intValue();
+        int j = p_183708_.getMinBlockZ().intValue();
         return !debugGenerateSquareTerrainWithoutNoise ? false : i > 8192 || i < 0 || j > 1024 || j < 0;
     }
 
