--- a/net/minecraft/server/level/ChunkMap.java
+++ b/net/minecraft/server/level/ChunkMap.java
@@ -116,9 +_,9 @@
     public static final int MIN_VIEW_DISTANCE = 2;
     public static final int MAX_VIEW_DISTANCE = 32;
     public static final int FORCED_TICKET_LEVEL = ChunkLevel.byStatus(FullChunkStatus.ENTITY_TICKING);
-    private final Long2ObjectLinkedOpenHashMap<ChunkHolder> updatingChunkMap = new Long2ObjectLinkedOpenHashMap<>();
-    private volatile Long2ObjectLinkedOpenHashMap<ChunkHolder> visibleChunkMap = this.updatingChunkMap.clone();
-    private final Long2ObjectLinkedOpenHashMap<ChunkHolder> pendingUnloads = new Long2ObjectLinkedOpenHashMap<>();
+    private final it.unimi.dsi.fastutil.objects.Object2ObjectLinkedOpenHashMap<ChunkPos, ChunkHolder> updatingChunkMap = new it.unimi.dsi.fastutil.objects.Object2ObjectLinkedOpenHashMap<>();
+    private volatile it.unimi.dsi.fastutil.objects.Object2ObjectLinkedOpenHashMap<ChunkPos, ChunkHolder> visibleChunkMap = this.updatingChunkMap.clone();
+    private final it.unimi.dsi.fastutil.objects.Object2ObjectLinkedOpenHashMap<ChunkPos, ChunkHolder> pendingUnloads = new it.unimi.dsi.fastutil.objects.Object2ObjectLinkedOpenHashMap<>();
     private final List<ChunkGenerationTask> pendingGenerationTasks = new ArrayList<>();
     final ServerLevel level;
     private final ThreadedLevelLightEngine lightEngine;
@@ -127,7 +_,7 @@
     private final ChunkGeneratorStructureState chunkGeneratorState;
     private final Supplier<DimensionDataStorage> overworldDataStorage;
     private final PoiManager poiManager;
-    final LongSet toDrop = new LongOpenHashSet();
+    final it.unimi.dsi.fastutil.objects.ObjectSet<ChunkPos> toDrop = new it.unimi.dsi.fastutil.objects.ObjectOpenHashSet<>();
     private boolean modified;
     private final ChunkTaskDispatcher worldgenTaskDispatcher;
     private final ChunkTaskDispatcher lightTaskDispatcher;
@@ -138,9 +_,9 @@
     private final String storageName;
     private final PlayerMap playerMap = new PlayerMap();
     private final Int2ObjectMap<ChunkMap.TrackedEntity> entityMap = new Int2ObjectOpenHashMap<>();
-    private final Long2ByteMap chunkTypeCache = new Long2ByteOpenHashMap();
-    private final Long2LongMap nextChunkSaveTime = new Long2LongOpenHashMap();
-    private final LongSet chunksToEagerlySave = new LongLinkedOpenHashSet();
+    private final it.unimi.dsi.fastutil.objects.Object2ByteMap<ChunkPos> chunkTypeCache = new it.unimi.dsi.fastutil.objects.Object2ByteOpenHashMap<>();
+    private final it.unimi.dsi.fastutil.objects.Object2LongMap<ChunkPos> nextChunkSaveTime = new it.unimi.dsi.fastutil.objects.Object2LongOpenHashMap<>();
+    private final it.unimi.dsi.fastutil.objects.ObjectSet<ChunkPos> chunksToEagerlySave = new it.unimi.dsi.fastutil.objects.ObjectLinkedOpenHashSet<>();
     private final Queue<Runnable> unloadQueue = Queues.newConcurrentLinkedQueue();
     private final AtomicInteger activeChunkWrites = new AtomicInteger();
     private int serverViewDistance;
@@ -205,7 +_,7 @@
     }
 
     private void setChunkUnsaved(ChunkPos p_381702_) {
-        this.chunksToEagerlySave.add(p_381702_.toLong());
+        this.chunksToEagerlySave.add(p_381702_);
     }
 
     protected ChunkGenerator generator() {
@@ -221,25 +_,25 @@
     }
 
     private static double euclideanDistanceSquared(ChunkPos p_140227_, Entity p_140228_) {
-        double d0 = (double)SectionPos.sectionToBlockCoord(p_140227_.x, 8);
-        double d1 = (double)SectionPos.sectionToBlockCoord(p_140227_.z, 8);
+        double d0 = (double)SectionPos.sectionToBlockCoord(p_140227_.x, 8).doubleValue();
+        double d1 = (double)SectionPos.sectionToBlockCoord(p_140227_.z, 8).doubleValue();
         double d2 = d0 - p_140228_.getX();
         double d3 = d1 - p_140228_.getZ();
         return d2 * d2 + d3 * d3;
     }
 
-    boolean isChunkTracked(ServerPlayer p_295366_, int p_294911_, int p_296247_) {
+    boolean isChunkTracked(ServerPlayer p_295366_, me.alphamode.mcbig.math.BigInteger p_294911_, me.alphamode.mcbig.math.BigInteger p_296247_) {
         return p_295366_.getChunkTrackingView().contains(p_294911_, p_296247_)
-            && !p_295366_.connection.chunkSender.isPending(ChunkPos.asLong(p_294911_, p_296247_));
+            && !p_295366_.connection.chunkSender.isPending(new ChunkPos(p_294911_, p_296247_));
     }
 
-    private boolean isChunkOnTrackedBorder(ServerPlayer p_295596_, int p_294838_, int p_295212_) {
+    private boolean isChunkOnTrackedBorder(ServerPlayer p_295596_, me.alphamode.mcbig.math.BigInteger p_294838_, me.alphamode.mcbig.math.BigInteger p_295212_) {
         if (!this.isChunkTracked(p_295596_, p_294838_, p_295212_)) {
             return false;
         } else {
             for (int i = -1; i <= 1; i++) {
                 for (int j = -1; j <= 1; j++) {
-                    if ((i != 0 || j != 0) && !this.isChunkTracked(p_295596_, p_294838_ + i, p_295212_ + j)) {
+                    if ((i != 0 || j != 0) && !this.isChunkTracked(p_295596_, p_294838_.add(i), p_295212_.add(j))) {
                         return true;
                     }
                 }
@@ -254,16 +_,16 @@
     }
 
     @Nullable
-    protected ChunkHolder getUpdatingChunkIfPresent(long p_140175_) {
+    protected ChunkHolder getUpdatingChunkIfPresent(ChunkPos p_140175_) {
         return this.updatingChunkMap.get(p_140175_);
     }
 
     @Nullable
-    protected ChunkHolder getVisibleChunkIfPresent(long p_140328_) {
+    protected ChunkHolder getVisibleChunkIfPresent(ChunkPos p_140328_) {
         return this.visibleChunkMap.get(p_140328_);
     }
 
-    protected IntSupplier getChunkQueueLevel(long p_140372_) {
+    protected IntSupplier getChunkQueueLevel(ChunkPos p_140372_) {
         return () -> {
             ChunkHolder chunkholder = this.getVisibleChunkIfPresent(p_140372_);
             return chunkholder == null
@@ -273,7 +_,7 @@
     }
 
     public String getChunkDebugData(ChunkPos p_140205_) {
-        ChunkHolder chunkholder = this.getVisibleChunkIfPresent(p_140205_.toLong());
+        ChunkHolder chunkholder = this.getVisibleChunkIfPresent(p_140205_);
         if (chunkholder == null) {
             return "null";
         } else {
@@ -306,7 +_,7 @@
             for (int j = -p_282030_; j <= p_282030_; j++) {
                 for (int k = -p_282030_; k <= p_282030_; k++) {
                     int l = Math.max(Math.abs(k), Math.abs(j));
-                    long i1 = ChunkPos.asLong(chunkpos.x + k, chunkpos.z + j);
+                    ChunkPos i1 = new ChunkPos(chunkpos.x.add(k), chunkpos.z.add(j));
                     ChunkHolder chunkholder = this.getUpdatingChunkIfPresent(i1);
                     if (chunkholder == null) {
                         return UNLOADED_CHUNK_LIST_FUTURE;
@@ -372,7 +_,7 @@
     }
 
     @Nullable
-    ChunkHolder updateChunkScheduling(long p_140177_, int p_140178_, @Nullable ChunkHolder p_140179_, int p_140180_) {
+    ChunkHolder updateChunkScheduling(ChunkPos p_140177_, int p_140178_, @Nullable ChunkHolder p_140179_, int p_140180_) {
         if (!ChunkLevel.isLoaded(p_140180_) && !ChunkLevel.isLoaded(p_140178_)) {
             return p_140179_;
         } else {
@@ -393,7 +_,7 @@
                 if (p_140179_ != null) {
                     p_140179_.setTicketLevel(p_140178_);
                 } else {
-                    p_140179_ = new ChunkHolder(new ChunkPos(p_140177_), p_140178_, this.level, this.lightEngine, this::onLevelChange, this);
+                    p_140179_ = new ChunkHolder(p_140177_, p_140178_, this.level, this.lightEngine, this::onLevelChange, this);
                 }
 
                 this.updatingChunkMap.put(p_140177_, p_140179_);
@@ -480,8 +_,8 @@
     }
 
     private void processUnloads(BooleanSupplier p_140354_) {
-        for (LongIterator longiterator = this.toDrop.iterator(); longiterator.hasNext(); longiterator.remove()) {
-            long i = longiterator.nextLong();
+        for (it.unimi.dsi.fastutil.objects.ObjectIterator<ChunkPos> longiterator = this.toDrop.iterator(); longiterator.hasNext(); longiterator.remove()) {
+            ChunkPos i = longiterator.next();
             ChunkHolder chunkholder = this.updatingChunkMap.get(i);
             if (chunkholder != null) {
                 this.updatingChunkMap.remove(i);
@@ -505,10 +_,10 @@
     private void saveChunksEagerly(BooleanSupplier p_381708_) {
         long i = Util.getMillis();
         int j = 0;
-        LongIterator longiterator = this.chunksToEagerlySave.iterator();
+        it.unimi.dsi.fastutil.objects.ObjectIterator<ChunkPos> longiterator = this.chunksToEagerlySave.iterator();
 
         while (j < 20 && this.activeChunkWrites.get() < 128 && p_381708_.getAsBoolean() && longiterator.hasNext()) {
-            long k = longiterator.nextLong();
+            ChunkPos k = longiterator.next();
             ChunkHolder chunkholder = this.visibleChunkMap.get(k);
             ChunkAccess chunkaccess = chunkholder != null ? chunkholder.getLatestChunk() : null;
             if (chunkaccess == null || !chunkaccess.isUnsaved()) {
@@ -520,7 +_,7 @@
         }
     }
 
-    private void scheduleUnload(long p_140182_, ChunkHolder p_140183_) {
+    private void scheduleUnload(ChunkPos p_140182_, ChunkHolder p_140183_) {
         CompletableFuture<?> completablefuture = p_140183_.getSaveSyncFuture();
         completablefuture.thenRunAsync(() -> {
             CompletableFuture<?> completablefuture1 = p_140183_.getSaveSyncFuture();
@@ -541,7 +_,7 @@
                     this.lightEngine.updateChunkStatus(chunkaccess.getPos());
                     this.lightEngine.tryScheduleUpdate();
                     this.progressListener.onStatusChange(chunkaccess.getPos(), null);
-                    this.nextChunkSaveTime.remove(chunkaccess.getPos().toLong());
+                    this.nextChunkSaveTime.removeLong(chunkaccess.getPos());
                 }
             }
         }, this.unloadQueue::add).whenComplete((p_381764_, p_381765_) -> {
@@ -614,15 +_,15 @@
     }
 
     private void markPositionReplaceable(ChunkPos p_140423_) {
-        this.chunkTypeCache.put(p_140423_.toLong(), (byte)-1);
+        this.chunkTypeCache.put(p_140423_, (byte)-1);
     }
 
     private byte markPosition(ChunkPos p_140230_, ChunkType p_332120_) {
-        return this.chunkTypeCache.put(p_140230_.toLong(), (byte)(p_332120_ == ChunkType.PROTOCHUNK ? -1 : 1));
+        return this.chunkTypeCache.put(p_140230_, (byte)(p_332120_ == ChunkType.PROTOCHUNK ? -1 : 1));
     }
 
     @Override
-    public GenerationChunkHolder acquireGeneration(long p_347661_) {
+    public GenerationChunkHolder acquireGeneration(ChunkPos p_347661_) {
         ChunkHolder chunkholder = this.updatingChunkMap.get(p_347661_);
         chunkholder.increaseGenerationRefCount();
         return chunkholder;
@@ -654,8 +_,8 @@
                 CrashReport crashreport = CrashReport.forThrowable(exception, "Exception generating new chunk");
                 CrashReportCategory crashreportcategory = crashreport.addCategory("Chunk to be generated");
                 crashreportcategory.setDetail("Status being generated", () -> p_347638_.targetStatus().getName());
-                crashreportcategory.setDetail("Location", String.format(Locale.ROOT, "%d,%d", chunkpos.x, chunkpos.z));
-                crashreportcategory.setDetail("Position hash", ChunkPos.asLong(chunkpos.x, chunkpos.z));
+                crashreportcategory.setDetail("Location", String.format(Locale.ROOT, "%s,%s", chunkpos.x, chunkpos.z));
+//                crashreportcategory.setDetail("Position hash", ChunkPos.asLong(chunkpos.x, chunkpos.z));
                 crashreportcategory.setDetail("Generator", this.generator());
                 this.mainThreadExecutor.execute(() -> {
                     throw new ReportedException(crashreport);
@@ -679,7 +_,7 @@
             if (completablefuture != null) {
                 completablefuture.thenRun(() -> this.runGenerationTask(p_347721_));
             }
-        }, generationchunkholder.getPos().toLong(), generationchunkholder::getQueueLevel);
+        }, generationchunkholder.getPos(), generationchunkholder::getQueueLevel);
     }
 
     @Override
@@ -739,7 +_,7 @@
             } else if (!chunkaccess.isUnsaved()) {
                 return false;
             } else {
-                long i = chunkaccess.getPos().toLong();
+                ChunkPos i = chunkaccess.getPos();
                 long j = this.nextChunkSaveTime.getOrDefault(i, -1L);
                 if (p_361555_ < j) {
                     return false;
@@ -792,14 +_,14 @@
                 this.markPosition(chunkpos, chunkstatus.getChunkType());
                 return true;
             } catch (Exception exception) {
-                this.level.getServer().reportChunkSaveFailure(exception, this.storageInfo(), chunkpos);
+                LOGGER.error("Failed to save chunk {},{}", chunkpos.x, chunkpos.z, exception);
                 return false;
             }
         }
     }
 
     private boolean isExistingChunkFull(ChunkPos p_140426_) {
-        byte b0 = this.chunkTypeCache.get(p_140426_.toLong());
+        byte b0 = this.chunkTypeCache.get(p_140426_);
         if (b0 != 0) {
             return b0 == 1;
         } else {
@@ -838,7 +_,7 @@
     }
 
     private void markChunkPendingToSend(ServerPlayer p_294638_, ChunkPos p_296183_) {
-        LevelChunk levelchunk = this.getChunkToSend(p_296183_.toLong());
+        LevelChunk levelchunk = this.getChunkToSend(p_296183_);
         if (levelchunk != null) {
             markChunkPendingToSend(p_294638_, levelchunk);
         }
@@ -853,7 +_,7 @@
     }
 
     @Nullable
-    public LevelChunk getChunkToSend(long p_300929_) {
+    public LevelChunk getChunkToSend(ChunkPos p_300929_) {
         ChunkHolder chunkholder = this.getVisibleChunkIfPresent(p_300929_);
         return chunkholder == null ? null : chunkholder.getChunkToSend();
     }
@@ -891,9 +_,9 @@
             .build(p_140275_);
         TickingTracker tickingtracker = this.distanceManager.tickingTracker();
 
-        for (Entry<ChunkHolder> entry : this.visibleChunkMap.long2ObjectEntrySet()) {
-            long i = entry.getLongKey();
-            ChunkPos chunkpos = new ChunkPos(i);
+        for(it.unimi.dsi.fastutil.objects.Object2ObjectMap.Entry<ChunkPos, ChunkHolder> entry : this.visibleChunkMap.object2ObjectEntrySet()) {
+//            ChunkPos i = entry.getKey();
+            ChunkPos chunkpos = entry.getKey();
             ChunkHolder chunkholder = entry.getValue();
             Optional<ChunkAccess> optional = Optional.ofNullable(chunkholder.getLatestChunk());
             Optional<LevelChunk> optional1 = optional.flatMap(
@@ -909,11 +_,11 @@
                 printFuture(chunkholder.getFullChunkFuture()),
                 printFuture(chunkholder.getTickingChunkFuture()),
                 printFuture(chunkholder.getEntityTickingChunkFuture()),
-                this.distanceManager.getTicketDebugString(i),
+                this.distanceManager.getTicketDebugString(chunkpos),
                 this.anyPlayerCloseEnoughForSpawning(chunkpos),
                 optional1.<Integer>map(p_214953_ -> p_214953_.getBlockEntities().size()).orElse(0),
-                tickingtracker.getTicketDebugString(i),
-                tickingtracker.getLevel(i),
+                tickingtracker.getTicketDebugString(chunkpos),
+                tickingtracker.getLevel(chunkpos),
                 optional1.<Integer>map(p_214946_ -> p_214946_.getBlockTicks().count()).orElse(0),
                 optional1.<Integer>map(p_214937_ -> p_214937_.getFluidTicks().count()).orElse(0)
             );
@@ -944,10 +_,10 @@
     }
 
     void forEachSpawnCandidateChunk(Consumer<ChunkHolder> p_365506_) {
-        LongIterator longiterator = this.distanceManager.getSpawnCandidateChunks();
+        it.unimi.dsi.fastutil.objects.ObjectIterator<ChunkPos> longiterator = this.distanceManager.getSpawnCandidateChunks();
 
         while (longiterator.hasNext()) {
-            long i = longiterator.nextLong();
+            ChunkPos i = longiterator.next();
             ChunkHolder chunkholder = this.visibleChunkMap.get(i);
             if (chunkholder != null && this.anyPlayerCloseEnoughForSpawningInternal(chunkholder.getPos())) {
                 p_365506_.accept(chunkholder);
@@ -956,7 +_,7 @@
     }
 
     boolean anyPlayerCloseEnoughForSpawning(ChunkPos p_183880_) {
-        return !this.distanceManager.hasPlayersNearby(p_183880_.toLong()) ? false : this.anyPlayerCloseEnoughForSpawningInternal(p_183880_);
+        return !this.distanceManager.hasPlayersNearby(p_183880_) ? false : this.anyPlayerCloseEnoughForSpawningInternal(p_183880_);
     }
 
     private boolean anyPlayerCloseEnoughForSpawningInternal(ChunkPos p_363995_) {
@@ -970,8 +_,8 @@
     }
 
     public List<ServerPlayer> getPlayersCloseForSpawning(ChunkPos p_183889_) {
-        long i = p_183889_.toLong();
-        if (!this.distanceManager.hasPlayersNearby(i)) {
+//        long i = p_183889_.toLong();
+        if (!this.distanceManager.hasPlayersNearby(p_183889_)) {
             return List.of();
         } else {
             Builder<ServerPlayer> builder = ImmutableList.builder();
@@ -1040,7 +_,7 @@
         SectionPos sectionpos1 = SectionPos.of(p_140185_);
         boolean flag = this.playerMap.ignored(p_140185_);
         boolean flag1 = this.skipPlayer(p_140185_);
-        boolean flag2 = sectionpos.asLong() != sectionpos1.asLong();
+        boolean flag2 = !sectionpos.equals(sectionpos1);
         if (flag2 || flag != flag1) {
             this.updatePlayerPos(p_140185_);
             if (!flag) {
@@ -1172,7 +_,7 @@
                 chunkmap$trackedentity.lastSectionPos = sectionpos1;
             }
 
-            if (flag || this.distanceManager.inEntityTickingRange(sectionpos1.chunk().toLong())) {
+            if (flag || this.distanceManager.inEntityTickingRange(sectionpos1.chunk())) {
                 chunkmap$trackedentity.serverEntity.sendChanges();
             }
         }
@@ -1233,7 +_,7 @@
     public void waitForLightBeforeSending(ChunkPos p_301194_, int p_301130_) {
         int i = p_301130_ + 1;
         ChunkPos.rangeClosed(p_301194_, i).forEach(p_300775_ -> {
-            ChunkHolder chunkholder = this.getVisibleChunkIfPresent(p_300775_.toLong());
+            ChunkHolder chunkholder = this.getVisibleChunkIfPresent(p_300775_);
             if (chunkholder != null) {
                 chunkholder.addSendDependency(this.lightEngine.waitForPendingTasks(p_300775_.x, p_300775_.z));
             }
@@ -1246,19 +_,19 @@
         }
 
         @Override
-        protected boolean isChunkToRemove(long p_140462_) {
+        protected boolean isChunkToRemove(ChunkPos p_140462_) {
             return ChunkMap.this.toDrop.contains(p_140462_);
         }
 
         @Nullable
         @Override
-        protected ChunkHolder getChunk(long p_140469_) {
+        protected ChunkHolder getChunk(ChunkPos p_140469_) {
             return ChunkMap.this.getUpdatingChunkIfPresent(p_140469_);
         }
 
         @Nullable
         @Override
-        protected ChunkHolder updateChunkScheduling(long p_140464_, int p_140465_, @Nullable ChunkHolder p_140466_, int p_140467_) {
+        protected ChunkHolder updateChunkScheduling(ChunkPos p_140464_, int p_140465_, @Nullable ChunkHolder p_140466_, int p_140467_) {
             return ChunkMap.this.updateChunkScheduling(p_140464_, p_140465_, p_140466_, p_140467_);
         }
     }
