--- a/net/minecraft/server/level/ChunkTaskPriorityQueue.java
+++ b/net/minecraft/server/level/ChunkTaskPriorityQueue.java
@@ -9,9 +_,9 @@
 
 public class ChunkTaskPriorityQueue {
     public static final int PRIORITY_LEVEL_COUNT = ChunkLevel.MAX_LEVEL + 2;
-    private final List<Long2ObjectLinkedOpenHashMap<List<Runnable>>> queuesPerPriority = IntStream.range(0, PRIORITY_LEVEL_COUNT)
-        .mapToObj(p_140520_ -> new Long2ObjectLinkedOpenHashMap<List<Runnable>>())
-        .toList();
+    private final List<it.unimi.dsi.fastutil.objects.Object2ObjectLinkedOpenHashMap<ChunkPos, List<Runnable>>> queuesPerPriority = IntStream.range(0, PRIORITY_LEVEL_COUNT)
+            .mapToObj(p_140520_ -> new it.unimi.dsi.fastutil.objects.Object2ObjectLinkedOpenHashMap<ChunkPos, List<Runnable>>())
+            .toList();
     private volatile int topPriorityQueueIndex = PRIORITY_LEVEL_COUNT;
     private final String name;
 
@@ -21,8 +_,8 @@
 
     protected void resortChunkTasks(int p_140522_, ChunkPos p_140523_, int p_140524_) {
         if (p_140522_ < PRIORITY_LEVEL_COUNT) {
-            Long2ObjectLinkedOpenHashMap<List<Runnable>> long2objectlinkedopenhashmap = this.queuesPerPriority.get(p_140522_);
-            List<Runnable> list = long2objectlinkedopenhashmap.remove(p_140523_.toLong());
+            it.unimi.dsi.fastutil.objects.Object2ObjectLinkedOpenHashMap<ChunkPos, List<Runnable>> long2objectlinkedopenhashmap = this.queuesPerPriority.get(p_140522_);
+            List<Runnable> list = long2objectlinkedopenhashmap.remove(p_140523_);
             if (p_140522_ == this.topPriorityQueueIndex) {
                 while (this.hasWork() && this.queuesPerPriority.get(this.topPriorityQueueIndex).isEmpty()) {
                     this.topPriorityQueueIndex++;
@@ -30,19 +_,19 @@
             }
 
             if (list != null && !list.isEmpty()) {
-                this.queuesPerPriority.get(p_140524_).computeIfAbsent(p_140523_.toLong(), p_140547_ -> Lists.newArrayList()).addAll(list);
+                this.queuesPerPriority.get(p_140524_).computeIfAbsent(p_140523_, p_140547_ -> Lists.newArrayList()).addAll(list);
                 this.topPriorityQueueIndex = Math.min(this.topPriorityQueueIndex, p_140524_);
             }
         }
     }
 
-    protected void submit(Runnable p_371364_, long p_140537_, int p_140538_) {
+    protected void submit(Runnable p_371364_, ChunkPos p_140537_, int p_140538_) {
         this.queuesPerPriority.get(p_140538_).computeIfAbsent(p_140537_, p_140545_ -> Lists.newArrayList()).add(p_371364_);
         this.topPriorityQueueIndex = Math.min(this.topPriorityQueueIndex, p_140538_);
     }
 
-    protected void release(long p_140531_, boolean p_140532_) {
-        for (Long2ObjectLinkedOpenHashMap<List<Runnable>> long2objectlinkedopenhashmap : this.queuesPerPriority) {
+    protected void release(ChunkPos p_140531_, boolean p_140532_) {
+        for (it.unimi.dsi.fastutil.objects.Object2ObjectLinkedOpenHashMap<ChunkPos, List<Runnable>> long2objectlinkedopenhashmap : this.queuesPerPriority) {
             List<Runnable> list = long2objectlinkedopenhashmap.get(p_140531_);
             if (list != null) {
                 if (p_140532_) {
@@ -66,8 +_,8 @@
             return null;
         } else {
             int i = this.topPriorityQueueIndex;
-            Long2ObjectLinkedOpenHashMap<List<Runnable>> long2objectlinkedopenhashmap = this.queuesPerPriority.get(i);
-            long j = long2objectlinkedopenhashmap.firstLongKey();
+            it.unimi.dsi.fastutil.objects.Object2ObjectLinkedOpenHashMap<ChunkPos, List<Runnable>> long2objectlinkedopenhashmap = this.queuesPerPriority.get(i);
+            ChunkPos j = long2objectlinkedopenhashmap.firstKey();
             List<Runnable> list = long2objectlinkedopenhashmap.removeFirst();
 
             while (this.hasWork() && this.queuesPerPriority.get(this.topPriorityQueueIndex).isEmpty()) {
@@ -87,6 +_,6 @@
         return this.name + " " + this.topPriorityQueueIndex + "...";
     }
 
-    public static record TasksForChunk(long chunkPos, List<Runnable> tasks) {
+    public static record TasksForChunk(ChunkPos chunkPos, List<Runnable> tasks) {
     }
 }
