--- a/net/minecraft/server/level/ServerPlayer.java
+++ b/net/minecraft/server/level/ServerPlayer.java
@@ -299,7 +_,7 @@
         BlockPos blockpos = p_9202_.getSharedSpawnPos();
         if (p_9202_.dimensionType().hasSkyLight() && p_9202_.getServer().getWorldData().getGameType() != GameType.ADVENTURE) {
             int i = Math.max(0, this.server.getSpawnRadius(p_9202_));
-            int j = Mth.floor(p_9202_.getWorldBorder().getDistanceToBorder((double)blockpos.getX(), (double)blockpos.getZ()));
+            int j = Mth.floor(p_9202_.getWorldBorder().getDistanceToBorder((double)blockpos.getBigX().doubleValue(), (double)blockpos.getBigZ().doubleValue()));
             if (j < i) {
                 i = j;
             }
@@ -318,7 +_,7 @@
                 int i2 = (k1 + j1 * l1) % i1;
                 int j2 = i2 % (i * 2 + 1);
                 int k2 = i2 / (i * 2 + 1);
-                BlockPos blockpos1 = PlayerRespawnLogic.getOverworldRespawnPos(p_9202_, blockpos.getX() + j2 - i, blockpos.getZ() + k2 - i);
+                BlockPos blockpos1 = PlayerRespawnLogic.getOverworldRespawnPos(p_9202_, blockpos.getBigX().add(j2 - i), blockpos.getBigZ().add(k2 - i));
                 if (blockpos1 != null) {
                     this.moveTo(blockpos1, 0.0F, 0.0F);
                     if (p_9202_.noCollision(this)) {
@@ -330,7 +_,7 @@
             this.moveTo(blockpos, 0.0F, 0.0F);
 
             while(!p_9202_.noCollision(this) && this.getY() < (double)(p_9202_.getMaxBuildHeight() - 1)) {
-                this.setPos(this.getX(), this.getY() + 1.0, this.getZ());
+                this.setPos(this.getBigX(), this.getY() + 1.0, this.getBigZ());
             }
         }
     }
@@ -364,7 +_,7 @@
         }
 
         if (p_9131_.contains("SpawnX", 99) && p_9131_.contains("SpawnY", 99) && p_9131_.contains("SpawnZ", 99)) {
-            this.respawnPosition = new BlockPos(p_9131_.getInt("SpawnX"), p_9131_.getInt("SpawnY"), p_9131_.getInt("SpawnZ"));
+            this.respawnPosition = new BlockPos(p_9131_.getByteArray("SpawnX"), p_9131_.getByteArray("SpawnY"), p_9131_.getByteArray("SpawnZ"));
             this.respawnForced = p_9131_.getBoolean("SpawnForced");
             this.respawnAngle = p_9131_.getFloat("SpawnAngle");
             if (p_9131_.contains("SpawnDimension")) {
@@ -409,9 +_,9 @@
         p_9197_.put("recipeBook", this.recipeBook.toNbt());
         p_9197_.putString("Dimension", this.level().dimension().location().toString());
         if (this.respawnPosition != null) {
-            p_9197_.putInt("SpawnX", this.respawnPosition.getX());
-            p_9197_.putInt("SpawnY", this.respawnPosition.getY());
-            p_9197_.putInt("SpawnZ", this.respawnPosition.getZ());
+            p_9197_.putByteArray("SpawnX", this.respawnPosition.getBigX().toByteArray());
+            p_9197_.putByteArray("SpawnY", this.respawnPosition.getBigY().toByteArray());
+            p_9197_.putByteArray("SpawnZ", this.respawnPosition.getBigZ().toByteArray());
             p_9197_.putBoolean("SpawnForced", this.respawnForced);
             p_9197_.putFloat("SpawnAngle", this.respawnAngle);
             ResourceLocation.CODEC
@@ -927,7 +_,7 @@
     protected PortalInfo findDimensionEntryPoint(ServerLevel p_8998_, boolean p_342019_) {
         PortalInfo portalinfo = super.findDimensionEntryPoint(p_8998_, p_342019_);
         if (portalinfo != null && this.level().dimension() == Level.OVERWORLD && p_8998_.dimension() == Level.END) {
-            Vec3 vec3 = portalinfo.pos.add(0.0, -1.0, 0.0);
+            me.alphamode.mcbig.core.BigVec3 vec3 = portalinfo.pos.add(0.0, -1.0, 0.0);
             return new PortalInfo(vec3, Vec3.ZERO, 90.0F, 0.0F);
         } else {
             return portalinfo;
@@ -1134,7 +_,7 @@
 
         super.stopSleepInBed(p_9165_, p_9166_);
         if (this.connection != null) {
-            this.connection.teleport(this.getX(), this.getY(), this.getZ(), this.getYRot(), this.getXRot());
+            this.connection.teleport(this.getBigX(), this.getY(), this.getBigZ(), this.getYRot(), this.getXRot());
         }
     }
 
@@ -1455,8 +_,14 @@
         this.connection.send(new ClientboundPlayerLookAtPacket(p_9112_, p_9113_.x, p_9113_.y, p_9113_.z));
     }
 
+    @Override
+    public void lookAt(EntityAnchorArgument.Anchor p_9112_, me.alphamode.mcbig.core.BigVec3 p_9113_) {
+        super.lookAt(p_9112_, p_9113_);
+        this.connection.send(new ClientboundPlayerLookAtPacket(p_9112_, p_9113_.x.doubleValue(), p_9113_.y, p_9113_.z.doubleValue()));
+    }
+
     public void lookAt(EntityAnchorArgument.Anchor p_9108_, Entity p_9109_, EntityAnchorArgument.Anchor p_9110_) {
-        Vec3 vec3 = p_9110_.apply(p_9109_);
+        Vec3 vec3 = p_9110_.apply(p_9109_).toVanilla();
         super.lookAt(p_9108_, vec3);
         this.connection.send(new ClientboundPlayerLookAtPacket(p_9108_, p_9109_, p_9110_));
     }
@@ -1536,12 +_,12 @@
     @Override
     public void teleportRelative(double p_251611_, double p_248861_, double p_252266_) {
         this.connection
-            .teleport(this.getX() + p_251611_, this.getY() + p_248861_, this.getZ() + p_252266_, this.getYRot(), this.getXRot(), RelativeMovement.ALL);
+            .teleport(this.getBigX().add(p_251611_), this.getY() + p_248861_, this.getBigZ().add(p_252266_), this.getYRot(), this.getXRot(), RelativeMovement.ALL);
     }
 
     @Override
     public boolean teleportTo(
-        ServerLevel p_265564_, double p_265424_, double p_265680_, double p_265312_, Set<RelativeMovement> p_265192_, float p_265059_, float p_265266_
+        ServerLevel p_265564_, me.alphamode.mcbig.math.BigDecimal p_265424_, double p_265680_, me.alphamode.mcbig.math.BigDecimal p_265312_, Set<RelativeMovement> p_265192_, float p_265059_, float p_265266_
     ) {
         ChunkPos chunkpos = new ChunkPos(BlockPos.containing(p_265424_, p_265680_, p_265312_));
         p_265564_.getChunkSource().addRegionTicket(TicketType.POST_TELEPORT, chunkpos, 1, this.getId());
@@ -1735,7 +_,7 @@
         if (entity != this.camera) {
             Level level = this.camera.level();
             if (level instanceof ServerLevel serverlevel) {
-                this.teleportTo(serverlevel, this.camera.getX(), this.camera.getY(), this.camera.getZ(), Set.of(), this.getYRot(), this.getXRot());
+                this.teleportTo(serverlevel, this.camera.getBigX(), this.camera.getY(), this.camera.getBigZ(), Set.of(), this.getYRot(), this.getXRot());
             }
 
             if (p_9214_ != null) {
@@ -1790,7 +_,7 @@
         return this.advancements;
     }
 
-    public void teleportTo(ServerLevel p_9000_, double p_9001_, double p_9002_, double p_9003_, float p_9004_, float p_9005_) {
+    public void teleportTo(ServerLevel p_9000_, me.alphamode.mcbig.math.BigDecimal p_9001_, double p_9002_, me.alphamode.mcbig.math.BigDecimal p_9003_, float p_9004_, float p_9005_) {
         this.setCamera(this);
         this.stopRiding();
         if (p_9000_ == this.level()) {
@@ -2015,7 +_,7 @@
             return false;
         } else {
             p_277395_.positionRider(this);
-            this.connection.teleport(this.getX(), this.getY(), this.getZ(), this.getYRot(), this.getXRot());
+            this.connection.teleport(this.getBigX(), this.getY(), this.getBigZ(), this.getYRot(), this.getXRot());
             if (p_277395_ instanceof LivingEntity livingentity) {
                 for(MobEffectInstance mobeffectinstance : livingentity.getActiveEffects()) {
                     this.connection.send(new ClientboundUpdateMobEffectPacket(p_277395_.getId(), mobeffectinstance, false));
