--- a/net/minecraft/server/level/ServerPlayer.java
+++ b/net/minecraft/server/level/ServerPlayer.java
@@ -360,8 +_,8 @@
                 int i2 = (k1 + j1 * l1) % i1;
                 int j2 = i2 % (i * 2 + 1);
                 int k2 = i2 / (i * 2 + 1);
-                int l2 = p_352202_.getX() + j2 - i;
-                int i3 = p_352202_.getZ() + k2 - i;
+                me.alphamode.mcbig.math.BigInteger l2 = p_352202_.getBigX().add(j2 - i);
+                me.alphamode.mcbig.math.BigInteger i3 = p_352202_.getBigZ().add(k2 - i);
 
                 try {
                     blockpos = PlayerRespawnLogic.getOverworldRespawnPos(p_352206_, l2, i3);
@@ -428,7 +_,7 @@
         }
 
         if (p_9131_.contains("SpawnX", 99) && p_9131_.contains("SpawnY", 99) && p_9131_.contains("SpawnZ", 99)) {
-            this.respawnPosition = new BlockPos(p_9131_.getInt("SpawnX"), p_9131_.getInt("SpawnY"), p_9131_.getInt("SpawnZ"));
+            this.respawnPosition = new BlockPos(p_9131_.getByteArray("SpawnX"), p_9131_.getByteArray("SpawnY"), p_9131_.getByteArray("SpawnZ"));
             this.respawnForced = p_9131_.getBoolean("SpawnForced");
             this.respawnAngle = p_9131_.getFloat("SpawnAngle");
             if (p_9131_.contains("SpawnDimension")) {
@@ -467,9 +_,9 @@
         p_9197_.put("recipeBook", this.recipeBook.toNbt());
         p_9197_.putString("Dimension", this.level().dimension().location().toString());
         if (this.respawnPosition != null) {
-            p_9197_.putInt("SpawnX", this.respawnPosition.getX());
-            p_9197_.putInt("SpawnY", this.respawnPosition.getY());
-            p_9197_.putInt("SpawnZ", this.respawnPosition.getZ());
+            p_9197_.putByteArray("SpawnX", this.respawnPosition.getBigX().toByteArray());
+            p_9197_.putByteArray("SpawnY", this.respawnPosition.getBigY().toByteArray());
+            p_9197_.putByteArray("SpawnZ", this.respawnPosition.getBigZ().toByteArray());
             p_9197_.putBoolean("SpawnForced", this.respawnForced);
             p_9197_.putFloat("SpawnAngle", this.respawnAngle);
             ResourceLocation.CODEC
@@ -982,7 +_,7 @@
             if (optional.isPresent()) {
                 ServerPlayer.RespawnPosAngle serverplayer$respawnposangle = optional.get();
                 return new TeleportTransition(
-                    serverlevel, serverplayer$respawnposangle.position(), Vec3.ZERO, serverplayer$respawnposangle.yaw(), 0.0F, p_379376_
+                    serverlevel, serverplayer$respawnposangle.position().toBig(), Vec3.ZERO, serverplayer$respawnposangle.yaw(), 0.0F, p_379376_
                 );
             } else {
                 return TeleportTransition.missingRespawnBlock(this.server.overworld(), this, p_379376_);
@@ -1201,7 +_,7 @@
 
         super.stopSleepInBed(p_9165_, p_9166_);
         if (this.connection != null) {
-            this.connection.teleport(this.getX(), this.getY(), this.getZ(), this.getYRot(), this.getXRot());
+            this.connection.teleport(this.getBigX(), this.getY(), this.getBigZ(), this.getYRot(), this.getXRot());
         }
     }
 
@@ -1503,7 +_,7 @@
     }
 
     public void lookAt(EntityAnchorArgument.Anchor p_9108_, Entity p_9109_, EntityAnchorArgument.Anchor p_9110_) {
-        Vec3 vec3 = p_9110_.apply(p_9109_);
+        Vec3 vec3 = p_9110_.apply(p_9109_).toVanilla();
         super.lookAt(p_9108_, vec3);
         this.connection.send(new ClientboundPlayerLookAtPacket(p_9108_, p_9109_, p_9110_));
     }
@@ -1592,7 +_,13 @@
     @Override
     public void teleportTo(double p_8969_, double p_8970_, double p_8971_) {
         this.connection
-            .teleport(new PositionMoveRotation(new Vec3(p_8969_, p_8970_, p_8971_), Vec3.ZERO, 0.0F, 0.0F), Relative.union(Relative.DELTA, Relative.ROTATION));
+                .teleport(new PositionMoveRotation(new Vec3(p_8969_, p_8970_, p_8971_), Vec3.ZERO, 0.0F, 0.0F), Relative.union(Relative.DELTA, Relative.ROTATION));
+    }
+
+    @Override
+    public void teleportTo(me.alphamode.mcbig.math.BigDecimal p_8969_, double p_8970_, me.alphamode.mcbig.math.BigDecimal p_8971_) {
+        this.connection
+            .teleport(new PositionMoveRotation(new me.alphamode.mcbig.core.BigVec3(p_8969_, p_8970_, p_8971_), Vec3.ZERO, 0.0F, 0.0F), Relative.union(Relative.DELTA, Relative.ROTATION));
     }
 
     @Override
@@ -1603,9 +_,9 @@
     @Override
     public boolean teleportTo(
         ServerLevel p_265564_,
-        double p_265424_,
+        me.alphamode.mcbig.math.BigDecimal p_265424_,
         double p_265680_,
-        double p_265312_,
+        me.alphamode.mcbig.math.BigDecimal p_265312_,
         Set<Relative> p_265192_,
         float p_265059_,
         float p_265266_,
@@ -1695,7 +_,7 @@
     public CommandSourceStack createCommandSourceStack() {
         return new CommandSourceStack(
             this.commandSource(),
-            this.position(),
+            this.bigPosition(),
             this.getRotationVector(),
             this.serverLevel(),
             this.getPermissionLevel(),
@@ -1833,7 +_,7 @@
         this.camera = (Entity)(p_9214_ == null ? this : p_9214_);
         if (entity != this.camera) {
             if (this.camera.level() instanceof ServerLevel serverlevel) {
-                this.teleportTo(serverlevel, this.camera.getX(), this.camera.getY(), this.camera.getZ(), Set.of(), this.getYRot(), this.getXRot(), false);
+                this.teleportTo(serverlevel, this.camera.getBigX(), this.camera.getY(), this.camera.getBigZ(), Set.of(), this.getYRot(), this.getXRot(), false);
             }
 
             if (p_9214_ != null) {
