--- a/net/minecraft/server/level/ThreadedLevelLightEngine.java
+++ b/net/minecraft/server/level/ThreadedLevelLightEngine.java
@@ -52,10 +_,10 @@
     public void checkBlock(BlockPos p_9357_) {
         BlockPos blockpos = p_9357_.immutable();
         this.addTask(
-            SectionPos.blockToSectionCoord(p_9357_.getX()),
-            SectionPos.blockToSectionCoord(p_9357_.getZ()),
-            ThreadedLevelLightEngine.TaskType.PRE_UPDATE,
-            Util.name(() -> super.checkBlock(blockpos), () -> "checkBlock " + blockpos)
+                SectionPos.blockToSectionCoord(p_9357_.getBigX()),
+                SectionPos.blockToSectionCoord(p_9357_.getBigZ()),
+                ThreadedLevelLightEngine.TaskType.PRE_UPDATE,
+                Util.name(() -> super.checkBlock(blockpos), () -> "checkBlock " + blockpos)
         );
     }
 
@@ -78,66 +_,66 @@
     @Override
     public void updateSectionStatus(SectionPos p_9364_, boolean p_9365_) {
         this.addTask(
-            p_9364_.x(),
-            p_9364_.z(),
-            () -> 0,
-            ThreadedLevelLightEngine.TaskType.PRE_UPDATE,
-            Util.name(() -> super.updateSectionStatus(p_9364_, p_9365_), () -> "updateSectionStatus " + p_9364_ + " " + p_9365_)
+                p_9364_.x(),
+                p_9364_.z(),
+                () -> 0,
+                ThreadedLevelLightEngine.TaskType.PRE_UPDATE,
+                Util.name(() -> super.updateSectionStatus(p_9364_, p_9365_), () -> "updateSectionStatus " + p_9364_ + " " + p_9365_)
         );
     }
 
     @Override
     public void propagateLightSources(ChunkPos p_285029_) {
         this.addTask(
-            p_285029_.x,
-            p_285029_.z,
-            ThreadedLevelLightEngine.TaskType.PRE_UPDATE,
-            Util.name(() -> super.propagateLightSources(p_285029_), () -> "propagateLight " + p_285029_)
+                p_285029_.x,
+                p_285029_.z,
+                ThreadedLevelLightEngine.TaskType.PRE_UPDATE,
+                Util.name(() -> super.propagateLightSources(p_285029_), () -> "propagateLight " + p_285029_)
         );
     }
 
     @Override
     public void setLightEnabled(ChunkPos p_9336_, boolean p_9337_) {
         this.addTask(
-            p_9336_.x,
-            p_9336_.z,
-            ThreadedLevelLightEngine.TaskType.PRE_UPDATE,
-            Util.name(() -> super.setLightEnabled(p_9336_, p_9337_), () -> "enableLight " + p_9336_ + " " + p_9337_)
+                p_9336_.x,
+                p_9336_.z,
+                ThreadedLevelLightEngine.TaskType.PRE_UPDATE,
+                Util.name(() -> super.setLightEnabled(p_9336_, p_9337_), () -> "enableLight " + p_9336_ + " " + p_9337_)
         );
     }
 
     @Override
     public void queueSectionData(LightLayer p_285046_, SectionPos p_285496_, @Nullable DataLayer p_285495_) {
         this.addTask(
-            p_285496_.x(),
-            p_285496_.z(),
-            () -> 0,
-            ThreadedLevelLightEngine.TaskType.PRE_UPDATE,
-            Util.name(() -> super.queueSectionData(p_285046_, p_285496_, p_285495_), () -> "queueData " + p_285496_)
+                p_285496_.x(),
+                p_285496_.z(),
+                () -> 0,
+                ThreadedLevelLightEngine.TaskType.PRE_UPDATE,
+                Util.name(() -> super.queueSectionData(p_285046_, p_285496_, p_285495_), () -> "queueData " + p_285496_)
         );
     }
 
-    private void addTask(int p_9313_, int p_9314_, ThreadedLevelLightEngine.TaskType p_9315_, Runnable p_9316_) {
-        this.addTask(p_9313_, p_9314_, this.chunkMap.getChunkQueueLevel(ChunkPos.asLong(p_9313_, p_9314_)), p_9315_, p_9316_);
+    private void addTask(me.alphamode.mcbig.math.BigInteger p_9313_, me.alphamode.mcbig.math.BigInteger p_9314_, ThreadedLevelLightEngine.TaskType p_9315_, Runnable p_9316_) {
+        this.addTask(p_9313_, p_9314_, this.chunkMap.getChunkQueueLevel(new ChunkPos(p_9313_, p_9314_)), p_9315_, p_9316_);
     }
 
-    private void addTask(int p_9318_, int p_9319_, IntSupplier p_9320_, ThreadedLevelLightEngine.TaskType p_9321_, Runnable p_9322_) {
+    private void addTask(me.alphamode.mcbig.math.BigInteger p_9318_, me.alphamode.mcbig.math.BigInteger p_9319_, IntSupplier p_9320_, ThreadedLevelLightEngine.TaskType p_9321_, Runnable p_9322_) {
         this.taskDispatcher.submit(() -> {
             this.lightTasks.add(Pair.of(p_9321_, p_9322_));
             if (this.lightTasks.size() >= 1000) {
                 this.runUpdate();
             }
-        }, ChunkPos.asLong(p_9318_, p_9319_), p_9320_);
+        }, new ChunkPos(p_9318_, p_9319_), p_9320_);
     }
 
     @Override
     public void retainData(ChunkPos p_9370_, boolean p_9371_) {
         this.addTask(
-            p_9370_.x,
-            p_9370_.z,
-            () -> 0,
-            ThreadedLevelLightEngine.TaskType.PRE_UPDATE,
-            Util.name(() -> super.retainData(p_9370_, p_9371_), () -> "retainData " + p_9370_)
+                p_9370_.x,
+                p_9370_.z,
+                () -> 0,
+                ThreadedLevelLightEngine.TaskType.PRE_UPDATE,
+                Util.name(() -> super.retainData(p_9370_, p_9371_), () -> "retainData " + p_9370_)
         );
     }
 
@@ -209,7 +_,7 @@
         }
     }
 
-    public CompletableFuture<?> waitForPendingTasks(int p_301257_, int p_300921_) {
+    public CompletableFuture<?> waitForPendingTasks(me.alphamode.mcbig.math.BigInteger p_301257_, me.alphamode.mcbig.math.BigInteger p_300921_) {
         return CompletableFuture.runAsync(() -> {
         }, p_300778_ -> this.addTask(p_301257_, p_300921_, ThreadedLevelLightEngine.TaskType.POST_UPDATE, p_300778_));
     }
