--- a/net/minecraft/server/level/WorldGenRegion.java
+++ b/net/minecraft/server/level/WorldGenRegion.java
@@ -104,13 +_,13 @@
     }
 
     @Override
-    public ChunkAccess getChunk(int p_9507_, int p_9508_) {
+    public ChunkAccess getChunk(me.alphamode.mcbig.math.BigInteger p_9507_, me.alphamode.mcbig.math.BigInteger p_9508_) {
         return this.getChunk(p_9507_, p_9508_, ChunkStatus.EMPTY);
     }
 
     @Nullable
     @Override
-    public ChunkAccess getChunk(int p_9514_, int p_9515_, ChunkStatus p_331853_, boolean p_9517_) {
+    public ChunkAccess getChunk(me.alphamode.mcbig.math.BigInteger p_9514_, me.alphamode.mcbig.math.BigInteger p_9515_, ChunkStatus p_331853_, boolean p_9517_) {
         int i = this.center.getPos().getChessboardDistance(p_9514_, p_9515_);
         ChunkStatus chunkstatus = i >= this.generatingStep.directDependencies().size() ? null : this.generatingStep.directDependencies().get(i);
         GenerationChunkHolder generationchunkholder;
@@ -144,7 +_,7 @@
     }
 
     @Override
-    public boolean hasChunk(int p_9574_, int p_9575_) {
+    public boolean hasChunk(me.alphamode.mcbig.math.BigInteger p_9574_, me.alphamode.mcbig.math.BigInteger p_9575_) {
         int i = this.center.getPos().getChessboardDistance(p_9574_, p_9575_);
         return i < this.generatingStep.directDependencies().size();
     }
@@ -176,7 +_,7 @@
     }
 
     @Override
-    public Holder<Biome> getUncachedNoiseBiome(int p_203787_, int p_203788_, int p_203789_) {
+    public Holder<Biome> getUncachedNoiseBiome(me.alphamode.mcbig.math.BigInteger p_203787_, me.alphamode.mcbig.math.BigInteger p_203788_, me.alphamode.mcbig.math.BigInteger p_203789_) {
         return this.level.getUncachedNoiseBiome(p_203787_, p_203788_, p_203789_);
     }
 
@@ -242,11 +_,11 @@
 
     @Override
     public boolean ensureCanWrite(BlockPos p_181031_) {
-        int i = SectionPos.blockToSectionCoord(p_181031_.getX());
-        int j = SectionPos.blockToSectionCoord(p_181031_.getZ());
+        me.alphamode.mcbig.math.BigInteger i = SectionPos.blockToSectionCoord(p_181031_.getBigX());
+        me.alphamode.mcbig.math.BigInteger j = SectionPos.blockToSectionCoord(p_181031_.getBigZ());
         ChunkPos chunkpos = this.getCenter();
-        int k = Math.abs(chunkpos.x - i);
-        int l = Math.abs(chunkpos.z - j);
+        int k = Math.abs(chunkpos.x.subtract(i).intValue());
+        int l = Math.abs(chunkpos.z.subtract(j).intValue());
         if (k <= this.generatingStep.blockStateWriteRadius() && l <= this.generatingStep.blockStateWriteRadius()) {
             if (this.center.isUpgrading()) {
                 LevelHeightAccessor levelheightaccessor = this.center.getHeightAccessorForGeneration();
@@ -293,9 +_,9 @@
                     }
                 } else {
                     CompoundTag compoundtag = new CompoundTag();
-                    compoundtag.putInt("x", p_9539_.getX());
-                    compoundtag.putInt("y", p_9539_.getY());
-                    compoundtag.putInt("z", p_9539_.getZ());
+                    compoundtag.putByteArray("x", p_9539_.getBigX().toByteArray());
+                    compoundtag.putByteArray("y", p_9539_.getBigY().toByteArray());
+                    compoundtag.putByteArray("z", p_9539_.getBigZ().toByteArray());
                     compoundtag.putString("id", "DUMMY");
                     chunkaccess.setBlockEntityNbt(compoundtag);
                 }
@@ -317,8 +_,8 @@
 
     @Override
     public boolean addFreshEntity(Entity p_9580_) {
-        int i = SectionPos.blockToSectionCoord(p_9580_.getBlockX());
-        int j = SectionPos.blockToSectionCoord(p_9580_.getBlockZ());
+        me.alphamode.mcbig.math.BigInteger i = SectionPos.blockToSectionCoord(p_9580_.getBlockX());
+        me.alphamode.mcbig.math.BigInteger j = SectionPos.blockToSectionCoord(p_9580_.getBlockZ());
         this.getChunk(i, j).addEntity(p_9580_);
         return true;
     }
@@ -361,11 +_,11 @@
 
     @Override
     public DifficultyInstance getCurrentDifficultyAt(BlockPos p_9585_) {
-        if (!this.hasChunk(SectionPos.blockToSectionCoord(p_9585_.getX()), SectionPos.blockToSectionCoord(p_9585_.getZ()))) {
-            throw new RuntimeException("We are asking a region for a chunk out of bound");
-        } else {
+//        if (!this.hasChunk(SectionPos.blockToSectionCoord(p_9585_.getBigX()), SectionPos.blockToSectionCoord(p_9585_.getBigZ()))) {
+//            throw new RuntimeException("We are asking a region for a chunk out of bound");
+//        } else {
             return new DifficultyInstance(this.level.getDifficulty(), this.level.getDayTime(), 0L, this.level.getMoonBrightness());
-        }
+//        }
     }
 
     @Nullable
@@ -405,8 +_,8 @@
     }
 
     @Override
-    public int getHeight(Heightmap.Types p_9535_, int p_9536_, int p_9537_) {
-        return this.getChunk(SectionPos.blockToSectionCoord(p_9536_), SectionPos.blockToSectionCoord(p_9537_)).getHeight(p_9535_, p_9536_ & 15, p_9537_ & 15)
+    public int getHeight(Heightmap.Types p_9535_, me.alphamode.mcbig.math.BigInteger p_9536_, me.alphamode.mcbig.math.BigInteger p_9537_) {
+        return this.getChunk(SectionPos.blockToSectionCoord(p_9536_), SectionPos.blockToSectionCoord(p_9537_)).getHeight(p_9535_, p_9536_.and(me.alphamode.mcbig.core.BigConstants.Ints.FIFTEEN), p_9537_.and(me.alphamode.mcbig.core.BigConstants.Ints.FIFTEEN))
             + 1;
     }
 
@@ -423,7 +_,7 @@
     }
 
     @Override
-    public void gameEvent(Holder<GameEvent> p_316792_, Vec3 p_215164_, GameEvent.Context p_215165_) {
+    public void gameEvent(Holder<GameEvent> p_215163_, me.alphamode.mcbig.core.BigVec3 p_215164_, GameEvent.Context p_215165_) {
     }
 
     @Override
