--- a/net/minecraft/client/renderer/chunk/RenderRegionCache.java
+++ b/net/minecraft/client/renderer/chunk/RenderRegionCache.java
@@ -12,24 +_,24 @@
 
 @OnlyIn(Dist.CLIENT)
 public class RenderRegionCache {
-    private final Long2ObjectMap<RenderRegionCache.ChunkInfo> chunkInfoCache = new Long2ObjectOpenHashMap<>();
+    private final it.unimi.dsi.fastutil.objects.Object2ObjectMap<ChunkPos, RenderRegionCache.ChunkInfo> chunkInfoCache = new it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap<>();
 
     @Nullable
     public RenderChunkRegion createRegion(Level p_200466_, SectionPos p_350879_) {
         RenderRegionCache.ChunkInfo renderregioncache$chunkinfo = this.getChunkInfo(p_200466_, p_350879_.x(), p_350879_.z());
-        if (renderregioncache$chunkinfo.chunk().isSectionEmpty(p_350879_.y())) {
+        if (renderregioncache$chunkinfo.chunk().isSectionEmpty(p_350879_.getY())) {
             return null;
         } else {
-            int i = p_350879_.x() - 1;
-            int j = p_350879_.z() - 1;
-            int k = p_350879_.x() + 1;
-            int l = p_350879_.z() + 1;
+            me.alphamode.mcbig.math.BigInteger i = p_350879_.x().subtract();
+            me.alphamode.mcbig.math.BigInteger j = p_350879_.z().subtract();
+            me.alphamode.mcbig.math.BigInteger k = p_350879_.x().add();
+            me.alphamode.mcbig.math.BigInteger l = p_350879_.z().add();
             RenderChunk[] arenderchunk = new RenderChunk[9];
 
-            for (int i1 = j; i1 <= l; i1++) {
-                for (int j1 = i; j1 <= k; j1++) {
+            for (me.alphamode.mcbig.math.BigInteger i1 = j; i1.compareTo(l) <= 0; i1 = i1.add()) {
+                for (me.alphamode.mcbig.math.BigInteger j1 = i; j1.compareTo(k) <= 0; j1 = j1.add()) {
                     int k1 = RenderChunkRegion.index(i, j, j1, i1);
-                    RenderRegionCache.ChunkInfo renderregioncache$chunkinfo1 = j1 == p_350879_.x() && i1 == p_350879_.z()
+                    RenderRegionCache.ChunkInfo renderregioncache$chunkinfo1 = j1.equals(p_350879_.x()) && i1.equals(p_350879_.z())
                         ? renderregioncache$chunkinfo
                         : this.getChunkInfo(p_200466_, j1, i1);
                     arenderchunk[k1] = renderregioncache$chunkinfo1.renderChunk();
@@ -40,11 +_,11 @@
         }
     }
 
-    private RenderRegionCache.ChunkInfo getChunkInfo(Level p_350834_, int p_350803_, int p_350907_) {
+    private RenderRegionCache.ChunkInfo getChunkInfo(Level p_350834_, me.alphamode.mcbig.math.BigInteger p_350803_, me.alphamode.mcbig.math.BigInteger p_350907_) {
         return this.chunkInfoCache
             .computeIfAbsent(
-                ChunkPos.asLong(p_350803_, p_350907_),
-                p_200464_ -> new RenderRegionCache.ChunkInfo(p_350834_.getChunk(ChunkPos.getX(p_200464_), ChunkPos.getZ(p_200464_)))
+                new ChunkPos(p_350803_, p_350907_),
+                p_200464_ -> new RenderRegionCache.ChunkInfo(p_350834_.getChunk(((ChunkPos)p_200464_).x, ((ChunkPos)p_200464_).z))
             );
     }
 
