--- a/net/minecraft/client/renderer/chunk/SectionRenderDispatcher.java
+++ b/net/minecraft/client/renderer/chunk/SectionRenderDispatcher.java
@@ -64,7 +_,7 @@
     private final TracingExecutor executor;
     ClientLevel level;
     final LevelRenderer renderer;
-    private Vec3 camera = Vec3.ZERO;
+    private me.alphamode.mcbig.core.BigVec3 camera = me.alphamode.mcbig.core.BigVec3.ZERO;
     final SectionCompiler sectionCompiler;
 
     public SectionRenderDispatcher(
@@ -138,11 +_,11 @@
         return this.bufferPool.getFreeBufferCount();
     }
 
-    public void setCamera(Vec3 p_296331_) {
+    public void setCamera(me.alphamode.mcbig.core.BigVec3 p_296331_) {
         this.camera = p_296331_;
     }
 
-    public Vec3 getCameraPosition() {
+    public me.alphamode.mcbig.core.BigVec3 getCameraPosition() {
         return this.camera;
     }
 
@@ -269,37 +_,37 @@
         private final Map<RenderType, VertexBuffer> buffers = RenderType.chunkBufferLayers()
             .stream()
             .collect(Collectors.toMap(p_295245_ -> (RenderType)p_295245_, p_373684_ -> new VertexBuffer(BufferUsage.STATIC_WRITE)));
-        private AABB bb;
+        private me.alphamode.mcbig.phys.BigAABB bb;
         private boolean dirty = true;
-        long sectionNode = SectionPos.asLong(-1, -1, -1);
+        SectionPos sectionNode = SectionPos.of(-1, -1, -1);
         final BlockPos.MutableBlockPos origin = new BlockPos.MutableBlockPos(-1, -1, -1);
         private boolean playerChanged;
 
-        public RenderSection(int p_295197_, long p_366428_) {
+        public RenderSection(int p_295197_, SectionPos p_366428_) {
             this.index = p_295197_;
             this.setSectionNode(p_366428_);
         }
 
-        private boolean doesChunkExistAt(long p_366835_) {
-            ChunkAccess chunkaccess = SectionRenderDispatcher.this.level.getChunk(SectionPos.x(p_366835_), SectionPos.z(p_366835_), ChunkStatus.FULL, false);
-            return chunkaccess != null && SectionRenderDispatcher.this.level.getLightEngine().lightOnInColumn(SectionPos.getZeroNode(p_366835_));
+        private boolean doesChunkExistAt(SectionPos p_366835_) {
+            ChunkAccess chunkaccess = SectionRenderDispatcher.this.level.getChunk(p_366835_.x(), p_366835_.z(), ChunkStatus.FULL, false);
+            return chunkaccess != null && SectionRenderDispatcher.this.level.getLightEngine().lightOnInColumn(p_366835_.zeroNode());
         }
 
         public boolean hasAllNeighbors() {
             int i = 24;
             return !(this.getDistToPlayerSqr() > 576.0)
                 ? true
-                : this.doesChunkExistAt(SectionPos.offset(this.sectionNode, Direction.WEST))
-                    && this.doesChunkExistAt(SectionPos.offset(this.sectionNode, Direction.NORTH))
-                    && this.doesChunkExistAt(SectionPos.offset(this.sectionNode, Direction.EAST))
-                    && this.doesChunkExistAt(SectionPos.offset(this.sectionNode, Direction.SOUTH))
-                    && this.doesChunkExistAt(SectionPos.offset(this.sectionNode, -1, 0, -1))
-                    && this.doesChunkExistAt(SectionPos.offset(this.sectionNode, -1, 0, 1))
-                    && this.doesChunkExistAt(SectionPos.offset(this.sectionNode, 1, 0, -1))
-                    && this.doesChunkExistAt(SectionPos.offset(this.sectionNode, 1, 0, 1));
+                : this.doesChunkExistAt(this.sectionNode.offset(Direction.WEST))
+                    && this.doesChunkExistAt(this.sectionNode.offset(Direction.NORTH))
+                    && this.doesChunkExistAt(this.sectionNode.offset(Direction.EAST))
+                    && this.doesChunkExistAt(this.sectionNode.offset(Direction.SOUTH))
+                    && this.doesChunkExistAt(this.sectionNode.offset(-1, 0, -1))
+                    && this.doesChunkExistAt(this.sectionNode.offset(-1, 0, 1))
+                    && this.doesChunkExistAt(this.sectionNode.offset(1, 0, -1))
+                    && this.doesChunkExistAt(this.sectionNode.offset(1, 0, 1));
         }
 
-        public AABB getBoundingBox() {
+        public me.alphamode.mcbig.phys.BigAABB getBoundingBox() {
             return this.bb;
         }
 
@@ -307,21 +_,21 @@
             return this.buffers.get(p_294497_);
         }
 
-        public void setSectionNode(long p_366482_) {
+        public void setSectionNode(SectionPos p_366482_) {
             this.reset();
             this.sectionNode = p_366482_;
-            int i = SectionPos.sectionToBlockCoord(SectionPos.x(p_366482_));
-            int j = SectionPos.sectionToBlockCoord(SectionPos.y(p_366482_));
-            int k = SectionPos.sectionToBlockCoord(SectionPos.z(p_366482_));
+            me.alphamode.mcbig.math.BigInteger i = SectionPos.sectionToBlockCoord(p_366482_.x());
+            int j = SectionPos.sectionToBlockCoord(p_366482_.getY());
+            me.alphamode.mcbig.math.BigInteger k = SectionPos.sectionToBlockCoord(p_366482_.z());
             this.origin.set(i, j, k);
-            this.bb = new AABB((double)i, (double)j, (double)k, (double)(i + 16), (double)(j + 16), (double)(k + 16));
+            this.bb = new me.alphamode.mcbig.phys.BigAABB(i.toBigDecimal(), (double)j, k.toBigDecimal(), (i.add(me.alphamode.mcbig.core.BigConstants.Ints.SIXTEEN)).toBigDecimal(), (double)(j + 16), (k.add(me.alphamode.mcbig.core.BigConstants.Ints.SIXTEEN)).toBigDecimal());
         }
 
         protected double getDistToPlayerSqr() {
             Camera camera = Minecraft.getInstance().gameRenderer.getMainCamera();
-            double d0 = this.bb.minX + 8.0 - camera.getPosition().x;
+            double d0 = this.bb.minX.add(me.alphamode.mcbig.core.BigConstants.EIGHT).subtract(camera.getPosition().x).doubleValue();
             double d1 = this.bb.minY + 8.0 - camera.getPosition().y;
-            double d2 = this.bb.minZ + 8.0 - camera.getPosition().z;
+            double d2 = this.bb.minZ.add(me.alphamode.mcbig.core.BigConstants.EIGHT).subtract(camera.getPosition().z).doubleValue();
             return d0 * d0 + d1 * d1 + d2 * d2;
         }
 
@@ -345,7 +_,7 @@
             return this.origin;
         }
 
-        public long getSectionNode() {
+        public SectionPos getSectionNode() {
             return this.sectionNode;
         }
 
@@ -368,8 +_,8 @@
             return this.dirty && this.playerChanged;
         }
 
-        public long getNeighborSectionNode(Direction p_366736_) {
-            return SectionPos.offset(this.sectionNode, p_366736_);
+        public SectionPos getNeighborSectionNode(Direction p_366736_) {
+            return this.sectionNode.offset(p_366736_.getStepX(), p_366736_.getStepY(), p_366736_.getStepZ());
         }
 
         public void resortTransparency(SectionRenderDispatcher p_294363_) {
@@ -399,7 +_,7 @@
 
         public SectionRenderDispatcher.RenderSection.CompileTask createCompileTask(RenderRegionCache p_295324_) {
             this.cancelTasks();
-            RenderChunkRegion renderchunkregion = p_295324_.createRegion(SectionRenderDispatcher.this.level, SectionPos.of(this.sectionNode));
+            RenderChunkRegion renderchunkregion = p_295324_.createRegion(SectionRenderDispatcher.this.level, this.sectionNode);
             boolean flag = this.compiled.get() != SectionRenderDispatcher.CompiledSection.UNCOMPILED;
             this.lastRebuildTask = new SectionRenderDispatcher.RenderSection.RebuildTask(renderchunkregion, flag);
             return this.lastRebuildTask;
@@ -435,9 +_,9 @@
         }
 
         VertexSorting createVertexSorting() {
-            Vec3 vec3 = SectionRenderDispatcher.this.getCameraPosition();
+            me.alphamode.mcbig.core.BigVec3 vec3 = SectionRenderDispatcher.this.getCameraPosition();
             return VertexSorting.byDistance(
-                (float)(vec3.x - (double)this.origin.getX()), (float)(vec3.y - (double)this.origin.getY()), (float)(vec3.z - (double)this.origin.getZ())
+                (float)(vec3.x.subtract(this.origin.getBigX().toBigDecimal())).floatValue(), (float)(vec3.y - (double)this.origin.getY()), (float)(vec3.z.subtract(this.origin.getBigZ().toBigDecimal())).floatValue()
             );
         }
 
@@ -628,15 +_,20 @@
         private int y;
         private int z;
 
-        public static SectionRenderDispatcher.TranslucencyPointOfView of(Vec3 p_371341_, long p_371400_) {
+        public static SectionRenderDispatcher.TranslucencyPointOfView of(me.alphamode.mcbig.core.BigVec3 p_371341_, SectionPos p_371400_) {
             return new SectionRenderDispatcher.TranslucencyPointOfView().set(p_371341_, p_371400_);
         }
 
-        public SectionRenderDispatcher.TranslucencyPointOfView set(Vec3 p_371567_, long p_371572_) {
-            this.x = getCoordinate(p_371567_.x(), SectionPos.x(p_371572_));
-            this.y = getCoordinate(p_371567_.y(), SectionPos.y(p_371572_));
-            this.z = getCoordinate(p_371567_.z(), SectionPos.z(p_371572_));
+        public SectionRenderDispatcher.TranslucencyPointOfView set(me.alphamode.mcbig.core.BigVec3 p_371567_, SectionPos p_371572_) {
+            this.x = getCoordinate(p_371567_.bigX(), p_371572_.x());
+            this.y = getCoordinate(p_371567_.y(), p_371572_.getY());
+            this.z = getCoordinate(p_371567_.bigZ(), p_371572_.z());
             return this;
+        }
+
+        private static int getCoordinate(me.alphamode.mcbig.math.BigDecimal p_371545_, me.alphamode.mcbig.math.BigInteger p_371464_) {
+            int i = SectionPos.blockToSectionCoord(p_371545_).subtract(p_371464_).intValue();
+            return Mth.clamp(i, -1, 1);
         }
 
         private static int getCoordinate(double p_371545_, int p_371464_) {
