--- a/net/minecraft/world/level/biome/Climate.java
+++ b/net/minecraft/world/level/biome/Climate.java
@@ -476,10 +_,10 @@
         DensityFunction weirdness,
         List<Climate.ParameterPoint> spawnTarget
     ) {
-        public Climate.TargetPoint sample(int p_186975_, int p_186976_, int p_186977_) {
-            int i = QuartPos.toBlock(p_186975_);
+        public Climate.TargetPoint sample(me.alphamode.mcbig.math.BigInteger p_186975_, int p_186976_, me.alphamode.mcbig.math.BigInteger p_186977_) {
+            me.alphamode.mcbig.math.BigInteger i = QuartPos.toBlock(p_186975_);
             int j = QuartPos.toBlock(p_186976_);
-            int k = QuartPos.toBlock(p_186977_);
+            me.alphamode.mcbig.math.BigInteger k = QuartPos.toBlock(p_186977_);
             DensityFunction.SinglePointContext densityfunction$singlepointcontext = new DensityFunction.SinglePointContext(i, j, k);
             return Climate.target(
                 (float)this.temperature.compute(densityfunction$singlepointcontext),
@@ -500,7 +_,7 @@
         Climate.SpawnFinder.Result result;
 
         SpawnFinder(List<Climate.ParameterPoint> p_207872_, Climate.Sampler p_207873_) {
-            this.result = getSpawnPositionAndFitness(p_207872_, p_207873_, 0, 0);
+            this.result = getSpawnPositionAndFitness(p_207872_, p_207873_, me.alphamode.mcbig.math.BigInteger.ZERO, me.alphamode.mcbig.math.BigInteger.ZERO);
             this.radialSearch(p_207872_, p_207873_, 2048.0F, 512.0F);
             this.radialSearch(p_207872_, p_207873_, 512.0F, 32.0F);
         }
@@ -511,8 +_,8 @@
             BlockPos blockpos = this.result.location();
 
             while(f1 <= p_207877_) {
-                int i = blockpos.getX() + (int)(Math.sin((double)f) * (double)f1);
-                int j = blockpos.getZ() + (int)(Math.cos((double)f) * (double)f1);
+                me.alphamode.mcbig.math.BigInteger i = blockpos.getBigX().add((int)(Math.sin((double)f) * (double)f1));
+                me.alphamode.mcbig.math.BigInteger j = blockpos.getBigZ().add((int)(Math.cos((double)f) * (double)f1));
                 Climate.SpawnFinder.Result climate$spawnfinder$result = getSpawnPositionAndFitness(p_207875_, p_207876_, i, j);
                 if (climate$spawnfinder$result.fitness() < this.result.fitness()) {
                     this.result = climate$spawnfinder$result;
@@ -527,11 +_,11 @@
         }
 
         private static Climate.SpawnFinder.Result getSpawnPositionAndFitness(
-            List<Climate.ParameterPoint> p_207880_, Climate.Sampler p_207881_, int p_207882_, int p_207883_
+            List<Climate.ParameterPoint> p_207880_, Climate.Sampler p_207881_, me.alphamode.mcbig.math.BigInteger p_207882_, me.alphamode.mcbig.math.BigInteger p_207883_
         ) {
             double d0 = Mth.square(2500.0);
             int i = 2;
-            long j = (long)((double)Mth.square(10000.0F) * Math.pow((double)(Mth.square((long)p_207882_) + Mth.square((long)p_207883_)) / d0, 2.0));
+            long j = (long)((double)Mth.square(10000.0F) * Math.pow((double)(Mth.square((long)p_207882_.longValue()) + Mth.square((long)p_207883_.longValue())) / d0, 2.0));
             Climate.TargetPoint climate$targetpoint = p_207881_.sample(QuartPos.fromBlock(p_207882_), 0, QuartPos.fromBlock(p_207883_));
             Climate.TargetPoint climate$targetpoint1 = new Climate.TargetPoint(
                 climate$targetpoint.temperature(),
