--- a/net/minecraft/world/entity/boss/enderdragon/EnderDragon.java
+++ b/net/minecraft/world/entity/boss/enderdragon/EnderDragon.java
@@ -624,7 +_,7 @@
     public int findClosestNode(double p_31171_, double p_31172_, double p_31173_) {
         float f = 10000.0F;
         int i = 0;
-        Node node = new Node(Mth.floor(p_31171_), Mth.floor(p_31172_), Mth.floor(p_31173_));
+        Node node = new Node(Mth.bigFloor(p_31171_), Mth.floor(p_31172_), Mth.bigFloor(p_31173_));
         int j = 0;
         if (this.dragonFight == null || this.dragonFight.getCrystalsAlive() == 0) {
             j = 12;
