--- a/net/minecraft/server/commands/ItemCommands.java
+++ b/net/minecraft/server/commands/ItemCommands.java
@@ -487,7 +_,7 @@
     private static ItemStack applyModifier(CommandSourceStack p_180284_, Holder<LootItemFunction> p_335682_, ItemStack p_180286_) {
         ServerLevel serverlevel = p_180284_.getLevel();
         LootParams lootparams = new LootParams.Builder(serverlevel)
-            .withParameter(LootContextParams.ORIGIN, p_180284_.getPosition())
+            .withParameter(LootContextParams.ORIGIN, p_180284_.getPosition().toVanilla())
             .withOptionalParameter(LootContextParams.THIS_ENTITY, p_180284_.getEntity())
             .create(LootContextParamSets.COMMAND);
         LootContext lootcontext = new LootContext.Builder(lootparams).create(Optional.empty());
