--- a/net/minecraft/server/commands/LootCommand.java
+++ b/net/minecraft/server/commands/LootCommand.java
@@ -397,10 +_,10 @@
         return list.size();
     }
 
-    private static int dropInWorld(CommandSourceStack p_137946_, Vec3 p_137947_, List<ItemStack> p_137948_, LootCommand.Callback p_137949_) throws CommandSyntaxException {
+    private static int dropInWorld(CommandSourceStack p_137946_, me.alphamode.mcbig.core.BigVec3 p_137947_, List<ItemStack> p_137948_, LootCommand.Callback p_137949_) throws CommandSyntaxException {
         ServerLevel serverlevel = p_137946_.getLevel();
         p_137948_.forEach(p_137884_ -> {
-            ItemEntity itementity = new ItemEntity(serverlevel, p_137947_.x, p_137947_.y, p_137947_.z, p_137884_.copy());
+            ItemEntity itementity = new ItemEntity(serverlevel, p_137947_.x.doubleValue(), p_137947_.y, p_137947_.z.doubleValue(), p_137884_.copy());
             itementity.setDefaultPickUpDelay();
             serverlevel.addFreshEntity(itementity);
         });
@@ -473,7 +_,7 @@
             lootparams$builder.withOptionalParameter(LootContextParams.DIRECT_KILLER_ENTITY, entity);
             lootparams$builder.withOptionalParameter(LootContextParams.KILLER_ENTITY, entity);
             lootparams$builder.withParameter(LootContextParams.THIS_ENTITY, p_137907_);
-            lootparams$builder.withParameter(LootContextParams.ORIGIN, commandsourcestack.getPosition());
+            lootparams$builder.withParameter(LootContextParams.ORIGIN, commandsourcestack.getPosition().toVanilla());
             LootParams lootparams = lootparams$builder.create(LootContextParamSets.ENTITY);
             LootTable loottable = commandsourcestack.getServer().getLootData().getLootTable(resourcelocation);
             List<ItemStack> list = loottable.getRandomItems(lootparams);
@@ -485,7 +_,7 @@
         CommandSourceStack commandsourcestack = p_137933_.getSource();
         LootParams lootparams = new LootParams.Builder(commandsourcestack.getLevel())
             .withOptionalParameter(LootContextParams.THIS_ENTITY, commandsourcestack.getEntity())
-            .withParameter(LootContextParams.ORIGIN, commandsourcestack.getPosition())
+            .withParameter(LootContextParams.ORIGIN, commandsourcestack.getPosition().toVanilla())
             .create(LootContextParamSets.CHEST);
         return drop(p_137933_, p_137934_, lootparams, p_137935_);
     }
