--- a/net/minecraft/server/commands/RaidCommand.java
+++ b/net/minecraft/server/commands/RaidCommand.java
@@ -111,8 +_,8 @@
     private static int playSound(CommandSourceStack p_180478_, @Nullable Component p_180479_) {
         if (p_180479_ != null && p_180479_.getString().equals("local")) {
             ServerLevel serverlevel = p_180478_.getLevel();
-            Vec3 vec3 = p_180478_.getPosition().add(5.0, 0.0, 0.0);
-            serverlevel.playSeededSound(null, vec3.x, vec3.y, vec3.z, SoundEvents.RAID_HORN, SoundSource.NEUTRAL, 2.0F, 1.0F, serverlevel.random.nextLong());
+            me.alphamode.mcbig.core.BigVec3 vec3 = p_180478_.getPosition().add(5.0, 0.0, 0.0);
+            serverlevel.playSeededSound(null, vec3.x.doubleValue(), vec3.y, vec3.z.doubleValue(), SoundEvents.RAID_HORN, SoundSource.NEUTRAL, 2.0F, 1.0F, serverlevel.random.nextLong());
         }
 
         return 1;
