--- a/net/minecraft/server/commands/TeleportCommand.java
+++ b/net/minecraft/server/commands/TeleportCommand.java
@@ -163,9 +_,9 @@
                 p_139033_,
                 entity,
                 (ServerLevel)p_139035_.level(),
-                p_139035_.getX(),
+                p_139035_.getBigX(),
                 p_139035_.getY(),
-                p_139035_.getZ(),
+                p_139035_.getBigZ(),
                 EnumSet.noneOf(RelativeMovement.class),
                 p_139035_.getYRot(),
                 p_139035_.getXRot(),
@@ -195,7 +_,7 @@
         @Nullable Coordinates p_139030_,
         @Nullable TeleportCommand.LookAt p_139031_
     ) throws CommandSyntaxException {
-        Vec3 vec3 = p_139029_.getPosition(p_139026_);
+        me.alphamode.mcbig.core.BigVec3 vec3 = p_139029_.getPosition(p_139026_);
         Vec2 vec2 = p_139030_ == null ? null : p_139030_.getRotation(p_139026_);
         Set<RelativeMovement> set = EnumSet.noneOf(RelativeMovement.class);
         if (p_139029_.isXRelative()) {
@@ -254,6 +_,10 @@
         return p_139027_.size();
     }
 
+    private static String formatDouble(me.alphamode.mcbig.math.BigDecimal p_142776_) {
+        return String.format(Locale.ROOT, "%f", p_142776_.getBacking());
+    }
+
     private static String formatDouble(double p_142776_) {
         return String.format(Locale.ROOT, "%f", p_142776_);
     }
@@ -262,9 +_,9 @@
         CommandSourceStack p_139015_,
         Entity p_139016_,
         ServerLevel p_139017_,
-        double p_139018_,
+        me.alphamode.mcbig.math.BigDecimal p_139018_,
         double p_139019_,
-        double p_139020_,
+        me.alphamode.mcbig.math.BigDecimal p_139020_,
         Set<RelativeMovement> p_139021_,
         float p_139022_,
         float p_139023_,
@@ -294,7 +_,7 @@
     }
 
     static class LookAt {
-        private final Vec3 position;
+        private final me.alphamode.mcbig.core.BigVec3 position;
         private final Entity entity;
         private final EntityAnchorArgument.Anchor anchor;
 
@@ -304,7 +_,7 @@
             this.position = p_139057_.apply(p_139056_);
         }
 
-        public LookAt(Vec3 p_139059_) {
+        public LookAt(me.alphamode.mcbig.core.BigVec3 p_139059_) {
             this.entity = null;
             this.position = p_139059_;
             this.anchor = null;
