--- a/net/minecraft/core/particles/VibrationParticleOption.java
+++ b/net/minecraft/core/particles/VibrationParticleOption.java
@@ -55,7 +_,7 @@
 
     @Override
     public String writeToString(HolderLookup.Provider p_323922_) {
-        Vec3 vec3 = this.destination.getPosition(null).get();
+        Vec3 vec3 = this.destination.getPosition(null).get().toVanilla();
         double d0 = vec3.x();
         double d1 = vec3.y();
         double d2 = vec3.z();
